---
title: "Integrating with TCRseq"
output: html_document
date: "2024-02-07"
---
# installing Monocle 3 dependency packages

``` {r}
BiocManager::install(c('BiocGenerics', 'DelayedArray', 'DelayedMatrixStats',
                       'limma', 'lme4', 'S4Vectors', 'SingleCellExperiment',
                       'SummarizedExperiment', 'batchelor', 'HDF5Array',
                       'terra', 'ggrastr'))
```


# Now install monocle 3

``` {r}
install.packages("devtools")
devtools::install_github('cole-trapnell-lab/monocle3')
require(monocle3)
require(Seurat)
install.packages("spatstat.utils")
```


# install Seurat/Wrapper

``` {r}
install.packages("R.utils")
remotes::install_github('satijalab/seurat-wrappers')
library(SeuratWrappers)
```

# Annotate all myeloid/T cell clusters! 

``` {r}
#load Seurat obj! 
library(Seurat)
library(tidyverse)
seurat.obj<- readRDS("seurat.obj.final.RDS")

#Dimplot ! 
umap<- DimPlot(seurat.obj, reduction = "umap", label = FALSE) +
  theme_classic() +
  NoLegend() + 
  theme(axis.title = element_text(size= 15, face = "bold"),
        axis.text = element_text(size = 14, face = "bold")) + 
  NoAxes()
umap
```

#feature plots of receptor/ligand for manuscript !! 

``` {r}
MIF<- FeaturePlot(seurat.obj, 
                  features = "MIF",
                  cols = c("lightgrey", "red")) + 
  NoAxes() + 
  theme(legend.position = "none") + 
  theme(legend.title = element_blank(), 
        plot.title = element_blank())
MIF
``` 
``` {r}
CCL5<- FeaturePlot(seurat.obj, 
                  features = "CCL5",
                  cols = c("lightgrey", "red")) + 
  NoAxes() + 
  theme(legend.title = element_blank(), 
        plot.title = element_blank())
CCL5
``` 
``` {r}
ggsave("legend.supfi.5.pdf", plot = CCL5, height=4, width =6, units = "in", dpi = 300)
``` 

``` {r}
CXCL13<- FeaturePlot(seurat.obj, 
                  features = "CXCL13",
                  cols = c("lightgrey", "red")) + 
  NoAxes() + 
  theme(legend.position = "none") + 
  theme(legend.title = element_blank(), 
        plot.title = element_blank())
CXCL13
```

``` {r}
CD74<- FeaturePlot(seurat.obj, 
                  features = "CD74",
                  cols = c("lightgrey", "red")) + 
  NoAxes() + 
  theme(legend.position = "none") + 
  theme(legend.title = element_blank(), 
        plot.title = element_blank())
CD74
```
``` {r}
CD44<- FeaturePlot(seurat.obj, 
                  features = "CD44",
                  cols = c("lightgrey", "red")) + 
  NoAxes() + 
  theme(legend.position = "none") + 
  theme(legend.title = element_blank(), 
        plot.title = element_blank())
CD44
```

``` {r}
CCR1<- FeaturePlot(seurat.obj, 
                  features = "CCR1",
                  cols = c("lightgrey", "red")) + 
  NoAxes() + 
  theme(legend.position = "none") + 
  theme(legend.title = element_blank(), 
        plot.title = element_blank())
CCR1
```
``` {r}
CCR5<- FeaturePlot(seurat.obj, 
                  features = "CCR5",
                  cols = c("lightgrey", "red")) + 
  NoAxes() + 
  theme(legend.position = "none") + 
  theme(legend.title = element_blank(), 
        plot.title = element_blank())
CCR5
```

# combine 
``` {r}
library(patchwork)
first7<- (MIF|CCL5|CXCL13|CD74|CD44|CCR1|CCR5)
ggsave("cellcommunicate.feature_1.pdf", plot = first7, height=2.5, width = 8, units = "in", dpi = 300)

```



``` {r}
CCR3<- FeaturePlot(seurat.obj, 
                  features = "CCR3",
                  cols = c("lightgrey", "red")) + 
  NoAxes() + 
  theme(legend.position = "none") + 
  theme(legend.title = element_blank(), 
        plot.title = element_blank())
CCR3
```



``` {r}
CXCR3<- FeaturePlot(seurat.obj, 
                  features = "CXCR3",
                  cols = c("lightgrey", "red")) + 
  NoAxes() + 
  theme(legend.position = "none") + 
  theme(legend.title = element_blank(), 
        plot.title = element_blank())
CXCR3
```
``` {r}
CXCR5<- FeaturePlot(seurat.obj, 
                  features = "CXCR5",
                  cols = c("lightgrey", "red")) + 
  NoAxes() + 
  theme(legend.position = "none") + 
  theme(legend.title = element_blank(), 
        plot.title = element_blank())
CXCR5
```
``` {r}
IL2<- FeaturePlot(seurat.obj, 
                  features = "IL2",
                  cols = c("lightgrey", "red")) + 
  NoAxes() + 
  theme(legend.position = "none") + 
  theme(legend.title = element_blank(), 
        plot.title = element_blank())
IL2
``` 

``` {r}
IL2RB<- FeaturePlot(seurat.obj, 
                  features = "IL2RB",
                  cols = c("lightgrey", "red")) + 
  NoAxes() + 
  theme(legend.position = "none") + 
  theme(legend.title = element_blank(), 
        plot.title = element_blank())
IL2RB
``` 

``` {r}
IL2RG<- FeaturePlot(seurat.obj, 
                  features = "IL2RG",
                  cols = c("lightgrey", "red")) + 
  NoAxes() + 
  theme(legend.position = "none") + 
  theme(legend.title = element_blank(), 
        plot.title = element_blank())
IL2RG
``` 

``` {r}
IL7<- FeaturePlot(seurat.obj, 
                  features = "IL7",
                  cols = c("lightgrey", "red")) + 
  NoAxes() + 
  theme(legend.position = "none") + 
  theme(legend.title = element_blank(), 
        plot.title = element_blank())
IL7
``` 


``` {r}
library(patchwork)
second7<- (CXCR5|CXCR3|CCR3|IL2|IL2RB|IL2RG|IL7)
ggsave("cellcommunicate.feature_2.pdf", plot = second7, height=2.5, width = 8, units = "in", dpi = 300)

```


``` {r}
IL7R<- FeaturePlot(seurat.obj, 
                  features = "IL7R",
                  cols = c("lightgrey", "red")) + 
  NoAxes() + 
  theme(legend.position = "none") + 
  theme(legend.title = element_blank(), 
        plot.title = element_blank())
IL7R
``` 

``` {r}
IL21<- FeaturePlot(seurat.obj, 
                  features = "IL21",
                  cols = c("lightgrey", "red")) + 
  NoAxes() + 
  theme(legend.position = "none") + 
  theme(legend.title = element_blank(), 
        plot.title = element_blank())
IL21
``` 

``` {r}
IL21R<- FeaturePlot(seurat.obj, 
                  features = "IL21R",
                  cols = c("lightgrey", "red")) + 
  NoAxes() + 
  theme(legend.position = "none") + 
  theme(legend.title = element_blank(), 
        plot.title = element_blank())
IL21R
``` 

``` {r}
IL15<- FeaturePlot(seurat.obj, 
                  features = "IL15",
                  cols = c("lightgrey", "red")) + 
  NoAxes() + 
  theme(legend.position = "none") + 
  theme(legend.title = element_blank(), 
        plot.title = element_blank())
IL15
``` 
``` {r}
IL15RA<- FeaturePlot(seurat.obj, 
                  features = "IL15RA",
                  cols = c("lightgrey", "red")) + 
  NoAxes() + 
  theme(legend.position = "none") + 
  theme(legend.title = element_blank(), 
        plot.title = element_blank())
IL15RA
``` 


``` {r}
library(patchwork)
last5<- (IL7R|IL21|IL21R|IL15|IL15RA)
ggsave("cellcommunicate.feature_3.pdf", plot = last5, height=2.5, width = 7, units = "in", dpi = 300)

```






# 2024 IOTOX
``` {r}
ggsave("umap.iotox.tiff", plot = umap, height=2, width = 2.5, units = "in", dpi = 300)
``` 


# Feature plots !!! 
``` {r}
FeaturePlot(seurat.obj, features = c("CD4", "CD8A", "KLRB1", "KLRG1"))
```



# Add metadata
``` {r}
view(seurat.obj@meta.data)
seurat.obj@meta.data$group <- ifelse(seurat.obj@meta.data$type %in% c("First arthritis", "Second arthritis"), "ICI-arthritis", "Osteoarthritis")
``` 


``` {r}
seurat.obj@meta.data$type<- factor(seurat.obj@meta.data$type, levels = c("Osteoarthritis", "First arthritis","Second arthritis"))

#Dimplot ! 
split_umap<- DimPlot(seurat.obj, reduction = "umap", label = FALSE, split.by = "type") +
  NoLegend() + 
  theme(axis.title = element_text(size= 15, face = "bold"),
        axis.text = element_text(size = 14, face = "bold")) + 
  NoAxes()
split_umap
``` 
# Save high quality plot
``` {r}
ggsave("umap.split.pdf", plot = split_umap, height=3, width = 5, units = "in", dpi = 300)

#2024 IOTOX
ggsave("umap.split.tiff", plot = split_umap, height=3, width = 5, units = "in", dpi = 300)

``` 


#subset only cluster 19 
``` {r}
Bcell<- subset(seurat.obj, subset = type %in% c("First arthritis", "Second arthritis"))
Bcell<- subset(Bcell, idents = c("19"))
view(Bcell@meta.data)
``` 


``` {r}
seurat.obj@meta.data$type<- factor(seurat.obj@meta.data$type, levels = c("Osteoarthritis", "First arthritis", "Second arthritis"))
VlnPlot(Bcell, features = c("MS4A1"), group.by = "type") +
  theme(axis.title.x  = element_blank(), 
        legend.position = "none", 
        axis.text.x  = element_blank(), 
        axis.title = element_text(face = "bold", size= 13)) 
```

# TNF
``` {r}
TNF<- VlnPlot(Bcell, features = c("TNF"), group.by = "type") +
  theme(axis.title.x  = element_blank(), 
        legend.position = "none", 
        axis.text.x  = element_blank(), 
        axis.title = element_text(face = "bold", size= 13)) 
TNF
``` 
# C79 ! CD79a and CD79b are components of the B cell receptor complex, crucial for antigen recognition 
``` {r}
CXCR5<- VlnPlot(Bcell, features = c("CXCR5"), group.by = "type") +
  theme(axis.title.x  = element_blank(), 
        legend.position = "none", 
        axis.text.x  = element_blank(), 
        axis.title = element_text(face = "bold", size= 13)) 
CXCR5

``` 

``` {r}
IGHM<- VlnPlot(Bcell, features = c("IGHM"), group.by = "type") +
  theme(axis.title.x  = element_blank(), 
        legend.position = "none", 
        axis.text.x  = element_blank(), 
        axis.title = element_text(face = "bold", size= 13)) 
IGHM
``` 

# CXCR5 ! 
``` {r}
VlnPlot(Bcell, features = c("CXCR5"), group.by = "type") +
  theme(axis.title.x  = element_blank(), 
        legend.position = "none", 
        axis.text.x  = element_blank(), 
        axis.title = element_text(face = "bold", size= 13)) 
``` 
# CD24 !! CXCR5 and CD38 for follicular B cells!! 
``` {r}
HLA<- VlnPlot(Bcell, features = c("HLA-DRB1"), group.by = "type") +
  theme(axis.title.x  = element_blank(), 
        legend.position = "none", 
        axis.text.x  = element_blank(), 
        axis.title = element_text(face = "bold", size= 13)) 
``` 
#combine the plots !! 
``` {r}
library(patchwork)
bcell_combined_plots<- (TNF|CXCR5|IGHM|HLA)
bcell_combined_plots
```

``` {r}
ggsave("bcell.features.pdf", plot = bcell_combined_plots, height=3, width = 6, units = "in", dpi = 300)
``` 





# CD38 
``` {r}
VlnPlot(Bcell, features = c("CD38"), group.by = "type") +
  theme(axis.title.x  = element_blank(), 
        legend.position = "none", 
        axis.text.x  = element_blank(), 
        axis.title = element_text(face = "bold", size= 13)) 


```









# CD30 and CD83 for activated B cells
``` {r}
VlnPlot(Bcell, features = c("CD38"), group.by = "type") +
  theme(axis.title.x  = element_blank(), 
        legend.position = "none", 
        axis.text.x  = element_blank(), 
        axis.title = element_text(face = "bold", size= 13)) 
``` 

# IRF8, BCL6, and CD22 for Germinal center B cells!! 

``` {r}
VlnPlot(Bcell, features = c("IRF8"), group.by = "type") +
  theme(axis.title.x  = element_blank(), 
        legend.position = "none", 
        axis.text.x  = element_blank(), 
        axis.title = element_text(face = "bold", size= 13)) 


``` 

# BCL6!! 

``` {r}
VlnPlot(Bcell, features = c("BCL6"), group.by = "type") +
  theme(axis.title.x  = element_blank(), 
        legend.position = "none", 
        axis.text.x  = element_blank(), 
        axis.title = element_text(face = "bold", size= 13)) 


``` 

# CD22 !! 

``` {r}
VlnPlot(Bcell, features = c("CD22"), group.by = "type") +
  theme(axis.title.x  = element_blank(), 
        legend.position = "none", 
        axis.text.x  = element_blank(), 
        axis.title = element_text(face = "bold", size= 13)) 


``` 

# plot all as dot plot!!! 
``` {r}
markers_bcell<- DotPlot(Bcell, features = genes, 
        dot.scale = 6, 
        split.by = "type", 
        cols="RdBu")  +
  geom_point(aes(size=pct.exp), shape = 21, colour="black", stroke=0.5) +
  scale_colour_viridis(option="inferno") +
  guides(size=guide_legend(override.aes=list(shape=21, colour="black", fill="white"))) +
  theme_classic() +
  RotatedAxis() + 
  coord_flip() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=15), 
        axis.text = element_text(size=8, face = "bold"), 
        axis.text.x = element_blank(), 
        axis.title = element_blank(),
        legend.title = element_text(size= 12, face = "bold"), 
        legend.text = element_text(size= 10, face = "bold")) 

markers_bcell
``` 

# save ggplot !! 
``` {r}
ggsave("Bcell_markers.pdf", plot = markers_bcell, height=6, width = 6, units = "in", dpi = 300)
``` 

#SITC postdoctoral fellowship application ! 

``` {r}
genes <- c("CD19","MS4A1","CD27","CD38", "CD28", "CD30", "CD83","IGHD","IGHM", "IGHG", "IGHE","PRDM1", "IRF4", "BCL6", "KMT2C", "POLH", "PAX5", "HLA-DRB1", "P2RY8", "GNA13", "SDC1", "PDCD1", "CTLA4","CXCR5", "ICOSLG", "TNFRSF1A", 
           "TNFRSF1B", "TNFRSF4","IFNGR1", "IFNGR2","TNFRSF5","CSF2RA", "CSF2RB", "IL17RA", "IL10", "IFNG", "IL4", "IL17A", "IL21", "TNF")

genes <- c("CD19","CD27","CD38", "IGHD","IGHM", "IGHG", "IGHE", "PAX5", "HLA-DRB1", "CXCR5" , "IL10", "IFNG", "IL4", "IL12", "IL17A",  "TNF")

genes<- c("FCRL3", "FCRL4", "FCRL5", "IL6", "IL10", "TNF", "IL23A", "IL12A", "CSF2", "TBX21", "CXCR3", "CD80", "CD86", "CD40", "ICOSLG")


#plot
bcell.dot_plot<- DotPlot(Bcell, features = genes, cols = c("blue", "red"), dot.scale = 6, group.by = "type") + 
  theme_classic() +
  RotatedAxis() + 
  coord_flip() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=15), 
        axis.title.y= element_blank(), 
        axis.text = element_text(), 
        axis.text.x = element_text(), 
        axis.title.x = element_blank(),
        legend.title = element_text(size= 14, face = "bold"), 
        legend.text = element_text(size= 14, face = "bold")) + 
  theme(legend.position = "right")
bcell.dot_plot
```
#save high quality plots !! 
``` {r}
ggsave("Bcell_markers.legend.pdf", plot = bcell.dot_plot, height=4, width = 4, units = "in", dpi = 300)

``` 

#view 
``` {r}
view(seurat.obj@meta.data)
table(seurat.obj@active.ident, seurat.obj@meta.data$orig.ident) #link: https://github.com/satijalab/seurat/issues/738 
```

# Select only first and second athritis ! 

``` {r}
view(seurat.obj@meta.data)
seurat.new<- subset(seurat.obj, subset = type %in% c("First arthritis", "Second arthritis"))

umap<- DimPlot(seurat.new, reduction = "umap",
        split.by = "type",
        label = FALSE) +
  theme_classic() +
  theme(axis.title = element_text(size= 15, face = "bold"),
        axis.text = element_text(size = 14, face = "bold")) + 
  NoAxes()
``` 


# save plot1 
``` {r}
ggsave("umap.pdf", plot = umap, height=5, width = 6, units = "in", dpi = 300)
```

# Umap 
``` {r}
#oder the level type! 
seurat.obj@meta.data$type<- factor(seurat.obj@meta.data$type, levels = c("First arthritis", "Second arthritis", "Osteoarthritis"))

#plot umap ! 
merged_umap<- DimPlot(seurat.new, 
        reduction = "umap", 
        group.by = "type", 
        label = FALSE, 
        ncol = 1) +
  theme_classic() +
  theme(axis.text = element_text(size = 14, face = "bold"),
        plot.title = element_blank(), 
        axis.title = element_text(size= 15, face = "bold")) +
  NoAxes() +
  NoLegend()
merged_umap
``` 


# save plot1 
``` {r}
ggsave("umap_merged.pdf", plot = merged_umap, height=5, width = 6, units = "in", dpi = 300)
```


# stacked violen plots of important markers! 
```  {r, fig.width= 5, fig.height= 5.5}
features <- c("PTPRC", "TRDV2","SAPRC", "CD3E", "CD3D", "CD4", "CD27", "PDCD1", "CTLA4", "CXCL","CXCL13","FOXP3", "CD8A", "CD69", "ITAGAE", "MKI67", "NKG7", "HLA-DRB1", "CLED9A","CD14"," FCGR3A","CD68", "SPP1", "SDC1", "CD19","IFNG", "IFNA", "IFNB", "IL4", "IL10", "IL17A", "IL21", "TNF", "GZMB", "GZMM", "GZMH", 
           "GZMK", "GNLY")

feature<- c("CXCL9", "CXCL10", "CXCL11")

#plot
ggplot<- VlnPlot(seurat.obj, features = features, stack = TRUE, flip = TRUE, adjust = 1) + 
  theme(legend.position = "none") + 
  RotatedAxis()

ggplot + theme(axis.text.x = element_text(angle = 90), 
               axis.text.y = element_text(size = 8), 
               axis.title.x = element_blank())
```

# Dotplot for cytokines! 
``` {r, fig.width= 9, fig.height= 4}
Idents(seurat.obj) <- factor(Idents(seurat.obj), levels = c("0", "1","2", "3", "4", "5", "6", "7",
                                                                  "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20"))
genes <- c("CD3E", "CD3D", "CD4", "CD8A", "IFNG", "IL4", "IL10", "IL17A", "IL21", "TGFB1","TNF", "GZMB", "GZMM", "GZMH", 
           "GZMK", "GNLY", "CSF2", "PRF1", "FOXP3")

# plot dotplot
DotPlot(seurat.obj, features = genes, cols = c("blue", "red"), dot.scale = 7) +
  theme_bw() + 
  coord_flip() +
  RotatedAxis() + 
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=15), 
        axis.title= element_text(size = 14, face = "bold"), 
        axis.text = element_text(size=10, face = "bold"), 
        axis.text.x = element_text()) + labs(y= "Cluster Identity")
``` 

*Annotation* 

*0= CD4 Central Memory*
*1= Classical Monocyte*
*2= Effector CD8*
*3= gdT*
*4= Treg* 
*5= CD1c+ Non-inflammatory DCs* 
*6= SPP1+ Macrophage*
*7= NKT* 
*8= NK*
*9= PD-1hi CXCL13hi CD4*
*10= Non-classical monocytes*
*11= Activated CD4*
*12= Trm effector CD8*
*13= MAIT*
*14= Cycling T*
*15= Neutrophil*
*16= CLEC9A+ DCs*
*17= pDC*
*18= Synovial cells*
*19= B cells*
*20= mDC*

``` {r}
all.annotated.subsets<- RenameIdents(seurat.obj, `0` ="CD4 Tcm",`11`="Recently activated CD4",  `9`= "PD1hi CXCL13hi CD4", `4`= "Treg", `2` ="Effector CD8", 
                                     `12`= "Trm effector CD8", `14`= "Cycing T",`13`= "MAIT",`3` ="Mixed T", `8`= "NK",`7`= "NK T",  `20`= "mDC",  `16`= "CLEC9A+ DCs", 
                                     `17`= " pDCs",  `5`= "Non-inflammatory DCs", `1` ="Classical mono", `10`= "Non-classical mono",`6`= "SPP1+ Mac", `15`= "Neutrophil",                                        `19`= "B cells", `18`= "Syn cells")

DimPlot(all.annotated.subsets)

# save obj 
saveRDS(all.annotated.subsets, file = "all.clusters.annotated.obj.rds")
```

``` {r, fig.width= 13, fig.height= 4}
# Dimplot 1
dim1<- DimPlot(seurat.obj, reduction = "umap", 
        label = TRUE, 
        label.size = 3.5) + 
  theme_minimal() + 
  theme(legend.text = element_text(size= 8)) + 
  NoLegend()

#Dimplot 2
dim2<- DimPlot(all.annotated.subsets, 
        reduction = "umap", 
        label = FALSE, 
        label.size = 3.5) +
  theme_minimal() + 
  theme(legend.text = element_text(size= 8))
dim1|dim2
``` 

# Main markers !=== 
``` {r, fig.width= 5, fig.height= 8}
features <- c("PTPRC", "TRDV2", "CD3E", "CD3D", "CD4", "CD27", "PDCD1", "CTLA4", "CXCL13","FOXP3", "CD8A", "CD69", "ITAGAE", "MKI67", "NKG7", "HLA-DRB1", "CLED9A","CD14"," FCGR3A","CD68", "SPP1","ITGAX", "SDC1", "CD19", "IL12A", "IL12B", "IL12RB1", "IL12RB1", "TBX21", "FCRL5")

  stacked_violen<- VlnPlot(all.annotated.subsets, 
                           features = feature, 
                           stack = TRUE, 
                           flip = TRUE, 
                           adjust = 0.5) &
  theme(legend.position = "none",
        axis.text.x =  element_blank(),
        axis.text = element_text(size = 10)) + 
  RotatedAxis() 

stacked_all<- stacked_violen + 
  theme(axis.text.x = element_text(angle=90,hjust=0.95,vjust=0.2, face = "bold"), 
               axis.text.y = element_text(size = 8), 
               axis.title.x = element_blank(), 
        legend)
stacked_all

#save pdf 
ggsave("stacked_violen.pdf", plot = stacked_all, height=7, width = 5, units = "in", dpi = 300)
```


#Dot plot of featureplot!! === for publication ===!!! 
``` {r}
genes <- c("GZMK","GZMH","GZMM","GZMB","IL2","IL1B","IL1A","TGFB1","IL21","IL17A","IL10","IL4","TNF","IFNG","IL12B","IL12A","CCR7","CCR6","CCR5","CCR4","CCR3","CCR1","CCL3","CCL4","CCL5","MIF","CXCR3","CX3CR1","CXCL13","FCRL5","TBX21","CD19","SDC1","ITGAX","SPP1","CD68","CD14","HLA-DRB1","NKG7","MKI67","CD69","CD8A","FOXP3","HAVCR2","LAG3","TGIT","CTLA4","PDCD1","CD27","TOX","TCF7","CD4","CD3E","CD3D")


#Dot plot!
install.packages("viridis")
library(viridis)
Dot.plot<- DotPlot(all.annotated.subsets, features = genes, dot.scale = 4.5) + 
 geom_point(aes(size=pct.exp), shape = 21, colour="black", stroke=0.5) +
  scale_colour_viridis(option="inferno") +
  guides(size=guide_legend(override.aes=list(shape=21, colour="black", fill="white"))) +
  RotatedAxis() + 
  coord_flip() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=15), 
        axis.title.y= element_blank(), 
        axis.text = element_text(size=10, face = "bold"), 
        axis.text.x = element_text(angle=90,hjust=0.95,vjust=0.2, face = "bold", size= 10), 
        axis.title.x = element_blank(),
        legend.title = element_text(size= 12, face = "bold"), 
        legend.text = element_text(size= 10, face = "bold"), 
        legend.position = "right") 
Dot.plot
```

#save pdf 
```{r}
ggsave("dot.cytokine.violen.pdf", plot = Dot.plot, height=9, width = 6, units = "in", dpi = 300)
``` 

# Cytokine expression 
``` {r, fig.width= 7, fig.height= 6.5}
genes <- c("CD3E", "CD3D", "CD4", "CD8A", "IFNG", "IL4", "IL10", "IL17A", "IL21", "TNF", "GZMB", "GZMM", "GZMH", 
           "GZMK", "GNLY", "CSF2", "PRF1", "FOXP3")

#plot
Dot.plot<- DotPlot(all.annotated.subsets, features = features, cols = c("blue", "red"), dot.scale = 6) + 
  theme_bw() +
  RotatedAxis() + 
  coord_flip() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=15), 
        axis.title.y= element_text(size = 14, face = "bold"), 
        axis.text = element_text(size=13, face = "bold"), 
        axis.text.x = element_text(angle=90,hjust=0.95,vjust=0.2, face = "bold", size= 12), 
        axis.title.x = element_blank(),
        legend.title = element_text(size= 12, face = "bold"), 
        legend.text = element_text(size= 10, face = "bold")) 
Dot.plot

```


# Feature plots! 

``` {r, fig.height= 7, fig.width= 7}
FeaturePlot(seurat.obj, features = c("SELL", "TCF7", "PDCD1", "CXCL13", "TOX", "CXCR5", "CCR5"), pt.size = 0.2,
    ncol = 2, label = TRUE, label.size = 3)
``` 


``` {r, fig.height= 10, fig.width= 7}
FeaturePlot(seurat.obj, features = c("CXCL9", "CXCL10", "CXCL11", "CCL5", "CCR5", "CCL4", "CCR1", "CCR3"), pt.size = 0.2,
    ncol = 2, label = TRUE, label.size = 3)
``` 

``` {r, }
FeaturePlot(seurat.obj, features = c("TNF", "IFNG"), pt.size = 0.2,
            label = TRUE, label.size = 3)
``` 


# Examine differentially expressed genes between first/second arthritis
``` {r}
# Identify differentially expressed genes across conditions
view (all.annotated.subsets@meta.data)
Idents(all.annotated.subsets)
all.annotated.subsets$celltypes<-Idents(all.annotated.subsets)
view (all.annotated.subsets@meta.data)
all.annotated.subsets$arthritis_celltype <- paste(all.annotated.subsets$type, sep = "_", Idents(all.annotated.subsets))
view (all.annotated.subsets@meta.data)
Idents(all.annotated.subsets) <- "arthritis_celltype"
Idents(all.annotated.subsets)

#Calculate differentially expressed genes between first and second arthritis! 
prepsct<- PrepSCTFindMarkers(all.annotated.subsets)
deg.bcell<- FindMarkers(prepsct,
                               assay = "SCT", 
                               ident.1 = "Second arthritis_B cells", 
                               ident.2 = "First arthritis_B cells", 
                               min.pct = 0.25, 
                               logfc.threshold = 0.25, 
                               verbose = FALSE)
head(deg.bcell, n = 20)
``` 
``` {r}
deg.bcell.save<- deg.bcell %>% arrange(desc(avg_log2FC))
head(deg.bcell.save)
``` 
``` {r}
write.csv(deg.bcell.save, file = "deg.bcell secondvsfirst.csv")

```

#deg plot for volcano! 
``` {r}
## Volcano plot @ 1 day 

volcano.b<- deg.bcell [order(deg.bcell $p_val_adj),]

# create data.frame
results.b<- as.data.frame(mutate(as.data.frame(volcano.b), 
                                    cut_off=ifelse(volcano.b$p_val_adj<0.05, "significant", "insignificant")), 
                             row.names=rownames(volcano.b))
## ggplot plus ggrepel
options(ggrepel.max.overlaps = Inf)
library(ggrepel)
volcano.b$cut_off<- factor(results.b$cut_off, levels = c('significant', 'insignificant'))

#ggplot !
vocalno_bcell<- ggplot(results.b, aes(avg_log2FC, -log10(p_val_adj))) + 
  theme_classic() +
  geom_point(aes(col = cut_off)) +
  scale_color_manual(values = c("black", "red")) +
  theme(plot.title = element_text(hjust = 0.5, size = 8, face = "bold"), 
        axis.title = element_text(face = "bold", size = 12), 
        axis.text = element_text(face = "bold", size=8), 
        legend.title = element_text(size = 8, face = "bold"), 
        legend.text = element_text(size = 6), 
        legend.position = c(0.2, 0.8)) + 
  geom_text_repel(data=results.b[1:15,], 
                  aes(label=rownames(results.b[1:15,])), size= 2) +
 ggtitle("Second vs. first arthritis (B cell)")
vocalno_bcell

``` 
# save high quality plot! 
``` {r}
ggsave("volcano.bcell.pdf", plot = vocalno_bcell, height=2.8, width = 3.25, units = "in", dpi = 300)
```

# plot enrichment 
``` {r}
library(tidyverse)

## 3 hrs
gseabcell<- read.csv("gsea_bcell.csv", stringsAsFactors = TRUE, header = TRUE)

# ggplot
gseabcell_plot<- ggplot(gseabcell, aes(reorder(Pathway, NES), NES, fill= padj)) + 
  theme_classic() +
  geom_bar(stat = "identity")  + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 8),
        axis.title.x = element_text(face = "bold", size = 12),
        axis.title.y  = element_blank(), 
        axis.text = element_text(face = "bold", size = 10), 
        plot.title = element_text(hjust = 0.5, face = "bold", size = 10)) + 
  labs(x="Hallmark pathways", y="normalized enrichment score",
       title="Second vs. first arthritis (B cell)") + 
  coord_flip() + 
  scale_fill_gradient(low = "red", high = "blue")
gseabcell_plot
``` 

``` {r}
ggsave("gseabcell_barplot.pdf", plot = gseabcell_plot, height=3, width =6.5, units = "in", dpi = 300)
``` 


## perform gene set enrichment 

``` {r}
BiocManager::install("fgsea")
library(fgsea)
deg.bcell$gene<- rownames(deg.bcell)
deg.bcell<- deg.bcell %>% arrange(desc(avg_log2FC))

#write csv file for deg
write.csv(DEG.activatedCD4, file = "Y:/Recurrent__Arthritis__Single cell/DEG.activatedCD4.csv")

fold_changes<- deg.bcell$avg_log2FC
names(fold_changes)<- deg.bcell$gene
``` 

#Load geneset ! 
``` {r}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("fgsea")
library(fgsea)
Reactome <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.reactome.v2023.1.Hs.symbols.gmt")
hallmark <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/h.all.v2023.1.Hs.symbols.gmt")
KEGG <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.kegg.v2023.1.Hs.symbols.gmt")
GO <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c5.go.bp.v2023.1.Hs.symbols.gmt")

#combine the four gsea
Four_gene_sets <- c(hallmark, KEGG, GO, Reactome)

#getset for all pathway 
Allgenesets <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/msigdb.v2023.1.Hs.symbols.gmt")

# GSEA analysis
gsea_Myo_bcell <- fgsea(pathways = hallmark,
                  stats = fold_changes,
                  eps = 0.0,
                  minSize=15,
                  maxSize=500)

head(gsea_Myo_bcell[order(pval), ]) 
```
# make a table plot for a bunch of selected pathways:

``` {r, fig.width= 7, fig.height= 3}
topPathwaysUp <- gsea_Myo[ES > 0][head(order(pval), n=10), pathway]
topPathwaysDown <- gsea_Myo[ES < 0][head(order(pval), n=10), pathway]
topPathways <- c(topPathwaysUp, rev(topPathwaysDown))
table<- plotGseaTable(GO [topPathways], fold_changes, gsea_Myo, 
              gseaParam=0.5)
print(table)
``` 

# make an enrichment plot for a pathway:
``` {r}
plotEnrichment(hallmark[["HALLMARK_TNFA_SIGNALING_VIA_NFKB"]],
               fold_changes) + labs(title="TNFA_Singlalling via NFKB")
``` 

#save pdf 
``` {r}
ggsave("dot.plot.pdf", plot = Dot.plot, height=8.5, width = 6.5, units = "in", dpi = 300)
``` 

#Cluster Frequency ======! https://rpubs.com/olneykimberly/LPS_brain_neuron_recluster !!! useful codes !! 

``` {r}
#view seurat obj
view(all.annotated.subsets)

#set file path and create directory 
figurePath <- file.path("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis")
if(!dir.exists(figurePath)){
    dir.create("cluster_frequency", recursive = T)
}

#set working directory if needed!
setwd(figurePath)

# patient 164S
p.164s <- subset(all.annotated.subsets, subset = orig.ident == "164S")
view(p.164s)

# cluster_frequency
cluster_frequency.164s <- p.164s@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.164s")

#write csv file to save the result
write.csv(cluster_frequency.164s, file="cluster_frequency/cluster_frequency.164S.CSV")

# patient 164S2
p.164S2 <- subset(all.annotated.subsets, subset = orig.ident == "164S2")
view(p.164S2)

# cluster_frequency
cluster_frequency.164S2 <- p.164S2@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.164S2")

write csv file to save the result
write.csv(cluster_frequency.164S2, file="cluster_frequency/cluster_frequency.164S2.CSV")


# patient 164S2
p.184S2 <- subset(all.annotated.subsets, subset = orig.ident == "184S2")
view(p.184S2)

# cluster_frequency
cluster_frequency.184S2 <- p.184S2@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.184S2")

#write csv file to save the result
write.csv(cluster_frequency.184S2, file="cluster_frequency/cluster_frequency.184S2.CSV")

# patient 164S2
p.184S3 <- subset(all.annotated.subsets, subset = orig.ident == "184S3")
view(p.184S3)

# cluster_frequency
cluster_frequency.184S3 <- p.184S3@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.184S3")

#write csv file to save the result
write.csv(cluster_frequency.184S3, file="cluster_frequency/cluster_frequency.184S3.CSV")


# patient 164S2
p.218S <- subset(all.annotated.subsets, subset = orig.ident == "218S")
view(p.218S)

# cluster_frequency
cluster_frequency.218S <- p.218S@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.218S")

#write csv file to save the result
write.csv(cluster_frequency.218S, file="cluster_frequency/cluster_frequency.218S.CSV")

 #patient 164S2
p.218S2 <- subset(all.annotated.subsets, subset = orig.ident == "218S2")
view(p.218S2)

# cluster_frequency
cluster_frequency.218S2 <- p.218S2@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.218S2")

#write csv file to save the result
write.csv(cluster_frequency.218S2, file="cluster_frequency/cluster_frequency.218S2.CSV")


# patient SA5
p.SA5 <- subset(all.annotated.subsets, subset = orig.ident == "SA5")
view(p.SA5)

# cluster_frequency
cluster_frequency.SA5 <- p.SA5@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.SA5")

#write csv file to save the result
write.csv(cluster_frequency.SA5, file="cluster_frequency/cluster_frequency.SA5.CSV")

# patient SA6
p.SA6 <- subset(seurat.obj.annota, subset = orig.ident == "SA6")
view(p.SA6)

# cluster_frequency
cluster_frequency.SA6 <- p.SA6@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.SA6")

#write csv file to save the result
write.csv(cluster_frequency.SA6, file="cluster_frequency/cluster_frequency.SA6.CSV")

# patient SA7
p.SA7 <- subset(all.annotated.subsets, subset = orig.ident == "SA7")
view(p.SA7)

# cluster_frequency
cluster_frequency.SA7 <- p.SA7@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.SA7")

#write csv file to save the result
write.csv(cluster_frequency.SA7, file="cluster_frequency/cluster_frequency.SA7.CSV")
``` 


#plot cluster frequency 
``` {r}
#read csv file back in 
require(tidyverse)
proportion<- read.csv("cluster_frequency/cluster_frequency.csv", stringsAsFactors = TRUE)
proportion
``` 

## cluster proportion
``` {r}
CD4.ctm<- proportion %>% filter(cluster_name == "CD4 central memory")
CD4.ctm$Group<- factor(CD4.ctm$Group, levels = c("osteoarthritis", "ICI_arthritis"))
CD4.ctm$Group<- factor(CD4.ctm$Group, levels = c("osteoarthritis", "ICI-arthritis"))
CD4.ctm$Arthritis<- factor(CD4.ctm$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))

#visualization ! 
p1<- ggplot(CD4.ctm, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% of Total Cells", title = "CD4 Tcm")
p1
``` 
#idependent T test
``` {r}
t.test(frequency ~ Group, data=cd4tcm)

``` 


# Mann-Whitney U Test
``` {r}
wilcox.test(frequency ~ Group, data=cd4tcm)
``` 


#one way analysis of variance! 
``` {r}
#parametric anova
CD4.ctm<- aov(frequency ~ Group, data=cd4tcm)
summary(CD4.ctm)
TukeyHSD(CD4.ctm)  #non-significant
```

``` {r}
#non-parametric anova 
kruskal.test(frequency ~ Group, data=cd4tcm)
``` 




``` {r}
ActivatedCD4<- proportion %>% filter(cluster_name == "recently activated CD4")
ActivatedCD4$Group<- factor(ActivatedCD4$Group, levels = c("osteoarthritis", "ICI_arthritis"))
ActivatedCD4$Arthritis<- factor(ActivatedCD4$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))

#visualization ! 
p2<- ggplot(ActivatedCD4, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=4.3
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% Activated CD4 of total cells", title = "recently activated CD4")
p2
``` 


#one way analysis of variance! 
``` {r}
#parametric anova
activated.cd4<- aov(frequency ~ type, data=ActivatedCD4)
summary(activated.cd4)
TukeyHSD(activated.cd4)  #non-significant

#non-parametric anova 
kruskal.test(frequency ~ type, data=ActivatedCD4)
``` 
#both test shows non-significant! 

#previous version 
``` {r}
TPH<- proportion %>% filter(cluster_name == "PD-1hi CXCL13hi CD4")
TPH$Group<- factor(TPH$Group, levels = c("osteoarthritis", "ICI_arthritis"))
TPH$Arthritis<- factor(TPH$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))


#visualization ! 
p3<- ggplot(TPH, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5
  ), axis.title.y =  element_text(size = 8, face = "bold"), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% of Total Cells", title = "PD1hi CXCL13hi CD4")
p3
``` 
#idependent T test
``` {r}
t.test(frequency ~ Group, data=TPH)
``` 

# Mann-Whitney U Test

``` {r}
wilcox.test(frequency ~ Group, data=TPH)
``` 


``` {r}
MAIT<- proportion %>% filter(cluster_name == "MAIT")
MAIT$Group<- factor(MAIT$Group, levels = c("osteoarthritis", "ICI_arthritis"))
MAIT$Arthritis<- factor(MAIT$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))

#visualization ! 
p4<- ggplot(MAIT, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_text(angle = 90, face = "bold", vjust = 0.5, hjust = 0.5), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% MAIT of total cells", title = "MAIT")
p4
``` 
``` {r}
Treg<- proportion %>% filter(cluster_name == "Treg")
Treg$Group<- factor(Treg$Group, levels = c("osteoarthritis", "ICI_arthritis"))
Treg$Arthritis<- factor(Treg$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))


#visualization ! 
p5<- ggplot(Treg, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% Treg of total cells", title = "Treg")
p5
``` 



``` {r}
#visualization ! 
Treg.f.s<- Treg %>% 
  filter(Arthritis %in% c("first arthritis", "second arthritis"))
# ggplot 
tregse_vs_fi <- ggplot(Treg.f.s, aes(x = Arthritis, 
                                     y = frequency, 
                                     shape = Arthritis, 
                                     fill = Arthritis)) + 
  theme_classic() +
  geom_boxplot(outlier.shape = NA, 
               alpha=0.5) + # Suppress outliers from the boxplot to avoid overlap with points
  geom_point(position = position_jitter(width = 0.2), size = 3, alpha = 0.7) + # Add jittered points
  theme(plot.title = element_blank(),
        axis.title.y = element_text(size = 13, face = "bold"), 
        axis.title.x = element_blank(),
        axis.text = element_text(size = 13, face = "bold"), 
        axis.text.x = element_blank()) + 
  theme(legend.position = "none") + 
  ylab("% of total cells") + 
  ggtitle("Regulatory T")
tregse_vs_fi

``` 

# save violen plot !!! 
``` {r}
ggsave("tregcell_second_vs_first.pdf", plot = tregse_vs_fi, height=3, width = 2.5, units = "in", dpi = 300)

``` 






#idependent T test
``` {r}
t.test(frequency ~ Group, data=Treg)
``` 
``` {r}
wilcox.test(frequency ~ Group, data=Treg)
```

``` {r}
effector.cd8<- proportion %>% filter(cluster_name == "Effector CD8")
effector.cd8$Group<- factor(effector.cd8$Group, levels = c("osteoarthritis", "ICI_arthritis"))
effector.cd8$Arthritis<- factor(effector.cd8$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))


#visualization ! 
p6<- ggplot(effector.cd8, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
 theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% of Total Cells", title = "Effector CD8")
p6
``` 
#idependent T test
``` {r}
t.test(frequency ~ Group, data=effector.cd8)
```
``` {r}
wilcox.test(frequency ~ Group, data=effector.cd8)
```

``` {r}
TRM.cd8<- proportion %>% filter(cluster_name == "Trm effector CD8 ")
TRM.cd8$Group<- factor(TRM.cd8$Group, levels = c("osteoarthritis", "ICI_arthritis"))
TRM.cd8$Arthritis<- factor(TRM.cd8$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))


#visualization ! 
p7<- ggplot(TRM.cd8, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% Trm effector CD8 of total cells", title = "Trm CD8")
p7
```

#one way analysis of variance! 
``` {r}
#parametric anova
cd8trm<- aov(frequency ~ Group, data=TRM.cd8)
summary(cd8trm)
TukeyHSD(cd8trm)  #non-significant

#non-parametric anova 
kruskal.test(frequency ~ Group, data=ActivatedCD4)
``` 

``` {r}
cyclingT<- proportion %>% filter(cluster_name == "Cycing T")
cyclingT$Group<- factor(cyclingT$Group, levels = c("osteoarthritis", "ICI_arthritis"))
cyclingT$Arthritis<- factor(cyclingT$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))


#visualization ! 
p8<- ggplot(cyclingT, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% cycling T of total cells", title = "Cycling T")
p8
```

#idependent T test
``` {r}
t.test(frequency ~ Group, data=cyclingT)
```
``` {r}
wilcox.test(frequency ~ Group, data=cyclingT)
```


``` {r}
NKT<- proportion %>% filter(cluster_name == "NKT")
NKT$Group<- factor(NKT$Group, levels = c("osteoarthritis", "ICI_arthritis"))
NKT$Arthritis<- factor(NKT$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))


#visualization ! 
p9<- ggplot(NKT, aes(x= Group, y= frequency, fill= Group)) +
   theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_text(angle = 90, face = "bold", vjust = 0.5, hjust = 0.5), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% NK T of total cells", title = "NK T")
p9
```
#idependent T test
``` {r}
t.test(frequency ~ Group, data=NKT)

``` 


``` {r}
ILC<- proportion %>% filter(cluster_name == "Mixed T")
ILC$Group<- factor(ILC$Group, levels = c("osteoarthritis", "ICI_arthritis"))
ILC$Arthritis<- factor(ILC$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))


#visualization ! 
p10<- ggplot(ILC, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_text(size = 8, face = "bold"), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_text(angle = 90, face = "bold", vjust = 0.5, hjust = 0.5)) + 
  labs(x= "Groups", y= "% of Total Cells", title = "Mixed T")
p10

p10<- ggplot(ILC, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_text(size = 8, face = "bold"), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_text(angle = 90, face = "bold", vjust = 0.5, hjust = 0.5),
  legend.position = "none") + 
  labs(x= "Groups", y= "% of Total Cells", title = "Mixed T")
p10
```
#idependent T test
``` {r}
t.test(frequency ~ Group, data=ILC)
``` 

``` {r}
Nk<- proportion %>% filter(cluster_name == "NK")
Nk$Group<- factor(Nk$Group, levels = c("osteoarthritis", "ICI_arthritis"))
Nk$Arthritis<- factor(Nk$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))

#visualization ! 
p11<- ggplot(Nk, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_text(angle = 90, face = "bold", vjust = 0.5, hjust = 0.5), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% of Totol Cells", title = "NK")
p11
```
#idependent T test
``` {r}
t.test(frequency ~ Group, data=Nk)
``` 
``` {r}
wilcox.test(frequency ~ Group, data=Nk)
```


``` {r}
Bcell<- proportion %>% filter(cluster_name == "B cells")
Bcell$Group<- factor(Bcell$Group, levels = c("osteoarthritis", "ICI_arthritis"))

#visualization ! 
p12<- ggplot(Bcell, aes(x= Group, y= frequency, fill= Group)) +
  geom_bar(position = 'dodge2', stat = 'summary', fun.y='mean') +
  geom_errorbar(stat = 'summary', width=0, size=0.5, position = position_dodge(0.9)) +
  theme_classic() + 
  geom_point(size= 6, aes(x=Group), shape=21, position = position_dodge(width = 0.9)) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=14
  ), axis.title.y = element_text(size=13, face = "bold"), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=10, face = "bold"), 
  axis.text.x = element_text(angle = 50, vjust = 1, hjust = 1), 
  legend.position = "none", 
  legend.title = element_text(size = 14, face = "bold"), 
  legend.text = element_text(size = 12, face = "bold")) +
  labs(x= "Groups", y= "% B cell of total cells", title = "B cell")
p12
```

#Combine 
``` {r}
grant<- p3 + p12
ggsave("bcell.pdf", plot = grant, height=4, width = 5, units = "in", dpi = 300)
``` 



``` {r}
Bcell<- proportion %>% filter(cluster_name == "B cells")
Bcell$Arthritis<- factor(Bcell$Arthritis, levels = c("osteoarthritis","first arthritis", "second arthritis"))
Bcell$Group<- factor(Bcell$Group, levels = c("osteoarthritis", "ICI_arthritis"))


#visualization ! 
p12<- ggplot(Bcell, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_text(angle = 90, face = "bold", vjust = 0.5, hjust = 0.5), 
  legend.position = "none") +
  labs(x= "Groups", y= "% B cell of total cells", title = "B cell")
p12
```
# plot B cell frequency of the second vs. first arthritis !! 
``` {r}
b_second_vs_first<- Bcell %>% 
  filter(Arthritis %in% c("first arthritis", 
                          "second arthritis"))

# ggplot 
bse_vs_fi<- ggplot(b_second_vs_first, aes(x= Arthritis, y=frequency, fill= Arthritis)) + 
  theme_classic() +
  geom_boxplot() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=14
  ), axis.title.y = element_text(size= 15, face = "bold"), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=15, face = "bold"), 
  axis.text.x = element_blank()) + 
  theme(legend.position = "none") + 
  ylab("% of total cells") + 
  ggtitle("B cell")
bse_vs_fi
``` 
# save violen plot !!! 
``` {r}
ggsave("bcell_second_vs_first.pdf", plot = bse_vs_fi, height=3, width = 3, units = "in", dpi = 300)

``` 

#idependent T test
``` {r}
t.test(frequency ~ Group, data=Bcell)
``` 

``` {r}
mono.macro<- proportion %>% filter(cluster_name == "Classical Monocytes")
mono.macro$Arthritis<- factor(mono.macro$Arthritis, levels = c("osteoarthritis","first arthritis", "second arthritis"))
mono.macro$Group<- factor(mono.macro$Group, levels = c("osteoarthritis", "ICI_arthritis"))


#visualization ! 
p13<- ggplot(mono.macro, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% classical mono of total cells", title = "Classical mono")
p13
```

#idependent T test
``` {r}
t.test(frequency ~ Group, data=mono.macro)
``` 

``` {r}
neutro<- proportion %>% filter(cluster_name == "Non-classical monocytes")
neutro$Arthritis<- factor(neutro$Arthritis, levels = c("osteoarthritis","first arthritis", "second arthritis"))
neutro$Group<- factor(neutro$Group, levels = c("osteoarthritis", "ICI_arthritis"))


#visualization ! 
p14<- ggplot(neutro, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.4
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% Non-class mono of all cells", title = "Non-classical mono")
p14
```
``` {r}
macrophage<- proportion %>% filter(cluster_name == "SPP1+ Macrophages")
macrophage$Arthritis<- factor(macrophage$Arthritis, levels = c("osteoarthritis","first arthritis", "second arthritis"))
macrophage$Group<- factor(macrophage$Group, levels = c("osteoarthritis", "ICI_arthritis"))


#visualization ! 
p15<- ggplot(macrophage, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% SPP1+ Macrophages of CD45", title = "SPP1+ Macrophages")
p15
```
#idependent T test
``` {r}
t.test(frequency ~ Group, data=macrophage)
wilcox.test(frequency ~ Group, data=macrophage)
```
``` {r}
wilcox.test(frequency ~ Group, data=macrophage)
```


``` {r}
monocyte<- proportion %>% filter(cluster_name == "CD1c+ Non-inflammatory DCs")
monocyte$Arthritis<- factor(monocyte$Arthritis, levels = c("osteoarthritis","first arthritis", "second arthritis"))
monocyte$Group<- factor(monocyte$Group, levels = c("osteoarthritis", "ICI_arthritis"))

#visualization ! 
p16<- ggplot(monocyte, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.4
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% of Total Cells", title = "Non-inflamed DC")
p16
```
#idependent T test
``` {r}
t.test(frequency ~ Group, data=monocyte)
wilcox.test(frequency ~ Group, data=monocyte)
```

``` {r}
CD1C<- proportion %>% filter(cluster_name == "CLEC9A+ DCs")
CD1C$Arthritis<- factor(CD1C$Arthritis, levels = c("osteoarthritis","first arthritis", "second arthritis"))
CD1C$Group<- factor(CD1C$Group, levels = c("osteoarthritis", "ICI_arthritis"))


#visualization ! 
p17<- ggplot(CD1C, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% CLEC9A+ DCs of all cells", title = "CLEC9A+ DCs")
p17
```
#idependent T test
``` {r}
t.test(frequency ~ Group, data=CD1C)
wilcox.test(frequency ~ Group, data=CD1C)
```

``` {r}
CDC2<- proportion %>% filter(cluster_name == "pDC")
CDC2$Arthritis<- factor(CDC2$Arthritis, levels = c("osteoarthritis","first arthritis", "second arthritis"))
CDC2$Group<- factor(CDC2$Group, levels = c("osteoarthritis", "ICI_arthritis"))


#visualization ! 
p18<- ggplot(CDC2, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_text(size = 8, face = "bold"), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% of Total Cells", title = "pDC")
p18
```
``` {r}
wilcox.test(frequency ~ Group, data=CDC2)
t.test(frequency ~ Group, data=CDC2)
``` 


```{r}
#parametric anova
pdc<- aov(frequency ~ type, data=CDC2)
summary(pdc)
TukeyHSD(pdc)  #non-significant

#non-parametric anova 
kruskal.test(frequency ~ type, data=ActivatedCD4)

```




``` {r}
mdC<- proportion %>% filter(cluster_name == "mDC")
mdC$Arthritis<- factor(mdC$Arthritis, levels = c("osteoarthritis","first arthritis", "second arthritis"))
mdC$Group<- factor(mdC$Group, levels = c("osteoarthritis", "ICI_arthritis"))

#visualization ! 
p19<- ggplot(mdC, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% mDC of CD45", title = "mDC")
p19
```
``` {r}
wilcox.test(frequency ~ Group, data=mdC)
t.test(frequency ~ Group, data=mdC)
``` 

``` {r}
Neutrophil<- proportion %>% filter(cluster_name == "Neutrophil")
Neutrophil$Arthritis<- factor(Neutrophil$Arthritis, levels = c("osteoarthritis","first arthritis", "second arthritis"))
Neutrophil$Group<- factor(Neutrophil$Group, levels = c("osteoarthritis", "ICI_arthritis"))

#visualization ! 
p20<- ggplot(Neutrophil, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_text(size = 8, face = "bold"), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% Total cells", title = "Neutrophil")
p20
```
``` {r}
wilcox.test(frequency ~ Group, data=Neutrophil)
t.test(frequency ~ Group, data=Neutrophil)
``` 

``` {r}
synovial<- proportion %>% filter(cluster_name == "Synovial cells")
synovial$Arthritis<- factor(synovial$Arthritis, levels = c("osteoarthritis","first arthritis", "second arthritis"))
synovial$Group<- factor(synovial$Group, levels = c("osteoarthritis", "ICI_arthritis"))

#visualization ! 
p21<- ggplot(synovial, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_text(angle = 90, face = "bold", vjust = 0.5, hjust = 0.5), 
  legend.position = "none", 
  legend.text = element_text(face = "bold"), 
  legend.title = element_text(face = "bold")) + 
  labs(x= "Groups", y= "% Synovial cells of all cells", title = "Synovial cells")


p21
```
``` {r}
wilcox.test(frequency ~ Group, data=synovial)
t.test(frequency ~ Group, data=synovial)
``` 

``` {r}
library(patchwork)

proportion1<- (p20|p13|p14|p15|p19)/(p18|p16|p17|p1|p2)
ggsave("immune_fraction.pdf", plot = proportion1, height=3, width = 5, units = "in", dpi = 300)

# 2024 IOTOX !! 
ggsave("immune_fraction.tiff", plot = proportion1, height=3, width = 5, units = "in", dpi = 300)

#Last 10 
proportion2<- (p3|p5|p6|p7|p8)/(p10|p4|p11|p9|p12|p21)
ggsave("immune_fraction1.pdf", plot = proportion2, height=3.3, width = 5, units = "in", dpi = 300)

#2024 IOTOX
ggsave("immune_fraction1.tiff", plot = proportion2, height=3.3, width = 5, units = "in", dpi = 300)

ggsave("legend.pdf", plot = p21, height=2, width = 9, units = "in", dpi = 300)
``` 


 

# Annotate T cell subclusters! 

``` {r}
#load Seurat obj! 
library(tidyverse)
library(Seurat)
seurat.obj.T<- readRDS("umap.obj.pc15.rds")

#Dimplot ! 
umapT<- DimPlot(seurat.obj.T, reduction = "umap", label = TRUE, split.by = "type") +
  theme_classic() +
  NoLegend() + 
  NoAxes()
umapT
``` 
# Annotation ! 
*0= Th17 like cell*
*1= CD8 effector*
*2= ILC*
*3= Treg*
*4= CD4 Tcm* 
*5= CD8 Tem* 
*6= PD-1hi CXCL13hi CD4+*
*7= Effector CD8* 
*8= Th1-like*
*9= Cytotoxic T*
*10= Recently activated CD4*
*11= Treg*
*12= NKG7+ CD8+ T cells*
*13= NKT*
*14= CD8 Trm*
*15= MAIT*
*16= Cycling T*
*17= KLRB1hi CD8 Tem*
*18= mixed CD4*


# save plot1 
``` {r}
ggsave("umap.T.pdf", plot = umapT, height=4.5, width = 5.7, units = "in", dpi = 300)
```


``` {r}
#load Seurat obj! 
library(tidyverse)
library(Seurat)
library(monocle3)
library(SeuratWrappers)
seurat.obj.T<- readRDS("umap.obj.pc15.rds")

#Dimplot ! 
DimPlot(seurat.obj.T, 
        split.by = "type", 
        reduction = "umap", label = TRUE) +
  theme_minimal() +
  NoLegend()
```


# Annotation ! 
*0= Th17 like cell*
*1= CD8 effector*
*2= ILC*
*3= Treg*
*4= CD4 Tcm* 
*5= CD8 Tem* 
*6= PD-1hi CXCL13hi CD4+*
*7= Effector CD8* 
*8= Th1-like*
*9= Cytotoxic T*
*10= Recently activated CD4*
*11= Treg*
*12= NKG7+ CD8+ T cells*
*13= NKT*
*14= CD8 Trm*
*15= MAIT*
*16= Cycling T*
*17= KLRB1hi CD8 Tem*
*18= mixed CD4*

# Check cell numbers! 
``` {r}
table(seurat.obj.T@active.ident, seurat.obj.T@meta.data$orig.ident)
``` 


``` {r}
T.annotated.subsets<- RenameIdents(seurat.obj.T, `0` ="Th17 like",`10`= "Activated CD4", `4`= "CD4 Tcm", 
                                   `6`= "PD-1hi CXCL13hi CD4", `8`= "Th1-like", `15`= "MAIT", `3` ="Treg", `11`="Treg", 
                                   `18`= "Naive CD4", `2` ="gd T", `7`= "mixed T", `5`= "CD8 Tem", `1` ="CD8 effector",
                                   `14`= "CD8 Trm",  `9`= "CD8 effector",  `17`= "KLRB1hi CD8 Tem",  `12`= "NKG7+ CD8", `16`= "Cycling T", 
                                   `13`= "NKT")


annotated.plot<- DimPlot(T.annotated.subsets, reduction = "umap", label = FALSE, label.size = 4) +
  theme_classic() + 
  theme(legend.text = element_text(size= 8)) + 
  NoAxes() + NoLegend() 
  #scale_color_manual(values = c("CD8 effector"= "#FF3349", "Th17 like"="#77FF33", "Treg"= "#3633FF", 
                                #"Activated CD4"="#FF33F3", "CD4 Tcm"="#33C4FF", "PD-1hi CXCL13hi CD4"= "#FFAF33", 
                                #"Th1-like"= "#33FCFF", "MAIT"= "#338DFF", "Naive CD4"= "#9933FF", "gd T"= "#FF33D1", 
                                #"CD8 Tem"= "#802D8A", "CD8 Trm"= "#33FFAC", "KLRB1hi CD8 Tem"= "#A833FF", "NKG7+ CD8"= "#993527", 
                                #"Cycling T"= "#58B36E", "NKT"= "#A817B9"))

annotated.plot

saveRDS(T.annotated.subsets, file = "umap.T.obj.annotated.rds")  #latest save on 5/3/2024
``` 

# save plot1 
``` {r}
ggsave("umap.T.pdf", plot = annotated.plot, height=4.5, width = 5.7, units = "in", dpi = 300)
```


# save this this annotated obj! 
``` {r}
ggsave("umap.T.subcluster.tiff", plot = annotated.plot, height=5, width = 8, units = "in", dpi = 300)
``` 

# Cytokine expression 

``` {r, fig.width= 7, fig.height= 6.5}
genes <- c( "IFNG","IL4", "IL10", "IL17A", "IL21", "TNF", "TGFB1","IL1A", "IL1B", "IL2","GZMB", "GZMM", "GZMH", 
           "GZMK", "GNLY", "CSF2", "PRF1","FCGR3A","TRGV2","TRGV9","MKI67","CX3CR1", "EOMES","TIGIT","HAVCR2","PDCD1","CTLA4","LAG3","TOX","KLRG1","IL7R", "IL2RA", "KLRB1", "CCL5","CCL4","CCL3","CXCL13", "CXCL10","CXCL9","CXCR6","CXCR5","CXCR3","CCR7","CCR6","CCR5","CCR4","CCR3","CCR1","RORC","GATA3","TBX21","CD38","CD69","CD274","NCAM1","FOXP3","CD8A","CD4","CD3E","CD3D")


#Dot plot!
install.packages("viridis")
library(viridis)
Dot.plot<- DotPlot(T.annotated.subsets, features = genes, dot.scale = 4.5) + 
 geom_point(aes(size=pct.exp), shape = 21, colour="black", stroke=0.5) +
  scale_colour_viridis(option="inferno") +
  guides(size=guide_legend(override.aes=list(shape=21, colour="black", fill="white"))) +
  RotatedAxis() + 
  coord_flip() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=15), 
        axis.title.y= element_text(size = 14, face = "bold"), 
        axis.text = element_text(size=10, face = "bold"), 
        axis.text.x = element_text(angle=90,hjust=0.95,vjust=0.2, face = "bold", size= 10), 
        axis.title.x = element_blank(),
        legend.title = element_text(size= 12, face = "bold"), 
        legend.text = element_text(size= 10, face = "bold"), 
        legend.position = "right") 
Dot.plot
```
#save pdf 
```{r}
ggsave("dot.cytokine.violen.pdf", plot = Dot.plot , height=9, width = 5.5, units = "in", dpi = 300)
``` 

``` {r}

genes <- c("CD4", "CD8A", "FOXP3", "CD69", "CD38", "TBX21", "GATA3", "RORC","CCR1", "CCR3", "CCR4", "CCR5", "CCR6","CCR7", "CXCR3","CXCR5","CXCR6",  
           "CXCL9","CXCL10","CXCL11", "CXCL13","CCL3","CCL4","CCL5","KLRB1", "IL2RA", "IL7R", "KLRG1", "TOX", "LAG3", "CTLA4", "PDCD1", "HAVCR2", "TIGIT", "EOMES", "CX3CR1",  "MKI67", "TRGV9", "TRGV2", "FCGR3A", "NCAM1", "CD3E","CD3D")

genes <- c("CD3E", "CD3D", "CD4", "CD8A", "FOXP3", "IFNG", "IL4", "IL10", "IL17A", "IL21", "TNF", "TGFB1","IL1A", "IL1B", "IL1B", "IL2","GZMB", "GZMM", "GZMH", 
           "GZMK", "GNLY", "CSF2", "PRF1")


#plot stacked!
stacked_violen_1<- VlnPlot(T.annotated.subsets, features = genes, stack = TRUE, flip = TRUE, adjust = 0.5) &
  theme(legend.position = "none",
        axis.text.x =  element_blank(),
        axis.text = element_text(size = 10)) + 
  RotatedAxis() 

stacked_all_2<- stacked_violen_1 + 
  theme(axis.text.x = element_text(angle=90,hjust=0.95,vjust=0.2, face = "bold"), 
               axis.text.y = element_text(size = 8), 
               axis.title.x = element_blank())

stacked_all_2

```

#save pdf 
```{r}
ggsave("stacked.violen.pdf", plot = stacked_all_2 , height=6, width = 5, units = "in", dpi = 300)
``` 

# Cytokine expression 
``` {r, fig.width= 7, fig.height= 8}
genes <- c("CD3E", "CD3D", "CD4", "CD8A", "FOXP3", "IFNG", "IL4", "IL10", "IL17A", "IL21", "TNF", "GZMB", "GZMM", "GZMH", 
           "GZMK", "GNLY", "CSF2", "PRF1")

#plot
Dot.plot<- DotPlot(T.annotated.subsets, features = genes, cols = c("blue", "red"), dot.scale = 4.5) + 
  theme_bw() +
  RotatedAxis() + 
  coord_flip() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=15), 
        axis.title.y= element_text(size = 14, face = "bold"), 
        axis.text = element_text(size=13, face = "bold"), 
        axis.text.x = element_text(angle=90,hjust=0.95,vjust=0.2, face = "bold", size= 12), 
        axis.title.x = element_blank(),
        legend.title = element_text(size= 12, face = "bold"), 
        legend.text = element_text(size= 10, face = "bold")) 
Dot.plot

```

#save pdf 
```{r}
ggsave("dot.cytokine.violen.pdf", plot = Dot.plot , height=7, width = 6.8, units = "in", dpi = 300)
``` 

# clean a bit more! 
``` {r}
#Dimplot ! 
DimPlot(seurat.obj.T,
        reduction = "umap", label = TRUE) +
  theme_minimal() +
  NoLegend()
``` 
``` {r}
plot<- DimPlot(seurat.obj.T, reduction= "umap", label = TRUE) +
  theme_minimal() +
  NoLegend()
seurat.object<- CellSelector(plot = plot, seurat.obj.T)

DimPlot(seurat.object, reduction= "umap", label = TRUE) +
  theme_minimal() +
  NoLegend()
```

#subset all cluster except the selectedCells! 
``` {r}
seurat.obj.T<- subset(seurat.object, idents = c("0", "1", "2", "3", "4", "5", "6", 
                                                    "7", "8", "9", 
                                                    "10", "11", "12", "13", "14", "15", "16", 
                                                 "17", "18"))
DimPlot(seurat.obj.T, reduction= "umap", label = TRUE) +
  theme_minimal() +
  NoLegend()
``` 
``` {r}
plot<- DimPlot(seurat.obj.T, reduction= "umap", label = TRUE) +
  theme_minimal() +
  NoLegend()

seurat.object<- CellSelector(plot = plot, seurat.obj.T)

DimPlot(seurat.object, reduction= "umap", label = TRUE) +
  theme_minimal() +
  NoLegend()
```
#subset all cluster except the selectedCells! 
``` {r}
seurat.obj.T<- subset(seurat.object, idents = c("0", "1", "2", "3", "4", "5", "6", 
                                                    "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18"))
DimPlot(seurat.obj.T, reduction= "umap", label = TRUE) +
  theme_minimal() +
  NoLegend()
```
# save this final object
``` {r}
saveRDS(seurat.obj.T, file = "umap.obj.pc15.rds")
``` 

# read rds file back in! 
``` {r}
seurat.obj.T<- readRDS("umap.obj.pc15.rds")
DimPlot(seurat.obj.T, label = TRUE) + 
  NoAxes() + 
  NoLegend() 
``` 


``` {r}
T.annotated.subsets<- RenameIdents(seurat.obj.T, `0` ="Th17 like",`10`= "Activated CD4", `4`= "CD4 Tcm", 
                                   `6`= "PD-1hi CXCL13hi CD4", `8`= "Th1-like", `15`= "MAIT", `3` ="Treg", `11`="Treg", 
                                   `18`= "MAIT", `2` ="gd T", `7`= "Mixed T", `5`= "CD8 Tem", `1` ="CD8 effector",
                                   `14`= "CD8 Trm",  `9`= "CD8 effector",  `17`= "KLRB1hi CD8 Tem",  `12`= "NKG7+ CD8", `16`= "Cycling T", 
                                   `13`= "NKT")


annotated.plot<- DimPlot(T.annotated.subsets, reduction = "umap", label = FALSE, label.size = 4) +
  theme_classic() + 
  theme(legend.text = element_text(size= 8)) + 
  NoAxes() + NoLegend() 
  #scale_color_manual(values = c("CD8 effector"= "#FF3349", "Th17 like"="#77FF33", "Treg"= "#3633FF", 
                                #"Activated CD4"="#FF33F3", "CD4 Tcm"="#33C4FF", "PD-1hi CXCL13hi CD4"= "#FFAF33", 
                                #"Th1-like"= "#33FCFF", "MAIT"= "#338DFF", "Naive CD4"= "#9933FF", "gd T"= "#FF33D1", 
                                #"CD8 Tem"= "#802D8A", "CD8 Trm"= "#33FFAC", "KLRB1hi CD8 Tem"= "#A833FF", "NKG7+ CD8"= "#993527", 
                                #"Cycling T"= "#58B36E", "NKT"= "#A817B9"))

annotated.plot

saveRDS(T.annotated.subsets, file = "umap.T.obj.annotated.rds")  #latest save on 5/3/2024
```

# Add metadata
``` {r}
view(T.annotated.subsets)
T.annotated.subsets@meta.data$group <- ifelse(T.annotated.subsets@meta.data$type %in% c("First arthritis", "Second arthritis"), "ICI-arthritis", "Osteoarthritis")
``` 


``` {r}
seurat.obj.T@meta.data$type<- factor(seurat.obj.T@meta.data$type, levels = c("Osteoarthritis", "First arthritis","Second arthritis"))

#Dimplot ! 
split_umap<- DimPlot(seurat.obj.T, reduction = "umap", label = FALSE, split.by = "group") +
  NoLegend() + 
  theme(axis.title = element_text(size= 15, face = "bold"),
        axis.text = element_text(size = 14, face = "bold")) + 
  NoAxes()
DimPlot(seurat)
``` 

# save plot1 
``` {r}
ggsave("umap.T.pdf", plot = annotated.plot, height=4.5, width = 5.7, units = "in", dpi = 300)
```


``` {r}
VlnPlot(SelectedCells, features = c("nFeature_RNA", "nCount_RNA"), ncol = 2)
```


``` {r, fig.width= 12, fig.height= 9}
FeaturePlot(SelectedCells, features = c("EPCAM", "PTPRC", "COL1A1", "CD79A", "CD3D", "S100A8", "S100A9", "CD8A"), pt.size = 0.2,
    ncol = 3)
```

# Examine differentially expressed genes between first/second arthritis
``` {r}
# Identify differentially expressed genes across conditions
view (T.annotated.subsets@meta.data)
Idents(T.annotated.subsets)
T.annotated.subsets$celltypes<-Idents(T.annotated.subsets)
view (T.annotated.subsets@meta.data)
T.annotated.subsets$arthritis_celltype <- paste(T.annotated.subsets$type, sep = "_", Idents(T.annotated.subsets))
view (T.annotated.subsets@meta.data)
Idents(T.annotated.subsets) <- "arthritis_celltype"
Idents(T.annotated.subsets)

#Calculate differentially expressed genes between first and second arthritis!
prepsct<- PrepSCTFindMarkers(T.annotated.subsets)
effector_cd8<- FindMarkers(T.annotated.subsets,assay = "SCT", ident.1 = "Second arthritis_CD8 effector", 
                               ident.2 = "First arthritis_CD8 effector", min.pct = 0.25, logfc.threshold = 0.25, 
                           verbose = FALSE, 
                           recorrect_umi = FALSE)
head(effector_cd8, n = 20)

# order it from highest log2FC to lowest
effector_cd8_order<- effector_cd8 %>% arrange(desc(avg_log2FC))
head(effector_cd8_order)


effector_cd8_order<- effector_cd8_order[effector_cd8_order$p_val_adj<0.05, ]

# Save deg of cluster 1
write.csv(effector_cd8_order, file = "deg.effector.CD8.csv")
```


``` {r}
#volcano plots !! 
volcano.effector.cd8 <- effector_cd8[order(effector_cd8$p_val_adj), ]

# Create data.frame with new categories
results <- as.data.frame(
  mutate(
    volcano.effector.cd8,
    cut_off = case_when(
      avg_log2FC > 0.5 & p_val_adj < 0.05 ~ "Upregulated",
      avg_log2FC < -0.5 & p_val_adj < 0.05 ~ "Downregulated",
      TRUE ~ "Not significant"
    )
  ),
  row.names = rownames(volcano.effector.cd8)
)

# Update factor levels for better control over plot appearance
library(ggrepel)
results$cut_off <- factor(results$cut_off, levels = c("Downregulated", "Not significant", "Upregulated"))

# Plotting
volcano_effector.cd8 <- ggplot(results, aes(avg_log2FC, -log10(p_val_adj))) + 
  theme_classic() +
  geom_point(aes(col = cut_off)) +
  scale_color_manual(values = c("#00AFBB", "grey", "#bb0c00")) +
  geom_vline(xintercept = c(-0.5, 0.5), col = "gray", linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), col = "gray", linetype = 'dashed') +
  theme(plot.title = element_text(hjust = 0.5, size = 10, face = "bold"), 
        axis.title = element_text(face = "bold", size = 10), 
        axis.text = element_text(face = "bold", size = 8), 
        legend.title = element_blank(), 
        legend.text = element_text(size = 6), 
        legend.position = c(0.18, 0.88)) + 
  geom_text_repel(data = results[rownames(results) %in% genes_to_label, ], 
                  aes(label = rownames(results[rownames(results) %in% genes_to_label, ])), 
                  size = 2) +
  ggtitle("Second vs. first arthritis (Th17 like)")
``` 

# new type of volcano plot! 
``` {r}
# Loading relevant libraries 
library(tidyverse) # includes ggplot2, for data visualisation. dplyr, for data manipulation.
library(RColorBrewer) # for a colourful plot
library(ggrepel) # for nice annotations

# Set input path
path <- "C:/Users/Documents/Biostatsquid/Volcano_plots/Tutorial/"  # I can set later
setwd(path)

# Import DGE results
deg.first.vs.second.effectorCD8<- read.csv("deg_between conditions/deg_second_vs_first_effector.CD8.csv", row.names = 1)
head(deg.first.vs.second.effectorCD8)
``` 

``` {r}
deg.first.vs.second.effectorCD8$diffexpressed<- "NO"
deg.first.vs.second.effectorCD8$diffexpressed[deg.first.vs.second.effectorCD8$avg_log2FC > 0.5 & deg.first.vs.second.effectorCD8$p_val_adj < 0.05] <- "UP"
deg.first.vs.second.effectorCD8$diffexpressed[deg.first.vs.second.effectorCD8$avg_log2FC < -0.5 & deg.first.vs.second.effectorCD8$p_val_adj < 0.05] <- "DOWN"
head(deg.ici.vs.osteo[order(deg.first.vs.second.effectorCD8$p_val_adj) & deg.first.vs.second.effectorCD8$diffexpressed == 'DOWN', ])
``` 

# Create a new column "delabel" to de, that will contain the name of the top 30 differentially expressed genes (NA in case they are not)
``` {r}
deg.first.vs.second.effectorCD8$delabel <- ifelse(deg.first.vs.second.effectorCD8$gene_symbol %in% head(deg.first.vs.second.effectorCD8[order(deg.first.vs.second.effectorCD8$p_val_adj), "gene_symbol"], 20), deg.first.vs.second.effectorCD8$gene_symbol, NA)
``` 



``` {r}
volcano_effectorcd8_secondvsfirst<- ggplot(data = deg.first.vs.second.effectorCD8,
                                      aes(x = avg_log2FC, y = -log10(p_val_adj), 
                                                                   col = diffexpressed, 
                                          label= delabel)) +
  geom_vline(xintercept = c(-0.5, 0.5), col = "gray", linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), col = "gray", linetype = 'dashed') +
  geom_point(size = 2) +
  theme(plot.title = element_text(hjust = 0.5, size = 8.5, face = "bold"), 
        axis.title = element_text(face = "bold", size = 12), 
        axis.text = element_text(face = "bold", size=11), 
        legend.title = element_blank(), 
        legend.text = element_blank(), 
        legend.position = ) +
  scale_color_manual(values = c("#00AFBB", "grey", "#bb0c00"), 
                     labels = c("Downregulated", "Not significant", "Upregulated"))  + 
 ggtitle("Second vs. first arthritis (effector CD8 cluster)") + 
   geom_text_repel(max.overlaps = Inf, size=2.3) # To show all labels 
volcano_effectorcd8_secondvsfirst
``` 

# save high quality plot! 
``` {r}
ggsave("volcano_secondvsfirst_effectorcd8.pdf", plot = volcano_effectorcd8_secondvsfirst, height=4, width = 4.9, units = "in", dpi = 300)
```
# save high quality plot! 
``` {r}
ggsave("volcano_secondvsfirst_effectorcd8new.pdf", plot = volcano_effectorcd8_secondvsfirst, height=4, width = 4, units = "in", dpi = 300)
```

#deg plot for volcano! 
``` {r}
## Volcano plot @ 1 day 

volcano<- effector_cd8[order(effector_cd8$p_val_adj),]

# create data.frame
results<- as.data.frame(mutate(as.data.frame(volcano), 
                                    cut_off=ifelse(volcano$p_val_adj<0.05, "significant", "insignificant")), 
                             row.names=rownames(volcano))
## ggplot plus ggrepel
options(ggrepel.max.overlaps = Inf)
library(ggrepel)
volcano$cut_off<- factor(results$cut_off, levels = c('significant', 'insignificant'))

# plot
volcano_effectorcd8<- ggplot(results, aes(avg_log2FC, -log10(p_val_adj))) + 
  theme_classic() +
  geom_point(aes(col = cut_off)) +
  scale_color_manual(values = c("black", "red")) +
  theme(plot.title = element_text(hjust = 0.5, size = 8, face = "bold"), 
        axis.title = element_text(face = "bold", size = 12), 
        axis.text = element_text(face = "bold", size=8), 
        legend.title = element_text(size = 8, face = "bold"), 
        legend.text = element_text(size = 6), 
        legend.position = c(0.88, 0.2)) + 
  geom_text_repel(data=results[1:20,], 
                  aes(label=rownames(results[1:20,])), size= 2) +
  ggtitle("Second vs. first arthritis (effector CD8 cluster)") 
volcano_effectorcd8
``` 

# save high quality plot! 
``` {r}
ggsave("volcano.pdf", plot = volcano_effectorcd8, height=2.8, width = 3.25, units = "in", dpi = 300)
``` 


# Perform gene set enrichment analysis. === Second vs. first arthritis! 
``` {r}
## perform gene set enrichment 

BiocManager::install("fgsea")
library(fgsea)
effector_cd8$gene<- rownames(effector_cd8)
effector_cd8<- effector_cd8 %>% arrange(desc(avg_log2FC))

#write csv file for deg
write.csv(DEG.activatedCD4, file = "Y:/Recurrent__Arthritis__Single cell/DEG.activatedCD4.csv")

fold_changes<- effector_cd8$avg_log2FC
names(fold_changes)<- effector_cd8$gene
``` 

#Load geneset ! 
``` {r}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("fgsea")
Reactome <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.reactome.v2023.1.Hs.symbols.gmt")
hallmark <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/h.all.v2023.1.Hs.symbols.gmt")
KEGG <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.kegg.v2023.1.Hs.symbols.gmt")
GO <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c5.go.bp.v2023.1.Hs.symbols.gmt")

#combine the four gsea
Four_gene_sets <- c(hallmark, KEGG, GO, Reactome)

#getset for all pathway 
Allgenesets <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/msigdb.v2023.1.Hs.symbols.gmt")

# GSEA analysis
gsea_Myo_effectorcd8 <- fgsea(pathways = hallmark,
                  stats = fold_changes,
                  eps = 0.0,
                  minSize=15,
                  maxSize=500)

head(gsea_Myo[order(pval), ]) 
```

# plot enrichment 
``` {r}
library(tidyverse)

## effector CD8_second vs. first arthritis !!! 
gseaicisecondvsfirst<- read.csv("enrichment.csv", stringsAsFactors = TRUE, header = TRUE)
gseaicisecondvsfirst<- gseaicisecondvsfirst %>% filter(group== "top")

# ggplot
gseaicisecondvsfirst<- ggplot(gseaicisecondvsfirst, aes(reorder(Pathway, NES), NES, fill= padj)) + 
  theme_classic() +
  geom_bar(stat = "identity")  + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 8),
        axis.title.x = element_text(face = "bold", size = 12),
        axis.title.y  = element_blank(), 
        axis.text = element_text(face = "bold", size = 10), 
        plot.title = element_text(hjust = 0.5, face = "bold", size = 10)) + 
  labs(x="Hallmark pathways", y="normalized enrichment score",
       title="Second vs. first arthritis (Effector CD8)") + 
  coord_flip() + 
  scale_fill_gradient(low = "red", high = "blue")
gseaicisecondvsfirst

 scale_fill_gradient(low = "#2c7fb8", high = "#edf8b1")
gseaicisecondvsfirst

# Dotplot !! 
gseaicisecondvsfirst <- ggplot(gseaicisecondvsfirst, aes(reorder(Pathway, NES), NES, fill = padj)) + 
  theme_bw() +
  geom_point(shape = 21, size = 5) +  # Dot plot with fixed size
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 12),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(), 
        axis.text = element_text(face = "bold", size = 12), 
        plot.title = element_blank(), 
        panel.border = element_rect(colour = "black", fill = NA, size = 1.5)) + 
  labs(x="Hallmark pathways", y="Normalized enrichment score",
       title="Second vs. first arthritis (Effector CD8)") + 
  coord_flip() + 
  scale_fill_gradient(low = "red", high = "blue")
gseaicisecondvsfirst
``` 


``` {r}
ggsave("gseasecondvsfirst.pdf", plot = gseaicisecondvsfirst, height=3, width =6, units = "in", dpi = 300)

#2024 IOTOX !! 
ggsave("gseasecondvsfirst.tiff", plot = gseaicisecondvsfirst, height=3.5, width =6, units = "in", dpi = 300)
```

# Th17 like cell!!! 

``` {r}
th17.like<- FindMarkers(T.annotated.subsets,assay = "SCT", ident.1 = "Second arthritis_Th17 like", 
                               ident.2 = "First arthritis_Th17 like", min.pct = 0.25, logfc.threshold = 0.25, 
                           verbose = FALSE, 
                           recorrect_umi = FALSE)
head(th17.like, n = 20)

# order it from highest log2FC to lowest
th17.like_order<- th17.like %>% arrange(desc(avg_log2FC))
head(th17.like_order)


th17.like_order<- th17.like_order[th17.like_order$p_val_adj<0.05, ]

# Save deg of cluster 1
write.csv(th17.like_order, file = "th17_deg_second_vs_first.csv")
```

# Th17 volcano plot. 
``` {r}
volcano.th17<- th17.like[order(th17.like$p_val_adj),]

# create data.frame
results<- as.data.frame(mutate(as.data.frame(volcano.th17), 
                                    cut_off=ifelse(volcano.th17$p_val_adj<0.05, "significant", "insignificant")), 
                             row.names=rownames(volcano.th17))
## ggplot plus ggrepel
options(ggrepel.max.overlaps = Inf)
library(ggrepel)
volcano.th17$cut_off<- factor(results$cut_off, levels = c('significant', 'insignificant'))

# plot
volcano_th17<- ggplot(results, aes(avg_log2FC, -log10(p_val_adj))) + 
  theme_classic() +
  geom_point(aes(col = cut_off)) +
  scale_color_manual(values = c("black", "red")) +
  geom_vline(xintercept = c(-0.5, 0.5), col = "gray", linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), col = "gray", linetype = 'dashed') +
  theme(plot.title = element_text(hjust = 0.5, size = 8, face = "bold"), 
        axis.title = element_text(face = "bold", size = 12), 
        axis.text = element_text(face = "bold", size=8), 
        legend.title = element_text(size = 8, face = "bold"), 
        legend.text = element_text(size = 6), 
        legend.position = c(0.88, 0.2)) + 
  geom_text_repel(data=results[1:20,], 
                  aes(label=rownames(results[1:20,])), size= 2) +
  ggtitle("Second vs. first arthritis (Th17 like)") 
volcano_th17
``` 


# new Th17 plot !! automatic labelling !! 
``` {r}
# Order your dataset
volcano.th17 <- th17.like[order(th17.like$p_val_adj), ]

# Create data.frame with new categories
results <- as.data.frame(
  mutate(
    volcano.th17,
    cut_off = case_when(
      avg_log2FC > 0.5 & p_val_adj < 0.05 ~ "Upregulated",
      avg_log2FC < -0.5 & p_val_adj < 0.05 ~ "Downregulated",
      TRUE ~ "Not significant"
    )
  ),
  row.names = rownames(volcano.th17)
)

# Update factor levels for better control over plot appearance
results$cut_off <- factor(results$cut_off, levels = c("Downregulated", "Not significant", "Upregulated"))

# Plotting
volcano_th17 <- ggplot(results, aes(avg_log2FC, -log10(p_val_adj))) + 
  theme_classic() +
  geom_point(aes(col = cut_off)) +
  scale_color_manual(values = c("#00AFBB", "grey", "#bb0c00")) +
  geom_vline(xintercept = c(-0.5, 0.5), col = "gray", linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), col = "gray", linetype = 'dashed') +
  theme(plot.title = element_text(hjust = 0.5, size = 8, face = "bold"), 
        axis.title = element_text(face = "bold", size = 12), 
        axis.text = element_text(face = "bold", size = 8), 
        legend.title = element_blank(), 
        legend.text = element_text(size = 6), 
        legend.position = c(0.88, 0.2)) + 
  geom_text_repel(data = results[results$cut_off != "Not significant", ][1:20, ], 
                  aes(label = rownames(results[results$cut_off != "Not significant", ][1:20, ])), 
                  size = 2) +
  ggtitle("Second vs. first arthritis (Th17 like)")

volcano_th17
``` 

# Manual labelling !!! 
``` {r}
# Specify the genes you want to label manually
genes_to_label <- c("IFNG", "TNF", "CD69", "JUN", "DNAJA1", "KLF6", "ATF3", "JUNB", "ICAM1", "IRF4", "ICOS", "HSP90AA1", "NFKBIZ", "H3F3B", "PNRC1", 
                    "CCR7", "MT1E", "GPR171", "TSC22D3", "MT1X", "PASK")  # Replace with actual gene names

# Order your dataset
volcano.th17 <- th17.like[order(th17.like$p_val_adj), ]

# Create data.frame with new categories
results <- as.data.frame(
  mutate(
    volcano.th17,
    cut_off = case_when(
      avg_log2FC > 0.5 & p_val_adj < 0.05 ~ "Upregulated",
      avg_log2FC < -0.5 & p_val_adj < 0.05 ~ "Downregulated",
      TRUE ~ "Not significant"
    )
  ),
  row.names = rownames(volcano.th17)
)

# Update factor levels for better control over plot appearance
results$cut_off <- factor(results$cut_off, levels = c("Downregulated", "Not significant", "Upregulated"))

# Plotting
volcano_th17 <- ggplot(results, aes(avg_log2FC, -log10(p_val_adj))) + 
  theme_classic() +
  geom_point(aes(col = cut_off)) +
  scale_color_manual(values = c("#00AFBB", "grey", "#bb0c00")) +
  geom_vline(xintercept = c(-0.5, 0.5), col = "gray", linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), col = "gray", linetype = 'dashed') +
  theme(plot.title = element_text(hjust = 0.5, size = 10, face = "bold"), 
        axis.title = element_text(face = "bold", size = 10), 
        axis.text = element_text(face = "bold", size = 8), 
        legend.title = element_blank(), 
        legend.text = element_text(size = 6), 
        legend.position = c(0.18, 0.88)) + 
  geom_text_repel(data = results[rownames(results) %in% genes_to_label, ], 
                  aes(label = rownames(results[rownames(results) %in% genes_to_label, ])), 
                  size = 2) +
  ggtitle("Second vs. first arthritis (Th17 like)") + 
  theme(legend.position = "none")

# Display the plot
volcano_th17
``` 

# save high quality plot! 
``` {r}
ggsave("volcano.Th17n.pdf", plot = volcano_th17, height=3, width = 3, units = "in", dpi = 300)
``` 

# gene set enrichment of Th17 like clusters !!! 

# Perform gene set enrichment analysis. === Second vs. first arthritis! 
``` {r}
## perform gene set enrichment 

BiocManager::install("fgsea")
library(fgsea)
th17.like$gene<- rownames(th17.like)
th17.like<- th17.like %>% arrange(desc(avg_log2FC))

#write csv file for deg
write.csv(DEG.activatedCD4, file = "Y:/Recurrent__Arthritis__Single cell/DEG.activatedCD4.csv")

fold_changes<- th17.like$avg_log2FC
names(fold_changes)<- th17.like$gene
``` 

#Load geneset ! 
``` {r}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("fgsea")
Reactome <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.reactome.v2023.1.Hs.symbols.gmt")
hallmark <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/h.all.v2023.1.Hs.symbols.gmt")
KEGG <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.kegg.v2023.1.Hs.symbols.gmt")
GO <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c5.go.bp.v2023.1.Hs.symbols.gmt")

#combine the four gsea
Four_gene_sets <- c(hallmark, KEGG, GO, Reactome)

#getset for all pathway 
Allgenesets <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/msigdb.v2023.1.Hs.symbols.gmt")

# GSEA analysis
gsea_Myo_th17 <- fgsea(pathways = hallmark,
                  stats = fold_changes,
                  eps = 0.0,
                  minSize=15,
                  maxSize=500)

head(gsea_Myo_th17[order(pval), ]) 
```
#save excel file !! 
``` {r}
library(openxlsx)
write.xlsx(gsea_Myo_th17, file = "GSEA.Th17.xlsx")
``` 

#publication !!1 

#Enrichment of th17 !! 
``` {r}
gseaicisecondvsfirst_th17<- read.csv("gsea.Th17.csv", stringsAsFactors = TRUE, header = TRUE)
gseaicisecondvsfirst_data<- gseaicisecondvsfirst_data %>% filter(group== "top")


# ggplot
gseaicisecondvsfirst.th17<- ggplot(gseaicisecondvsfirst_th17, aes(reorder(Pathway, NES), NES, fill= p.adjust))  +
  theme_classic() +
  geom_bar(stat = "identity")  + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 8),
        axis.title.x = element_text(face = "bold", size = 12),
        axis.title.y  = element_blank(), 
        axis.text = element_text(face = "bold", size = 10), 
        plot.title = element_text(hjust = 0.5, face = "bold", size = 10), 
        legend.title = element_text(size= 12, face="bold"), 
        legend.text = element_text(size= 12, face="bold")) + 
  labs(x="Hallmark pathways", y="normalized enrichment score") + 
  coord_flip() + 
  scale_fill_gradient(low = "red", high = "blue", 
                      limits = c(0, max(gseaicisecondvsfirst_th17$p.adjust, na.rm = TRUE))) 
gseaicisecondvsfirst.th17
``` 

#publication 
``` {r}
ggsave("gseasecondvsfirst.th17.pdf", plot = gseaicisecondvsfirst.th17, height=4, width =6, units = "in", dpi = 300)
``` 




``` {r}
TNF_th17<- plotEnrichment(hallmark[["HALLMARK_TNFA_SIGNALING_VIA_NFKB"]],
               fold_changes) + labs(title="TNFA_SIGNALING_VIA_NFKB") + 
  theme_classic() + 
  theme(plot.title = element_blank(), 
        axis.title = element_text(size= 13, face= "bold"), 
        axis.text = element_text(size= 10, face= "bold"))
  
TNF_th17
``` 
# save high quality plot !! 
``` {r}
ggsave("th17_enrichment_TNFA.pdf", plot = TNF_th17, height=3, width = 4, units = "in", dpi = 300)
``` 

# Il2 
``` {r}
IL2_th17<- plotEnrichment(hallmark[["HALLMARK_IL2_STAT5_SIGNALING"]],
               fold_changes) + labs(title="IL2_STAT5_SIGNALING") + 
  theme_classic() + 
  theme(plot.title = element_blank(), 
        axis.title = element_text(size= 13, face= "bold"), 
        axis.text = element_text(size= 10, face= "bold"))
  
IL2_th17
``` 

# HALLMARK_P53_PATHWAY
``` {r}
p53_th17<- plotEnrichment(hallmark[["HALLMARK_P53_PATHWAY"]],
               fold_changes) + labs(title="TNFA_SIGNALING_VIA_NFKB") + 
  theme_classic() + 
  theme(plot.title = element_blank(), 
        axis.title = element_text(size= 13, face= "bold"), 
        axis.text = element_text(size= 10, face= "bold"))
  
p53_th17
``` 
## HALLMARK_MTORC1_SIGNALING

``` {r}
MTORC1_th17<- plotEnrichment(hallmark[["HALLMARK_MTORC1_SIGNALING"]],
               fold_changes) + labs(title="HALLMARK_MTORC1_SIGNALING") + 
  theme_classic() + 
  theme(plot.title = element_blank(), 
        axis.title = element_text(size= 13, face= "bold"), 
        axis.text = element_text(size= 10, face= "bold"))
  
MTORC1_th17
``` 
# combine all the three plots !! 
``` {r}
library(patchwork)
all<- IL2_th17 + p53_th17 + MTORC1_th17
all
``` 


``` {r}
ggsave("th17_enrichment_all.pdf", plot = all, height=3, width = 6, units = "in", dpi = 300)
``` 


# Examine differentially expressed genes ICI-arthritis/Osteoarthritis! 
``` {r}
# Identify differentially expressed genes across conditions
view (T.annotated.subsets@meta.data)
Idents(T.annotated.subsets)
T.annotated.subsets$celltypes<-Idents(T.annotated.subsets)
view (T.annotated.subsets@meta.data)
T.annotated.subsets$arthritis_celltype <- paste(T.annotated.subsets$group, sep = "_", Idents(T.annotated.subsets))
view (T.annotated.subsets@meta.data)
Idents(T.annotated.subsets) <- "arthritis_celltype"
Idents(T.annotated.subsets)

#Calculate differentially expressed genes between first and second arthritis!
prepsct<- PrepSCTFindMarkers(T.annotated.subsets)
effector_cd8_icivsosteo<- FindMarkers(T.annotated.subsets,assay = "SCT", ident.1 = "ICI-arthritis_CD8 effector", 
                               ident.2 = "Osteoarthritis_CD8 effector", min.pct = 0.25, logfc.threshold = 0.25, 
                           verbose = FALSE, 
                           recorrect_umi = FALSE)
head(effector_cd8_icivsosteo, n = 20)


# order it from highest log2FC to lowest

effector_cd8_icivsosteo_order<- effector_cd8_icivsosteo %>% arrange(desc(avg_log2FC))
head(effector_cd8_icivsosteo_order)

effector_cd8_icivsosteo_order<-effector_cd8_icivsosteo_order[effector_cd8_icivsosteo_order$p_val_adj<0.05, ]

# Save deg of cluster 1
write.csv(effector_cd8_icivsosteo_order, file = "degicivsosteo.effector.CD8.csv")
``` 

#deg plot for volcano! 
``` {r}
## Volcano plot @ 1 day 

volcano<- effector_cd8_icivsosteo[order(effector_cd8_icivsosteo$p_val_adj),]

# create data.frame
results<- as.data.frame(mutate(as.data.frame(volcano), 
                                    cut_off=ifelse(volcano$p_val_adj<0.05, "significant", "insignificant")), 
                             row.names=rownames(volcano))
## ggplot plus ggrepel
options(ggrepel.max.overlaps = Inf)
library(ggrepel)
volcano$cut_off<- factor(results$cut_off, levels = c('significant', 'insignificant'))

# plot
volcano_effectorcd8icivsoste<- ggplot(results, aes(avg_log2FC, -log10(p_val_adj))) + 
  theme_classic() +
  geom_point(aes(col = cut_off)) +
  scale_color_manual(values = c("black", "red")) +
  theme(plot.title = element_text(hjust = 0.5, size = 8, face = "bold"), 
        axis.title = element_text(face = "bold", size = 12), 
        axis.text = element_text(face = "bold", size=8), 
        legend.title = element_text(size = 8, face = "bold"), 
        legend.text = element_text(size = 6), 
        legend.position = c(0.88, 0.2)) + 
  geom_text_repel(data=results[1:20,], 
                  aes(label=rownames(results[1:20,])), size= 2) +
  ggtitle("ICI-arthritis vs. Osteoarthritis (effector CD8 cluster)") 
volcano_effectorcd8icivsoste
``` 
# save high quality plot! 
``` {r}
ggsave("volcanoiciosteo_effectorcd8.pdf", plot = volcano_effectorcd8icivsoste, height=2.8, width = 3.25, units = "in", dpi = 300)
```

# new type of volcano plot! 
``` {r}
# Loading relevant libraries 
library(tidyverse) # includes ggplot2, for data visualisation. dplyr, for data manipulation.
library(RColorBrewer) # for a colourful plot
library(ggrepel) # for nice annotations

# Set input path
path <- "C:/Users/Documents/Biostatsquid/Volcano_plots/Tutorial/"  # I can set later
setwd(path)

# Import DGE results
deg.ici.vs.osteo<- read.csv("deg_between conditions/deg_ici_vs_osteo.effector.CD8.csv", row.names = 1)
head(deg.ici.vs.osteo)
``` 



``` {r}
#volcano 
ggplot(data = deg.ici.vs.osteo, aes(avg_log2FC, y= -log10(p_val_adj))) + 
  theme_classic() +
  geom_point() + 
  geom_vline(xintercept = c(-0.5, 0.5), col = "gray", linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), col = "gray", linetype = 'dashed') + 
  geom_point() +  
  theme_set(theme_classic(base_size = 20) +
            theme(
              axis.title.y = element_text(face = "bold", margin = margin(0,20,0,0), size = rel(1.1), color = 'black'),
              axis.title.x = element_text(hjust = 0.5, face = "bold", margin = margin(20,0,0,0), size = rel(1.1), color = 'black'),
              plot.title = element_text(hjust = 0.5)
            ))

```

``` {r}
deg.ici.vs.osteo$diffexpressed<- "NO"
deg.ici.vs.osteo$diffexpressed[deg.ici.vs.osteo$avg_log2FC > 0.5 & deg.ici.vs.osteo$p_val_adj < 0.05] <- "UP"
deg.ici.vs.osteo$diffexpressed[deg.ici.vs.osteo$avg_log2FC < -0.5 & deg.ici.vs.osteo$p_val_adj < 0.05] <- "DOWN"
head(deg.ici.vs.osteo[order(deg.ici.vs.osteo$p_val_adj) & deg.ici.vs.osteo$diffexpressed == 'DOWN', ])
``` 

# Create a new column "delabel" to de, that will contain the name of the top 30 differentially expressed genes (NA in case they are not)
``` {r}
deg.ici.vs.osteo$delabel <- ifelse(deg.ici.vs.osteo$gene_symbol %in% head(deg.ici.vs.osteo[order(deg.ici.vs.osteo$p_val_adj), "gene_symbol"], 20), deg.ici.vs.osteo$gene_symbol, NA)
``` 



``` {r}
volcano_effectorcd8icivsoste<- ggplot(data = deg.ici.vs.osteo,
                                      aes(x = avg_log2FC, y = -log10(p_val_adj), 
                                                                   col = diffexpressed, 
                                          label= delabel)) +
  geom_vline(xintercept = c(-0.5, 0.5), col = "gray", linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), col = "gray", linetype = 'dashed') +
  geom_point(size = 2) +
  theme(plot.title = element_text(hjust = 0.5, size = 11, face = "bold"), 
        axis.title = element_text(face = "bold", size = 12), 
        axis.text = element_text(face = "bold", size=11), 
        legend.title = element_blank(), 
        legend.text = element_text(size = 6.5), 
        legend.position = c(0.90, 0.23)) +
  scale_color_manual(values = c("#00AFBB", "grey", "#bb0c00"), 
                     labels = c("Downregulated", "Not significant", "Upregulated"))  + 
 ggtitle("ICI-arthritis vs. Osteoarthritis (effector CD8 cluster)") + 
   geom_text_repel(max.overlaps = Inf, size=2.3) # To show all labels 
volcano_effectorcd8icivsoste
``` 

# save high quality plot! 
``` {r}
ggsave("volcanoiciosteo_effectorcd8.pdf", plot = volcano_effectorcd8icivsoste, height=4, width = 4.9, units = "in", dpi = 300)
```



``` {r}
volcano_effectorcd8icivsoste<- ggplot(data = deg.ici.vs.osteo,
                                      aes(x = avg_log2FC, y = -log10(p_val_adj), 
                                                                   col = diffexpressed, 
                                          label= delabel)) +
  geom_vline(xintercept = c(-0.5, 0.5), col = "gray", linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), col = "gray", linetype = 'dashed') +
  geom_point(size = 2) +
  theme(plot.title = element_text(hjust = 0.5, size = 8.5, face = "bold"), 
        axis.title = element_text(face = "bold", size = 12), 
        axis.text = element_text(face = "bold", size=11), 
        legend.title = element_blank(), 
        legend.text = element_text(size = 7.3), 
        legend.position = c(0.85, 0.23)) +
  scale_color_manual(values = c("#00AFBB", "grey", "#bb0c00"), 
                     labels = c("Downregulated", "Not significant", "Upregulated"))  + 
 ggtitle("ICI-arthritis vs. Osteoarthritis (effector CD8 cluster)") + 
   geom_text_repel(max.overlaps = Inf, size=2.3) # To show all labels 
volcano_effectorcd8icivsoste
``` 

# new shape 
``` {r}
ggsave("volcanoiciosteo_effectorcd8.new.pdf", plot = volcano_effectorcd8icivsoste, height=4, width = 4, units = "in", dpi = 300)
```

# Perform gene set enrichment analysis. === Second vs. first arthritis! 
``` {r}
## perform gene set enrichment 

BiocManager::install("fgsea")
library(fgsea)
effector_cd8_icivsosteo$gene<- rownames(effector_cd8_icivsosteo)
effector_cd8_icivsosteo<- effector_cd8_icivsosteo %>% arrange(desc(avg_log2FC))

#write csv file for deg
write.csv(DEG.activatedCD4, file = "Y:/Recurrent__Arthritis__Single cell/DEG.activatedCD4.csv")

fold_changes<- effector_cd8_icivsosteo$avg_log2FC
names(fold_changes)<- effector_cd8_icivsosteo$gene
``` 

#Load geneset ! 
``` {r}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("fgsea")
Reactome <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.reactome.v2023.1.Hs.symbols.gmt")
hallmark <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/h.all.v2023.1.Hs.symbols.gmt")
KEGG <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.kegg.v2023.1.Hs.symbols.gmt")
GO <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c5.go.bp.v2023.1.Hs.symbols.gmt")

#combine the four gsea
Four_gene_sets <- c(hallmark, KEGG, GO, Reactome)

#getset for all pathway 
Allgenesets <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/msigdb.v2023.1.Hs.symbols.gmt")

# GSEA analysis
gsea_Myo <- fgsea(pathways = hallmark,
                  stats = fold_changes,
                  eps = 0.0,
                  minSize=15,
                  maxSize=500)

head(gsea_Myo[order(pval), ]) 
```

#save csv file! 
``` {r}
write.csv(gsea_Myo, file = "gsea_ici_osteo.csv")
``` 

# plot enrichment 
``` {r}
library(tidyverse)

## 3 hrs
gseaici_oste<- read.csv("gseaici_osteo.csv", stringsAsFactors = TRUE, header = TRUE)

# ggplot
gseaici<- ggplot(gseaici_oste, aes(reorder(Pathway, NES), NES, fill= padj)) + 
  theme_classic() +
  geom_bar(stat = "identity")  + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 8),
        axis.title.x = element_text(face = "bold", size = 12),
        axis.title.y  = element_blank(), 
        axis.text = element_text(face = "bold", size = 10), 
        plot.title = element_text(hjust = 0.5, face = "bold", size = 10)) + 
  labs(x="Hallmark pathways", y="normalized enrichment score",
       title="ICI-arthritis vs. Osteoarthritis (Effector CD8)") + 
  coord_flip() + 
  scale_fill_gradient(low = "red", high = "blue")
gseaici

# dotplot !! 
gseaicisecondvsfirst <- ggplot(gseaicisecondvsfirst_data, aes(reorder(Pathway, NES), NES, fill = padj)) + 
  theme_bw() +
  geom_point(shape = 21, size = 5) +  # Dot plot with fixed size
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 12),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(), 
        axis.text = element_text(face = "bold", size = 12), 
        plot.title = element_blank()) + 
  labs(x="Hallmark pathways", y="Normalized enrichment score",
       title="Second vs. first arthritis (Effector CD8)") + 
  coord_flip() + 
  scale_fill_gradient(low = "red", high = "blue")
gseaicisecondvsfirst


``` 

``` {r}
ggsave("gseaici.pdf", plot = gseaici, height=3, width =6.5, units = "in", dpi = 300)
``` 

#new shape! 
``` {r}
ggsave("gseaici_new.pdf", plot = gseaici, height=3, width =6.5, units = "in", dpi = 300)
``` 


# make a table plot for a bunch of selected pathways:

``` {r, fig.width= 7, fig.height= 3}
topPathwaysUp <- gsea_Myo[ES > 0][head(order(pval), n=10), pathway]
topPathwaysDown <- gsea_Myo[ES < 0][head(order(pval), n=10), pathway]
topPathways <- c(topPathwaysUp, rev(topPathwaysDown))
table<- plotGseaTable(GO [topPathways], fold_changes, gsea_Myo, 
              gseaParam=0.5)
print(table)
``` 

# KEGG

``` {r}
# GSEA analysis
gsea_Myo_kegg <- fgsea(pathways = GO,
                  stats = fold_changes,
                  eps = 0.0,
                  minSize=15,
                  maxSize=500)

head(gsea_Myo_kegg[order(pval), ]) 

``` 

``` {r}
go<- plotEnrichment(GO[["GOBP_ALPHA_BETA_T_CELL_ACTIVATION"]],
               fold_changes) + labs(title="ALPHA_BETA_T_CELL_ACTIVATION") +
  theme_minimal() + 
  theme(plot.title = element_text(hjust = 0.5, face = "bold"), 
        axis.text = element_text(face = "bold", size= 12), 
        axis.title = element_text(face = "bold", size = 14))

``` 

# Save high quality plot! 
``` {r}
ggsave("go.pdf", plot = go, height=3.5, width = 5, units = "in", dpi = 300)
``` 


# plot enrichment 
``` {r}
library(tidyverse)

## 3 hrs
GSEA<- read.csv("enrichment.csv", stringsAsFactors = TRUE, header = TRUE)

# ggplot
GSEA<- ggplot(GSEA, aes(reorder(pathway, NES), NES, fill= padj)) + 
  theme_classic() +
  geom_bar(stat = "identity")  + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 8),
        axis.title.x = element_text(face = "bold", size = 12),
        axis.title.y  = element_blank(), 
        axis.text = element_text(face = "bold", size = 10), 
        plot.title = element_text(hjust = 0.5, face = "bold", size = 10)) + 
  labs(x="Hallmark pathways", y="Normalized Enrichment score",
       title="Second vs. first arthritis (CD8 effector)") + 
  coord_flip() + 
  scale_fill_gradient(low = "red", high = "blue")
GSEA
``` 


``` {r}
ggsave("GSEA.pdf", plot = GSEA_3hrs, height=3, width =6.5, units = "in", dpi = 300)
```

#Th1 like ! 
``` {r}
# Identify differentially expressed genes across conditions
view (T.annotated.subsets@meta.data)
Idents(T.annotated.subsets)
T.annotated.subsets$celltypes<-Idents(T.annotated.subsets)
view (T.annotated.subsets@meta.data)
T.annotated.subsets$arthritis_celltype <- paste(T.annotated.subsets$type, sep = "_", Idents(T.annotated.subsets))
view (T.annotated.subsets@meta.data)
Idents(T.annotated.subsets) <- "arthritis_celltype"
Idents(T.annotated.subsets)

#Calculate differentially expressed genes between first and second arthritis! 
prepsct<- PrepSCTFindMarkers(T.annotated.subsets)
Th1<-  FindMarkers(T.annotated.subsets,
                                assay = "SCT", 
                                ident.1 = "Second arthritis_Th1-like", 
                               ident.2 = "First arthritis_Th1-like", 
                    min.pct = 0.25, 
                    logfc.threshold = 0.25, 
                    verbose = FALSE, 
                    recorrect_umi = FALSE)
head(Th1, n = 20)


# order it from highest log2FC to lowest
Th1icivsosteo_order<- Th1 %>% arrange(desc(avg_log2FC))
head(Th1icivsosteo_order)


Th1icivsosteo_order<- Th1icivsosteo_order[Th1icivsosteo_order$p_val_adj<0.05, ]

# Save deg of cluster 1
write.csv(Th1icivsosteo_order, file = "deg.th1.csv")
``` 

# new volcano plot for CXCL13 

``` {r}
# Loading relevant libraries 
library(tidyverse) # includes ggplot2, for data visualisation. dplyr, for data manipulation.
library(RColorBrewer) # for a colourful plot
library(ggrepel) # for nice annotations

# Set input path
path <- "C:/Users/Documents/Biostatsquid/Volcano_plots/Tutorial/"  # I can set later
setwd(path)

# Import DGE results
deg.first.vs.second.cxcl13CD4<- read.csv("deg_between conditions/deg_second_vs_first_CXCL13cd4.csv")
head(deg.first.vs.second.cxcl13CD4)
``` 

``` {r}
deg.first.vs.second.cxcl13CD4<- CXCL13secondvsfirst

deg.first.vs.second.cxcl13CD4$diffexpressed<- "NO"
deg.first.vs.second.cxcl13CD4$diffexpressed[deg.first.vs.second.cxcl13CD4$avg_log2FC > 0.5 & deg.first.vs.second.cxcl13CD4$p_val_adj < 0.05] <- "UP"
deg.first.vs.second.cxcl13CD4$diffexpressed[deg.first.vs.second.cxcl13CD4$avg_log2FC < -0.5 & deg.first.vs.second.cxcl13CD4$p_val_adj < 0.05] <- "DOWN"
head(deg.first.vs.second.cxcl13CD4[order(deg.first.vs.second.cxcl13CD4$p_val_adj) & deg.first.vs.second.cxcl13CD4$diffexpressed == 'DOWN', ])
``` 

# Create a new column "delabel" to de, that will contain the name of the top 30 differentially expressed genes (NA in case they are not)
``` {r}
deg.first.vs.second.cxcl13CD4$delabel <- ifelse(deg.first.vs.second.cxcl13CD4$gene_symbol %in% head(deg.first.vs.second.cxcl13CD4[order(deg.first.vs.second.cxcl13CD4$p_val_adj), "gene_symbol"], 30), deg.first.vs.second.cxcl13CD4$gene_symbol, NA)
``` 



``` {r}
volcano_cxcl13CD4_secondvsfirst<- ggplot(data = deg.first.vs.second.cxcl13CD4,
                                      aes(x = avg_log2FC, y = -log10(p_val_adj), 
                                                                   col = diffexpressed, 
                                          label= delabel)) +
  geom_vline(xintercept = c(-0.5, 0.5), col = "gray", linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), col = "gray", linetype = 'dashed') +
  geom_point(size = 2) +
  theme(plot.title = element_text(hjust = 0.5, size = 11, face = "bold"), 
        axis.title = element_text(face = "bold", size = 12), 
        axis.text = element_text(face = "bold", size=11), 
        legend.title = element_blank(), 
        legend.text = element_text(size = 6.5), 
        legend.position = c(0.90, 0.23)) +
  scale_color_manual(values = c("#00AFBB", "grey", "#bb0c00"), 
                     labels = c("Downregulated", "Not significant", "Upregulated"))  + 
 ggtitle("Second vs. first arthritis (PD1hi CXCL13hi CD4)") + 
   geom_text_repel(max.overlaps = Inf, size=2.3) # To show all labels 
volcano_cxcl13CD4_secondvsfirst
``` 

# save high quality plot! 
``` {r}
ggsave("volcano_secondvsfirst_cxcl13CD4.pdf", plot = volcano_cxcl13CD4_secondvsfirst, height=4, width = 4.9, units = "in", dpi = 300)
```


#new volcano for Th1! 
``` {r}
# Loading relevant libraries 
library(tidyverse) # includes ggplot2, for data visualisation. dplyr, for data manipulation.
library(RColorBrewer) # for a colourful plot
library(ggrepel) # for nice annotations

# Set input path
path <- "C:/Users/Documents/Biostatsquid/Volcano_plots/Tutorial/"  # I can set later
setwd(path)

# Import DGE results
deg.first.vs.second.th1<- read.csv("deg_between conditions/deg_second_vs_first_th1.csv")
head(deg.first.vs.second.th1)
``` 

``` {r}
deg.first.vs.second.th1$diffexpressed<- "NO"
deg.first.vs.second.th1$diffexpressed[deg.first.vs.second.th1$avg_log2FC > 0.5 & deg.first.vs.second.th1$p_val_adj < 0.05] <- "UP"
deg.first.vs.second.th1$diffexpressed[deg.first.vs.second.th1$avg_log2FC < -0.5 & deg.first.vs.second.th1$p_val_adj < 0.05] <- "DOWN"
head(deg.first.vs.second.th1[order(deg.first.vs.second.th1$p_val_adj) & deg.first.vs.second.th1$diffexpressed == 'DOWN', ])
``` 

# Create a new column "delabel" to de, that will contain the name of the top 30 differentially expressed genes (NA in case they are not)
``` {r}
deg.first.vs.second.th1$delabel <- ifelse(deg.first.vs.second.th1$gene_symbol %in% head(deg.first.vs.second.th1[order(deg.first.vs.second.th1$p_val_adj), "gene_symbol"], 30), deg.first.vs.second.th1$gene_symbol, NA)
``` 



``` {r}
volcano_th1_secondvsfirst<- ggplot(data = deg.first.vs.second.th1,
                                      aes(x = avg_log2FC, y = -log10(p_val_adj), 
                                                                   col = diffexpressed, 
                                          label= delabel)) +
  geom_vline(xintercept = c(-0.5, 0.5), col = "gray", linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), col = "gray", linetype = 'dashed') +
  geom_point(size = 2) +
  theme(plot.title = element_text(hjust = 0.5, size = 11, face = "bold"), 
        axis.title = element_text(face = "bold", size = 12), 
        axis.text = element_text(face = "bold", size=11), 
        legend.title = element_blank(), 
        legend.text = element_text(size = 6.5), 
        legend.position = c(0.88, 0.8)) +
  scale_color_manual(values = c("#00AFBB", "grey", "#bb0c00"), 
                     labels = c("Downregulated", "Not significant", "Upregulated"))  + 
 ggtitle("Second vs. first arthritis (Th1 like)") + 
   geom_text_repel(max.overlaps = Inf, size=2.3) # To show all labels 
volcano_th1_secondvsfirst
``` 

# save high quality plot! 
``` {r}
ggsave("volcano_secondvsfirst_th1.pdf", plot = volcano_th1_secondvsfirst, height=4, width = 4.9, units = "in", dpi = 300)
```


#deg plot for volcano! 
``` {r}
## Volcano plot @ 1 day 

volcano_Th1<- Th1[order(Th1$p_val_adj),]

# create data.frame
results_Th1<- as.data.frame(mutate(as.data.frame(volcano_Th1), 
                                    cut_off=ifelse(volcano_Th1$p_val_adj<0.05, "significant", "insignificant")), 
                             row.names=rownames(volcano_Th1))
## ggplot plus ggrepel
options(ggrepel.max.overlaps = Inf)
library(ggrepel)
volcano_Th1$cut_off<- factor(results_Th1$cut_off, levels = c('significant', 'insignificant'))

# plot
volcano_Th1plot<- ggplot(results_Th1, aes(avg_log2FC, -log10(p_val_adj))) + 
  theme_classic() +
  geom_point(aes(col = cut_off)) +
  scale_color_manual(values = c("black", "red")) +
  theme(plot.title = element_text(hjust = 0.5, size = 8, face = "bold"), 
        axis.title = element_text(face = "bold", size = 12), 
        axis.text = element_text(face = "bold", size=8), 
        legend.title = element_text(size = 8, face = "bold"), 
        legend.text = element_text(size = 6), 
        legend.position = c(0.2, 0.8)) + 
  geom_text_repel(data=results_Th1[1:30,], 
                  aes(label=rownames(results_Th1[1:30,])), size= 2) +
  ggtitle("Second vs. first arthritis (Th1-like)")
volcano_Th1plot
``` 
# save high quality plot! 
``` {r}
ggsave("volcano.th1.pdf", plot = volcano_Th1plot, height=2.8, width = 3.25, units = "in", dpi = 300)
```


# Perform gene set enrichment analysis. === Second vs. first arthritis! 
``` {r}
## perform gene set enrichment 

BiocManager::install("fgsea")
library(fgsea)
Th1$gene<- rownames(Th1)
Th1<- Th1 %>% arrange(desc(avg_log2FC))

#write csv file for deg
write.csv(DEG.activatedCD4, file = "Y:/Recurrent__Arthritis__Single cell/DEG.activatedCD4.csv")

fold_changes<- Th1$avg_log2FC
names(fold_changes)<- Th1$gene
``` 

#Load geneset ! 
``` {r}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("fgsea")
library(fgsea)
Reactome <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.reactome.v2023.1.Hs.symbols.gmt")
hallmark <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/h.all.v2023.1.Hs.symbols.gmt")
KEGG <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.kegg.v2023.1.Hs.symbols.gmt")
GO <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c5.go.bp.v2023.1.Hs.symbols.gmt")

#combine the four gsea
Four_gene_sets <- c(hallmark, KEGG, GO, Reactome)

#getset for all pathway 
Allgenesets <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/msigdb.v2023.1.Hs.symbols.gmt")

# GSEA analysis
gsea_Myo_th1<- fgsea(pathways = hallmark,
                  stats = fold_changes,
                  eps = 0.0,
                  minSize=15,
                  maxSize=500)

# order it from highest log2FC to lowest
gsea_Myo_cxcl13allgofilter<- gsea_Myo_cxcl13allgo %>% arrange(desc(padj))
head(gsea_Myo_cxcl13allgofilter)


gsea_Myo_cxcl13allgofilter<- gsea_Myo_cxcl13allgofilter[gsea_Myo_cxcl13allgofilter$padj<0.05, ]

# Save enrichment pathways! 
write.csv(gsea_Myo_cxcl13allgofilter, file = "enrichmentgo.cxcl13.csv")
```

# plot enrichment 
``` {r}
library(tidyverse)

## 3 hrs
th1enrich<- read.csv("th1_enrich.csv", stringsAsFactors = TRUE, header = TRUE)

# ggplot
th1path<- ggplot(th1enrich, aes(reorder(Pathway, NES), NES, fill= padj)) + 
  theme_classic() +
  geom_bar(stat = "identity")  + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 8),
        axis.title.x = element_text(face = "bold", size = 12),
        axis.title.y  = element_blank(), 
        axis.text = element_text(face = "bold", size = 10), 
        plot.title = element_text(hjust = 0.5, face = "bold", size = 10)) + 
  labs(x="Hallmark pathways", y="normalized enrichment score",
       title="Second vs. first arthritis (Th1-like)") + 
  coord_flip() + 
  scale_fill_gradient(low = "red", high = "blue")
th1path
``` 

``` {r}
ggsave("gseath1.pdf", plot = th1path, height=3, width =6.5, units = "in", dpi = 300)
``` 

#TNF OF TH1
``` {R}
TNF_TH1<- plotEnrichment(hallmark[["HALLMARK_TNFA_SIGNALING_VIA_NFKB"]],
               fold_changes) + labs(title="IL2_STAT5_SIGNALING") + 
  theme_classic() + 
  theme(plot.title = element_blank(), 
        axis.title = element_text(size= 12, face= "bold"), 
        axis.text = element_text(size= 10, face= "bold"))
TNF_TH1
``` 

#IL2 OF TH1
``` {R}
IL2_TH1<- plotEnrichment(hallmark[["HALLMARK_IL2_STAT5_SIGNALING"]],
               fold_changes) + labs(title="IL2_STAT5_SIGNALING") + 
  theme_classic() + 
  theme(plot.title = element_blank(), 
        axis.title = element_text(size= 12, face= "bold"), 
        axis.text = element_text(size= 10, face= "bold"))
IL2_TH1
``` 
# COMBINE
``` {r}
library(patchwork)
enrich.th1<- TNF_TH1 + IL2_TH1 
enrich.th1

``` 
``` {r}
ggsave("Th1.enrichment.pdf", plot = enrich.th1, height=2.8, width = 6, units = "in", dpi = 300)
```


# cxcl13 second vs. first ! 
``` {r}
# Identify differentially expressed genes across conditions
view (T.annotated.subsets@meta.data)
Idents(T.annotated.subsets)
T.annotated.subsets$celltypes<-Idents(T.annotated.subsets)
view (T.annotated.subsets@meta.data)
T.annotated.subsets$arthritis_celltype <- paste(T.annotated.subsets$type, sep = "_", Idents(T.annotated.subsets))
view (T.annotated.subsets@meta.data)
Idents(T.annotated.subsets) <- "arthritis_celltype"
Idents(T.annotated.subsets)

#Calculate differentially expressed genes between first and second arthritis! 
prepsct<- PrepSCTFindMarkers(T.annotated.subsets)
CXCL13secondvsfirst<-  FindMarkers(T.annotated.subsets,assay = "SCT", ident.1 = "Second arthritis_PD-1hi CXCL13hi CD4", 
                               ident.2 = "First arthritis_PD-1hi CXCL13hi CD4", 
                    min.pct = 0.25, 
                    logfc.threshold = 0.25, 
                    verbose = FALSE, 
                    recorrect_umi = FALSE)
head(CXCL13secondvsfirst, n = 20)
``` 

#deg plot for volcano! 
``` {r}
## Volcano plot @ 1 day 

volcano_cxcl13s.vs.f<- CXCL13secondvsfirst[order(CXCL13secondvsfirst$p_val_adj),]

# create data.frame
results_cxcl13s.vs.f<- as.data.frame(mutate(as.data.frame(volcano_cxcl13s.vs.f), 
                                    cut_off=ifelse(volcano_cxcl13s.vs.f$p_val_adj<0.05, "significant", "insignificant")), 
                             row.names=rownames(volcano_cxcl13s.vs.f))
## ggplot plus ggrepel
options(ggrepel.max.overlaps = Inf)
library(ggrepel)
volcano_cxcl13s.vs.f$cut_off<- factor(results_cxcl13s.vs.f$cut_off, levels = c('significant', 'insignificant'))

# plot
volcano_cxcl13s.vs.f.plot<- ggplot(results_cxcl13s.vs.f, aes(avg_log2FC, -log10(p_val_adj))) + 
  theme_classic() +
  geom_point(aes(col = cut_off)) +
  scale_color_manual(values = c("black", "red")) +
  theme(plot.title = element_text(hjust = 0.5, size = 8, face = "bold"), 
        axis.title = element_text(face = "bold", size = 12), 
        axis.text = element_text(face = "bold", size=8), 
        legend.title = element_text(size = 8, face = "bold"), 
        legend.text = element_text(size = 6), 
        legend.position = c(0.17, 0.85)) + 
  geom_text_repel(data=results_cxcl13s.vs.f[1:20,], 
                  aes(label=rownames(results_cxcl13s.vs.f[1:20,])), size= 2) +
  ggtitle("Second vs. first arthritis (PD1hi CXCL13hi CD4)")
volcano_cxcl13s.vs.f.plot
```
# save high quality plot! 
``` {r}
ggsave("volcano.cxcl13s.vs.f.pdf", plot = volcano_cxcl13s.vs.f.plot, height=2.8, width = 3.25, units = "in", dpi = 300)

```


# Perform gene set enrichment analysis. 
``` {r}
## perform gene set enrichment 

BiocManager::install("fgsea")
library(fgsea)
CXCL13secondvsfirst$gene<- rownames(CXCL13secondvsfirst)
CXCL13secondvsfirst<- CXCL13secondvsfirst %>% arrange(desc(avg_log2FC))

#write csv file for deg
write.csv(DEG.activatedCD4, file = "Y:/Recurrent__Arthritis__Single cell/DEG.activatedCD4.csv")

fold_changes<- CXCL13secondvsfirst$avg_log2FC
names(fold_changes)<- CXCL13secondvsfirst$gene
``` 

#Load geneset ! 
``` {r}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("fgsea")
Reactome <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.reactome.v2023.1.Hs.symbols.gmt")
hallmark <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/h.all.v2023.1.Hs.symbols.gmt")
KEGG <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.kegg.v2023.1.Hs.symbols.gmt")
GO <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c5.go.bp.v2023.1.Hs.symbols.gmt")

#combine the four gsea
Four_gene_sets <- c(hallmark, KEGG, GO, Reactome)

#getset for all pathway 
Allgenesets <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/msigdb.v2023.1.Hs.symbols.gmt")

# GSEA analysis
gsea_Myo_cxcl13svs.f <- fgsea(pathways = hallmark,
                  stats = fold_changes,
                  eps = 0.0,
                  minSize=15,
                  maxSize=500)

head(gsea_Myo_cxcl13svs.f[order(pval), ]) 
```

# plot as dot plots !! 

``` {r}
## 3 hrs
gseaicisecondvsfirst_data<- read.csv("enrichment_cxcl13.csv", stringsAsFactors = TRUE, header = TRUE)
gseaicisecondvsfirst_data<- gseaicisecondvsfirst_data %>% filter(group== "top")

# ggplot
gseaicisecondvsfirst<- ggplot(gseaicisecondvsfirst_data, aes(reorder(Pathway, NES), NES, fill= padj)) + 
  theme_classic() +
  geom_bar(stat = "identity")  + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 8),
        axis.title.x = element_text(face = "bold", size = 12),
        axis.title.y  = element_blank(), 
        axis.text = element_text(face = "bold", size = 10), 
        plot.title = element_text(hjust = 0.5, face = "bold", size = 10)) + 
  labs(x="Hallmark pathways", y="normalized enrichment score",
       title="Second vs. first arthritis (Effector CD8)") + 
  coord_flip() + 
  scale_fill_gradient(low = "red", high = "blue")
gseaicisecondvsfirst

# dot plots !! 
gseaicisecondvsfirst <- ggplot(gseaicisecondvsfirst_data, aes(reorder(Pathway, NES), NES, fill = padj)) + 
  theme_bw() +
  geom_point(shape = 21, size = 5) +  # Dot plot with fixed size
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 12),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(), 
        axis.text = element_text(face = "bold", size = 12), 
        plot.title = element_blank()) + 
  labs(x="Hallmark pathways", y="Normalized enrichment score",
       title="Second vs. first arthritis (Effector CD8)") + 
  coord_flip() + 
  scale_fill_gradient(low = "red", high = "blue")
gseaicisecondvsfirst
ggsave("gseasecondvsfirst_cxcl13.tiff", plot = gseaicisecondvsfirst, height=4, width =7, units = "in", dpi = 300)
``` 

``` {r}
TNF_cxcl13<- plotEnrichment(hallmark[["HALLMARK_TNFA_SIGNALING_VIA_NFKB"]],
               fold_changes) + labs(title="IL2_STAT5_SIGNALING") + 
  theme_classic() + 
theme(plot.title = element_blank(), 
        axis.title = element_text(size= 10, face= "bold"), 
        axis.text = element_text(size= 6, face= "bold"))
  
TNF_cxcl13
``` 
#inflammatory responses !!!
``` {r}
inflamm_cxcl13<- plotEnrichment(hallmark[["HALLMARK_INFLAMMATORY_RESPONSE"]],
               fold_changes) + labs(title="IL2_STAT5_SIGNALING") + 
  theme_classic() + 
 theme(plot.title = element_blank(), 
        axis.title = element_text(size= 10, face= "bold"), 
        axis.text = element_text(size= 6, face= "bold"))
inflamm_cxcl13
``` 

#IFNG !!!
``` {r}
IFNG_cxcl13<- plotEnrichment(hallmark[["HALLMARK_INTERFERON_GAMMA_RESPONSE"]],
               fold_changes) + labs(title="IL2_STAT5_SIGNALING") + 
  theme_classic() + 
  theme(plot.title = element_blank(), 
        axis.title = element_text(size= 10, face= "bold"), 
        axis.text = element_text(size= 6, face= "bold"))
IFNG_cxcl13
``` 
#il2 !!!
``` {r}
il2_cxcl13<- plotEnrichment(hallmark[["HALLMARK_IL2_STAT5_SIGNALING"]],
               fold_changes) + labs(title="IL2_STAT5_SIGNALING") + 
  theme_classic() + 
  theme(plot.title = element_blank(), 
        axis.title = element_text(size= 10, face= "bold"), 
        axis.text = element_text(size= 6, face= "bold"))
il2_cxcl13
``` 
# Combine all the figures !!! 
``` {r}
library(patchwork)
cxcl13.enrichment<- (TNF_cxcl13 + inflamm_cxcl13 + IFNG_cxcl13 + il2_cxcl13) + 
  plot_layout(ncol = 4)
cxcl13.enrichment
``` 
``` {r}
ggsave("cxcl13.enrichment.pdf", plot = cxcl13.enrichment, height=2.8, width = 6, units = "in", dpi = 300)
``` 


# Treg Deg 

# Examine differentially expressed genes between first/second arthritis
``` {r}
# Identify differentially expressed genes across conditions
view (T.annotated.subsets@meta.data)
Idents(all.annotated.subsets)
all.annotated.subsets$celltypes<-Idents(T.annotated.subsets)
view (all.annotated.subsets@meta.data)
T.annotated.subsets$arthritis_celltype <- paste(T.annotated.subsets$type, sep = "_", Idents(all.annotated.subsets))
view (all.annotated.subsets@meta.data)
Idents(T.annotated.subsets) <- "arthritis_celltype"
Idents(T.annotated.subsets)

#Calculate differentially expressed genes between first and second arthritis! 
prepsct<- PrepSCTFindMarkers(T.annotated.subsets)
Treg<-  FindMarkers(T.annotated.subsets,assay = "SCT", ident.1 = "Second arthritis_Treg", 
                               ident.2 = "First arthritis_Treg", 
                    min.pct = 0.25, 
                    logfc.threshold = 0.25, 
                    verbose = FALSE)
head(Treg, n = 20)
``` 

#deg plot for volcano! 
``` {r}
## Volcano plot @ 1 day 

volcano_treg<- Treg[order(Treg$p_val_adj),]

# create data.frame
results_treg<- as.data.frame(mutate(as.data.frame(volcano_treg), 
                                    cut_off=ifelse(volcano_treg$p_val_adj<0.05, "significant", "insignificant")), 
                             row.names=rownames(volcano_treg))
## ggplot plus ggrepel
options(ggrepel.max.overlaps = Inf)
library(ggrepel)
volcano_treg$cut_off<- factor(results_treg$cut_off, levels = c('significant', 'insignificant'))

# plot
volcano_treg<- ggplot(results_treg, aes(avg_log2FC, -log10(p_val_adj))) + 
  theme_classic() +
  geom_point(aes(col = cut_off)) +
  scale_color_manual(values = c("black", "red")) +
  theme(plot.title = element_text(hjust = 0.5, size = 8, face = "bold"), 
        axis.title = element_text(face = "bold", size = 12), 
        axis.text = element_text(face = "bold", size=8), 
        legend.title = element_text(size = 8, face = "bold"), 
        legend.text = element_text(size = 6), 
        legend.position = c(0.2, 0.8)) + 
  geom_text_repel(data=results_treg[1:20,], 
                  aes(label=rownames(results_treg[1:20,])), size= 2) +
  ggtitle("Second vs. first arthritis (Treg cluster)")
volcano_treg
``` 
# save high quality plot! 
``` {r}
ggsave("volcano.Treg.pdf", plot = volcano_treg, height=2.8, width = 3, units = "in", dpi = 300)

```

# Perform gene set enrichment analysis. 
``` {r}
## perform gene set enrichment 

BiocManager::install("fgsea")
library(fgsea)
Treg$gene<- rownames(Treg)
Treg<- Treg %>% arrange(desc(avg_log2FC))

#write csv file for deg
write.csv(DEG.activatedCD4, file = "Y:/Recurrent__Arthritis__Single cell/DEG.activatedCD4.csv")

fold_changes<- Treg$avg_log2FC
names(fold_changes)<- Treg$gene
``` 

#Load geneset ! 
``` {r}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("fgsea")
Reactome <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.reactome.v2023.1.Hs.symbols.gmt")
hallmark <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/h.all.v2023.1.Hs.symbols.gmt")
KEGG <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.kegg.v2023.1.Hs.symbols.gmt")
GO <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c5.go.bp.v2023.1.Hs.symbols.gmt")

#combine the four gsea
Four_gene_sets <- c(hallmark, KEGG, GO, Reactome)

#getset for all pathway 
Allgenesets <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/msigdb.v2023.1.Hs.symbols.gmt")

# GSEA analysis
gsea_Myo_treg <- fgsea(pathways = hallmark,
                  stats = fold_changes,
                  eps = 0.0,
                  minSize=15,
                  maxSize=500)

head(gsea_Myo_treg[order(pval), ]) 
```
``` {r}
treg_enrich<- as.data.frame(gsea_Myo_treg)
write.csv(treg_enrich, file = "enrichment_treg.csv", row.names = FALSE)
write.table(gsea_Myo_treg, file = "gsea_treg.csv", 
            sep = ",", 
            quote = TRUE, 
            row.names = FALSE)

# write rds
saveRDS(gsea_Myo_treg, file = "Treg.enrich.rds")
treg_enrich<- read_rds("Treg.enrich.rds")
``` 


# plot 
``` {r}
library(tidyverse)

## 3 hrs
GSEA_treg<- read.csv("gsea_treg.csv", stringsAsFactors = TRUE, header = TRUE)

# ggplot
GSEA_treg<- ggplot(GSEA_treg, aes(reorder(Pathway, NES), NES, fill= padj)) + 
  theme_classic() +
  geom_bar(stat = "identity")  + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 8),
        axis.title.x = element_text(face = "bold", size = 12),
        axis.title.y  = element_blank(), 
        axis.text = element_text(face = "bold", size = 10), 
        plot.title = element_text(hjust = 0.5, face = "bold", size = 10)) + 
  labs(x="Hallmark pathways", y="Normalized Enrichment score",
       title="Second vs. first arthritis (Regulatory T)") + 
  coord_flip() + 
  scale_fill_gradient(low = "red", high = "blue")
GSEA_treg
``` 

``` {r}
ggsave("GSEA_treg.pdf", plot = GSEA_treg, height=3, width =6.5, units = "in", dpi = 300)
``` 


#save pdf 
``` {r}
ggsave("dot.plot.pdf", plot = Dot.plot, height=7, width = 6.5, units = "in", dpi = 300)
``` 

#Perform over-representation !
``` {r}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("org.Hs.eg.db")
library(clusterProfiler)
significant_genes_map<- clusterProfiler::bitr(geneID = effector_cd8,
                      fromType="SYMBOL", toType="ENTREZID",
                      OrgDb="org.Hs.eg.db")

enrichgo<- clusterProfiler::enrichGO(effector_cd8, "org.Hs.eg.db", keyType = "SYMBOL", 
                                     ont = "BP", 
                                     minGSSize = 50)
```

# Calculate cluster frequency of T cell subclusters! 

#Cluster Frequency ======! 

``` {r}
#view seurat obj
view(all.annotated.subsets)

#set file path and create directory 
figurePath <- file.path("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis")
if(!dir.exists(figurePath)){
    dir.create("cluster_frequency_T_cell", recursive = T)
}

#set working directory if needed!
setwd(figurePath)

# patient 164S
p.164s <- subset(T.annotated.subsets, subset = orig.ident == "164S")
view(p.164s)

# cluster_frequency
cluster_frequency.164s <- p.164s@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.164s")

#write csv file to save the result
write.csv(cluster_frequency.164s, file="cluster_frequency_T_cell/cluster_frequency.164S.CSV")


# patient 164S2
p.164S2 <- subset(T.annotated.subsets, subset = orig.ident == "164S2")
view(p.164S2)

# cluster_frequency
cluster_frequency.164S2 <- p.164S2@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.164S2")

#write csv file to save the result
write.csv(cluster_frequency.164S2, file="cluster_frequency_T_cell/cluster_frequency.164S2.CSV")


# patient 184S2
p.184S2 <- subset(T.annotated.subsets, subset = orig.ident == "184S2")
view(p.184S2)

# cluster_frequency
cluster_frequency.184S2 <- p.184S2@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.184S2")

#write csv file to save the result
write.csv(cluster_frequency.184S2, file="cluster_frequency_T_cell/cluster_frequency.184S2.CSV")

# patient 184S3
p.184S3 <- subset(T.annotated.subsets, subset = orig.ident == "184S3")
view(p.184S3)

# cluster_frequency
cluster_frequency.184S3 <- p.184S3@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.184S3")

#write csv file to save the result
write.csv(cluster_frequency.184S3, file="cluster_frequency_T_cell/cluster_frequency.184S3.CSV")


# patient 218S
p.218S <- subset(T.annotated.subsets, subset = orig.ident == "218S")
view(p.218S)

# cluster_frequency
cluster_frequency.218S <- p.218S@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.218S")

#write csv file to save the result
write.csv(cluster_frequency.218S, file="cluster_frequency_T_cell/cluster_frequency.218S.CSV")

 #patient 164S2
p.218S2 <- subset(T.annotated.subsets, subset = orig.ident == "218S2")
view(p.218S2)

# cluster_frequency
cluster_frequency.218S2 <- p.218S2@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.218S2")

#write csv file to save the result
write.csv(cluster_frequency.218S2, file="cluster_frequency_T_cell/cluster_frequency.218S2.CSV")


# patient SA5
p.SA5 <- subset(T.annotated.subsets, subset = orig.ident == "SA5")
view(p.SA5)

# cluster_frequency
cluster_frequency.SA5 <- p.SA5@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.SA5")

#write csv file to save the result
write.csv(cluster_frequency.SA5, file="cluster_frequency_T_cell/cluster_frequency.SA5.CSV")

# patient SA6
p.SA6 <- subset(T.annotated.subsets, subset = orig.ident == "SA6")
view(p.SA6)

# cluster_frequency
cluster_frequency.SA6 <- p.SA6@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.SA6")

#write csv file to save the result
write.csv(cluster_frequency.SA6, file="cluster_frequency_T_cell/cluster_frequency.SA6.CSV")

# patient SA7
p.SA7 <- subset(T.annotated.subsets, subset = orig.ident == "SA7")
view(p.SA7)

# cluster_frequency
cluster_frequency.SA7 <- p.SA7@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.SA7")

#write csv file to save the result
write.csv(cluster_frequency.SA7, file="cluster_frequency_T_cell/cluster_frequency.SA7.CSV")
``` 

#plot cluster frequency 
``` {r}
#read csv file back in 
library(tidyverse)
proportion<- read.csv("cluster_frequency_T_cell/cluster_frequency.T.csv", stringsAsFactors = TRUE)
print(proportion)
``` 
## Th17 like 
``` {r}
Th17 <- proportion %>% filter_all(any_vars(. == " Th17 like cell"))
Th17$Group<- factor(Th17$Group, levels = c("osteoarthritis", "ICI-arthritis"))
Th17$Arthritis<- factor(Th17$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))


#visualization ! 
p1<- ggplot(Th17, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_text(size = 8, face = "bold"), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% of total cells", title = "Th17 like")
p1

``` 

# CD4 Tcm

``` {r}
cd4tcm <- proportion %>% filter(cluster_name == "CD4 Tcm")
cd4tcm$Group<- factor(cd4tcm$Group, levels = c("osteoarthritis", "ICI-arthritis"))
cd4tcm$Arthritis<- factor(cd4tcm$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))


#visualization ! 
p2<- ggplot(cd4tcm, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% of total CD3", title = "CD4 Tcm")
p2
``` 
``` {r}
wilcox.test(frequency ~ Group, data=cd4tcm)
t.test(frequency ~ Group, data=cd4tcm)
``` 

# CD4 Tcm

``` {r}
activatedcd4 <- proportion %>% filter(cluster_name == "Naive CD4")
activatedcd4$Group<- factor(activatedcd4$Group, levels = c("osteoarthritis", "ICI-arthritis"))
activatedcd4$Arthritis<- factor(activatedcd4$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))


#visualization ! 
p3<- ggplot(activatedcd4, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none")  + 
  labs(x= "Groups", y= "% Contaminated CD4 of CD3", title = "Naive CD4")
p3
``` 

# PD-1hi CXCL13hi CD4+

``` {r}
CD4CXCL13 <- proportion %>% filter(cluster_name == "PD-1hi CXCL13hi CD4")
CD4CXCL13$Group<- factor(CD4CXCL13$Group, levels = c("osteoarthritis", "ICI-arthritis"))
CD4CXCL13$Arthritis<- factor(CD4CXCL13$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))


#visualization ! 
p4<- ggplot(CD4CXCL13, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% of total CD3", title = "PD1CXCL13hi CD4")
p4
``` 

#  Recently activated CD4

``` {r}
RecentActivated <- proportion %>% filter(cluster_name== " Recently activated CD4")
RecentActivated$Group<- factor(RecentActivated$Group, levels = c("osteoarthritis", "ICI-arthritis"))
RecentActivated$Arthritis<- factor(RecentActivated$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))


#visualization ! 
p5<- ggplot(RecentActivated, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% of total CD3", title = "Recently Activated CD4")
p5
``` 
#  Th1 like
``` {r}
Th1 <- proportion %>% filter(cluster_name== "Th1-like")
Th1$Group<- factor(Th1$Group, levels = c("osteoarthritis", "ICI-arthritis"))
Th1$Arthritis<- factor(Th1$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))

#visualization ! 
p6<- ggplot(Th1, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none")+ 
  labs(x= "Groups", y= "% Th1 of CD3", title = "Th1 like")
p6
``` 
``` {r}
wilcox.test(frequency ~ Group, data=Th1)
t.test(frequency ~ Group, data=Th1)
``` 

#  Treg
``` {r}
Treg <- proportion %>% filter(cluster_name== "Treg")
Treg$Group<- factor(Treg$Group, levels = c("osteoarthritis", "ICI-arthritis"))
Treg$Arthritis<- factor(Treg$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))


#visualization !
p7<- ggplot(Treg, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% Treg of CD3", title = "Treg")
p7
``` 
``` {r}
wilcox.test(frequency ~ Group, data=Treg)
t.test(frequency ~ Group, data=Treg)
``` 

#  ILC
``` {r}
ILC <- proportion %>% filter(cluster_name== "gdT")
ILC$Group<- factor(ILC$Group, levels = c("osteoarthritis", "ICI-arthritis"))
ILC$Arthritis<- factor(ILC$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))

#visualization ! 
p8<- ggplot(ILC, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% gamma/delta of CD3", title = "gd T")
p8
``` 
#   CD8 effector
``` {r}
CD8effector <- proportion %>% filter(cluster_name == " CD8 effector")
CD8effector$Group<- factor(CD8effector$Group, levels = c("osteoarthritis", "ICI-arthritis"))
CD8effector$Arthritis<- factor(CD8effector$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))

#visualization ! 
p9<- ggplot(CD8effector, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% effector CD8 of CD3", title = "Effector CD8")
p9
``` 
``` {r}
proportion1<- (p1|p2|p3|p4)/(p5|p6|p7|p8|p9)
ggsave("immune_T.pdf", plot = proportion1, height=3, width = 5, units = "in", dpi = 300)
``` 

#   CD8 effector
``` {r}
effectorcd8 <- proportion %>% filter(cluster_name == "Effector CD8")
effectorcd8$Group<- factor(effectorcd8$Group, levels = c("osteoarthritis", "ICI-arthritis"))
effectorcd8$Arthritis<- factor(effectorcd8$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))

#visualization ! 
p10<- ggplot(effectorcd8, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.5
  ), axis.title.y = element_text(size = 8, face = "bold"), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=7), 
  axis.text.x = element_text(angle=90,hjust=0.95,vjust=0.2, face = "bold", size= 6), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% of total cells", title = "Effector CD8")
p10
``` 

#   CD8 Tem
``` {r}
CD8Tem<- proportion %>% filter(cluster_name == "CD8 Tem")
CD8Tem$Group<- factor(CD8Tem$Group, levels = c("osteoarthritis", "ICI-arthritis"))
CD8Tem$Arthritis<- factor(CD8Tem$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))

#visualization ! 
p11<- ggplot(CD8Tem, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5)+
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.4
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text.y = element_text(size = 7),
  axis.text.x = element_text(angle=90,hjust=0.95,vjust=0.2, face = "bold", size= 6), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% CD8Tem of total cells", title = "CD8Tem")
p11
```
``` {r}
wilcox.test(frequency ~ Group, data=CD8Tem)
t.test(frequency ~ Group, data=CD8Tem)
``` 


#  CD8 Trm
``` {r}
CD8Trm<- proportion %>% filter(cluster_name == "CD8 Trm")
CD8Trm$Group<- factor(CD8Trm$Group, levels = c("osteoarthritis", "ICI-arthritis"))
CD8Trm$Arthritis<- factor(CD8Trm$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))

#visualization ! 
p12<- ggplot(CD8Trm, aes(x= Group, y= frequency, fill= Group)) +
 theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5)+
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.4
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text.y = element_text(size = 7),
  axis.text.x = element_text(angle=90,hjust=0.95,vjust=0.2, face = "bold", size= 6), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% CD8 Trm of CD3", title = "CD8 Trm")
p12
```
``` {r}
wilcox.test(frequency ~ Group, data=CD8Trm)
t.test(frequency ~ Group, data=CD8Trm)
``` 


#  cytotoxic T
``` {r}
cytotoxicT<- proportion %>% filter(cluster_name == "Cytotoxic T")
cytotoxicT$Group<- factor(cytotoxicT$Group, levels = c("osteoarthritis", "ICI-arthritis"))
cytotoxicT$Arthritis<- factor(cytotoxicT$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))

#visualization ! 
p13<- ggplot(cytotoxicT, aes(x= Group, y= frequency, fill= Group)) +
   theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5)+
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.4
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text.y = element_text(size = 7),
  axis.text.x = element_text(angle=90,hjust=0.95,vjust=0.2, face = "bold", size= 6), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% Cytotoxic T of CD3", title = "Cytotoxic T")
p13
```

#  NKG7+ CD8+ T cells
``` {r}
NKG7cd8<- proportion %>% filter(cluster_name == "NKG7+ CD8+ T cells")
NKG7cd8$Group<- factor(NKG7cd8$Group, levels = c("osteoarthritis", "ICI-arthritis"))
NKG7cd8$Arthritis<- factor(NKG7cd8$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))

#visualization ! 
p14<- ggplot(NKG7cd8, aes(x= Group, y= frequency, fill= Group)) +
 theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5)+
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.4
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text.y = element_text(size = 7),
  axis.text.x = element_text(angle=90,hjust=0.95,vjust=0.2, face = "bold", size= 6), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% of total CD3", title = "NKG7+ CD8")
p14
```

# KLRB1hi CD8 Tem
``` {r}
KLRB1hicd8<- proportion %>% filter(cluster_name == " KLRB1hi CD8 Tem")
KLRB1hicd8$Group<- factor(KLRB1hicd8$Group, levels = c("osteoarthritis", "ICI-arthritis"))
KLRB1hicd8$Arthritis<- factor(KLRB1hicd8$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))

#visualization ! 
p15<- ggplot(KLRB1hicd8, aes(x= Group, y= frequency, fill= Group)) +
   theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5)+
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.4
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text.y = element_text(size = 7),
  axis.text.x = element_text(angle=90,hjust=0.95,vjust=0.2, face = "bold", size= 6), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% KLRB1hi CD8 Tem of CD3", title = "KLRB1hi CD8 Tem")
p15
```
# MAIT
``` {r}
MAIT<- proportion %>% filter(cluster_name == "MAIT")
MAIT$Group<- factor(MAIT$Group, levels = c("osteoarthritis", "ICI-arthritis"))
MAIT$Arthritis<- factor(MAIT$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))

#visualization ! 
p16<- ggplot(MAIT, aes(x= Group, y= frequency, fill= Group)) +
 theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5)+
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.4
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text.y = element_text(size = 7),
  axis.text.x = element_text(angle=90,hjust=0.95,vjust=0.2, face = "bold", size= 6), 
  legend.position = "none")  + 
  labs(x= "Groups", y= "% MAIT of CD3", title = "MAIT")
p16
```
# CyclingT
``` {r}
CyclingT<- proportion %>% filter(cluster_name == "Cycling T")
CyclingT$Group<- factor(CyclingT$Group, levels = c("osteoarthritis", "ICI-arthritis"))
CyclingT$Arthritis<- factor(CyclingT$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))

#visualization ! 
p17<- ggplot(CyclingT, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5)+
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.4
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text.y = element_text(size = 7),
  axis.text.x = element_text(angle=90,hjust=0.95,vjust=0.2, face = "bold", size= 6), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% CyclingT of CD3", title = "CyclingT")
p17
```
# NKT 
``` {r}
NKT <- proportion %>% filter(cluster_name == " NKT")
NKT$Group<- factor(NKT$Group, levels = c("osteoarthritis", "ICI-arthritis"))
NKT$Arthritis<- factor(NKT$Arthritis, levels = c("osteoarthritis", "first arthritis", "second arthritis"))

#visualization ! 
p18<- ggplot(NKT, aes(x= Group, y= frequency, fill= Group)) +
  theme_classic() +
  geom_boxplot(size= 0.5, alpha= 0.5) +
   geom_point(position=position_jitterdodge(jitter.width=0.3, dodge.width = 0.3), 
             aes(color=Arthritis), show.legend = TRUE, size= 2.5)+
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.4
  ), axis.title.y = element_blank(), 
  axis.title.x = element_blank(),
  axis.text.y = element_text(size = 7),
  axis.text.x = element_text(angle=90,hjust=0.95,vjust=0.2, face = "bold", size= 6), 
  legend.position = "none")  + 
  labs(x= "Groups", y= "% NKT of CD3", title = "NKT")
p18
```

``` {r}
#Last 10 
proportion2<- (p1|p2|p3|p4|p5|p6|p7|p8|p9)/(p10|p11|p12|p13|p14|p15|p16|p17|p18)
ggsave("immune_T.1.pdf", plot = proportion2, height=3.4, width = 8.5, units = "in", dpi = 300)
``` 

#plot stack bargraph! 
``` {r}
stacked_T<- read.csv("cluster_frequency_T_cell/cluster_frequency.T.csv", stringsAsFactors = TRUE)
head(stacked_T)
``` 

# Stack_bar graph! 
``` {r}
library(tidyverse)
colours = c( "#F9062B", "#062FF9","#F6AE2D","#F26157", "#F9ECCC", "#679289", "#33658A","#86BBD8")

# order the factor level 
stacked_T$Arthritis<- factor(stacked_T$Arthritis, 
                                levels = c("osteoarthritis", "first arthritis", "second arthritis"))

stacked_T$cluster_name<- factor(stacked_T$cluster_name, 
                                levels = c("CD8 effector", 
                                           "CD8 Tem",
                                           "NKG7 CD8", 
                                           "CD8 Trm",
                                           " KLRB1hi CD8 Tem",
                                           "gdT", 
                                           " Th17 like cell", 
                                           "Treg",
                                           "PD-1hi CXCL13hi CD4", 
                                           "CD4 Tcm", 
                                           "Th1-like",
                                           "Recently activated CD4", 
                                           "MAIT", 
                                           "Naive CD4", 
                                           "Cycling T", 
                                           " NKT"
                                           ))
#ggplot
CD3_fraction<- ggplot(stacked_T, aes(x= cluster_name, y= frequency, fill= Arthritis)) + 
  theme_bw() + 
  geom_bar(position = "stack", stat = "identity", color = "black") + 
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.4
  ), axis.title.y = element_text(size = 12, face = "bold"), 
  axis.title.x = element_blank(),
  axis.text.y = element_text(size = 12, face = "bold"),
  axis.text.x = element_text(angle=90,hjust=0.95,vjust=0.2, face = "bold", size= 10), 
  legend.position = "top", 
  legend.title = element_blank(), 
  legend.text = element_text(face = "bold", size= 13), 
  panel.grid = element_blank())  +
 scale_fill_manual(values = colours) +
  labs(y= "Fraction of total CD3")
CD3_fraction
```


#save high quality plot! 
``` {r}
# save plot1 
ggsave("CD3_fraction.pdf", plot = CD3_fraction, height=4.5, width = 6, units = "in", dpi = 300)
```


# Try boxplot 
``` {r}
CD3_fraction= ggplot(stacked_T, aes(x= cluster_name, y= frequency, fill= Arthritis)) + 
  theme_bw() + 
  geom_boxplot(alpha= 0.5) + 
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.4
  ), axis.title.y = element_text(size = 12, face = "bold"), 
  axis.title.x = element_blank(),
  axis.text.y = element_text(size = 12, face = "bold"),
  axis.text.x = element_text(angle=90,hjust=0.95,vjust=0.2, face = "bold", size= 10), 
  legend.position = "top", 
  legend.title = element_blank(), 
  legend.text = element_text(face = "bold", size= 13)) + 
  labs(y= "Fraction of total CD3")
CD3_fraction
``` 
#plot individual patients

``` {r}
CD3_fraction= ggplot(stacked_T, aes(x= data_set, y= frequency, fill= cluster_name)) + 
  theme_minimal() + 
 geom_bar(position = "stack", stat = "identity", color= "black") + 
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=5.4
  ), axis.title.y = element_text(size = 12, face = "bold"), 
  axis.title.x = element_blank(),
  axis.text.y = element_text(size = 12, face = "bold"),
  axis.text.x = element_text(angle=90,hjust=0.95,vjust=0.2, face = "bold", size= 12), 
  legend.position = "right", 
  legend.title = element_text(face = "bold"), 
  legend.text = element_text(face = "bold", size= 7), 
  legend.key.size = unit(0.5, "cm")) + 
  labs(y= "Fraction of total CD3")
CD3_fraction
``` 


``` {r}
ggsave("CD3_fraction.pdf", plot = CD3_fraction, height=4.5, width = 7, units = "in", dpi = 300)
```


# Subset CD4 
``` {r}
annotated.plot
``` 


# CD4 cluster
``` {r}
CD4<- subset(T.annotated.subsets, idents = c("Activated CD4", "Th17 like", "CD4 Tcm", "Treg", "Naive CD4", "PD-1hi CXCL13hi CD4", "Th1-like", "MAIT"))
DimPlot(CD4, label = TRUE) +
  theme_void() +
  NoLegend() +
  ggtitle("subset all CD4") +
  theme(plot.title = element_text(hjust = 0.5))
``` 


# Run SCT again 
``` {r}
#SCTransform
CD4.subcluster<- SCTransform(CD4, 
                         method= "glmGamPoi", 
                         verbose = FALSE)

#Run PCA
CD4.subcluster<- RunPCA(CD4.subcluster, 
                        verbose = FALSE)

saveRDS(T.subcluster, file = "T.subcluster.sct.pca.rds")  #save on 01/31/2024

#Elbow plot
T.subcluster<- read_rds("T.subcluster.sct.pca.rds")

ElbowPlot(CD4.subcluster, ndims = 50,reduction = "pca") +
  ggtitle("Number of principle component")  + 
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))
``` 

# Trajectory analysis: Perform Monocle3 

``` {r}
#load Seurat obj! 
library(Seurat)
library(monocle3)
library(SeuratWrappers)

#load the Seurat obj! 
seurat.obj.T<- read_rds("umap.obj.pc15.rds")

#Dimplot ! 
DimPlot(seurat.obj.T, reduction = "umap", label = TRUE) +
  theme_minimal() +
  NoLegend()
``` 

``` {r, fig.height= 7, fig.width= 7}
FeaturePlot(seurat.obj.T, features = c("SELL", "TCF7", "PDCD1", "CXCL13", "TOX", "CXCR5"), pt.size = 0.2,
    ncol = 2, label = TRUE, label.size = 3)
``` 

``` {r}
ggsave("Feature.tiff", plot = featureplot, height=5, width = 6, units = "in", dpi = 300)
``` 

# Seurat Wrap 

``` {r}
# 2. Convert to Seurat object to cell_data_set object ------
cds <- as.cell_data_set(seurat.obj.T, group.by='ident') # SeuratWrappers
DefaultAssay(seurat.obj.T)
``` 


# Learn the trajectory graph
``` {r}
cds <- cluster_cells(cds, resolution = 1e-5)  #fit with Seurat ! 

plot_cells(cds, color_cells_by = 'ident', 
           label_groups_by_cluster = FALSE,
           group_label_size = 5) +
  theme(legend.position = "right")  # already run and get the same graph! 
```


# trajectory analysis
``` {r}
cds <- learn_graph(cds, use_partition = FALSE)

# plot
plot_cells(cds, color_cells_by = 'ident', 
           label_groups_by_cluster = FALSE,
           group_label_size = 5) +
  theme_minimal() +
  theme(legend.position = "none")
```
 

``` {r}
ggsave("cd.traj.tiff", plot = cd, height=5, width = 5.5, units = "in", dpi = 300)
```


# Create gene annotation file
``` {r}
fData(cds) # gene_annotation
rownames(fData(cds))[1:10]
fData(cds)$gene_short_name <- rownames(fData(cds))
fData(cds)
``` 

# Order the cells by pseudotime : https://ucdavis-bioinformatics-training.github.io/2021-August-Advanced-Topics-in-Single-Cell-RNA-Seq-Trajectory-and-Velocity/data_analysis/monocle_fixed

# Annotation ! 
*0= Th17 like cell*
*1= CD8 effector*
*2= ILC*
*3= Treg*
*4= CD4 Tcm* 
*5= CD8 Tem* 
*6= PD-1hi CXCL13hi CD4+*
*7= Effector CD8* 
*8= Th1-like*
*9= Cytotoxic T*
*10= Recently activated CD4*
*11= Treg*
*12= NKG7+ CD8+ T cells*
*13= NKT*
*14= CD8 Trm*
*15= MAIT*
*16= Cycling T*
*17= KLRB1hi CD8 Tem*
*18= mixed CD4*

# one node! 
``` {r, message= FALSE}
cds <- order_cells(cds)

#plot_cells in pseudotime
trjectory<- plot_cells(cds, color_cells_by = 'pseudotime', 
           label_branch_points = FALSE, label_leaves = FALSE) + 
  theme_classic() + 
  NoAxes()
trjectory
``` 


``` {r}
ggsave("trajectory.pdf", plot = trjectory, height=5, width = 4, units = "in", dpi = 300)
```

# two nodes! 

``` {r}
cds <- order_cells(cds)

#plot_cells in pseudotime
plot_cells(cds, color_cells_by = 'pseudotime', 
           label_branch_points = FALSE, label_leaves = FALSE) + 
  theme_minimal()

``` 

#Perform grid optimization 
``` {r}
tarfile <- "monocle3/monocle3.tar.gz"
untar(tarfile = tarfile, exdir = "monocle3/extract folder")

# run monocle3 
print('<==== trajectory ====>')
install.packages("optparse")

suppressMessages({
    library(optparse)
    library(Seurat)
    library(SeuratWrappers)
    library(monocle3)
    library(tidyverse)
    library(Matrix)
    library(ggplot2)
})

option_list = list(
    make_option(c("-e","--edr"),
                type = 'double',
                default = c(0.5, 0.6, 0.7, 0.8, 0.9, 1, 1.1, 1.2, 1.3, 1.4, 1.5),
                help = 'euclidean_distance_ratio',
                metavar = 'double'),
    make_option(c("-m","--mbl"),
                type = 'integer',
                default = c(5, 10, 15),
                help = 'minimal_branch_len',
                metavar = 'integer'),
    make_option(c("-g","--gdr"),
                type = 'double',
                default = c(0.1, 0.2, 0.3, 0.4, 0.5),
                help = 'geodesic_distance_ratio',
                metavar = 'double')
);

opt_parser = OptionParser(option_list = option_list);
opt = parse_args(opt_parser);

seurat_obj <- readRDS("yanshuo_monocle3/umap.T.obj.annotated.rds")

figure_path <- file.path("yanshuo_monocle3/result",
                         paste0("e_", opt$edr),
                         paste0("m_", opt$mbl),
                         paste0("g_", opt$gdr))

if (!dir.exists(figure_path)) {
  dir.create(figure_path, recursive = T)
}
setwd(figure_path)

print("Begin generate monocle obj")
monocle.cds <- as.cell_data_set(seurat_obj)
monocle.cds <- cluster_cells(cds = monocle.cds, reduction_method = "UMAP")
print("Begin learn graph")
monocle.cds <- learn_graph(
    monocle.cds,
    learn_graph_control = list(
        euclidean_distance_ratio = opt$edr,
        minimal_branch_len = opt$mbl,
        geodesic_distance_ratio = opt$gdr))

startCells <- Cells(seurat_obj)[seurat_obj$seurat_clusters == 10]
monocle.cds <- order_cells(monocle.cds, reduction_method = "UMAP", root_cells = startCells)

png(file.path(figure_path, paste0("e_", opt$edr, "_m_", opt$mbl, "_g_", opt$gdr, "_trajectory.png")))
plot_cells(cds = monocle.cds,
           color_cells_by = "pseudotime",
           show_trajectory_graph = T,
           trajectory_graph_color = "grey",
           trajectory_graph_segment_size = 0.5,
           graph_label_size = 2,
           cell_size = 1,
           label_cell_groups = F,
           label_groups_by_cluster = F,
           label_branch_points = F,
           label_roots = F,
           label_leaves = F)
dev.off()

``` 


#monocle3 for cd8 cluster! 
# Trajectory analysis: Perform Monocle3 ==== old script ======= not for publiciation

``` {r}
#load Seurat obj!
library(tidyverse)
library(Seurat)
library(monocle3)
library(SeuratWrappers)

#load the Seurat obj! 
seurat_obj_cd8<- readRDS("cd8_subclu.rds")

#Dimplot ! 
DimPlot(seurat_obj_cd8, reduction = "umap", label = TRUE) +
  theme_classic() +
  NoLegend() + 
  NoAxes()
``` 


# Filter only first and second arthritis ! 
``` {r}
seurat_obj_cd8<- subset(seurat_obj_cd8, subset = type %in% c("First arthritis", "Second arthritis"))  # for group selection 
``` 


``` {r}
suppressPackageStartupMessages({
  library(slingshot); library(SingleCellExperiment)
  library(RColorBrewer); library(scales)
  library(viridis); library(UpSetR)
  library(pheatmap); library(msigdbr)
  library(fgsea); library(knitr)
  library(ggplot2); library(gridExtra)
  library(tradeSeq)
})
``` 


## convert back to singleCellExperiment
``` {r}
# good tutorial: https://kstreet13.github.io/bioc2020trajectories/articles/workshopTrajectories.html 
seurat_obj_cd8 <- as.SingleCellExperiment(seurat_obj_cd8, assay = "SCT")

#Try slignthot !
sce <- slingshot(seurat_obj_cd8, 
                 reducedDim = 'UMAP',
                 start.clus = '1', 
                 end.clus= '2')


colors <- colorRampPalette(brewer.pal(11,'Spectral')[-6])(100)
plotcol <- colors[cut(sce$slingPseudotime_1, breaks=100)]

plot(reducedDims(sce)$UMAP, col = plotcol, pch=16, asp = 1)
lines(SlingshotDataSet(sce), lwd=2, col='black')
```

# Fit the GAM based on slighshot!
``` {r}
#pseudotime obj
# Extract pseudotime and assume uniform cell weights
pseudotime <- as.numeric(pseudotime(cds))
cellWeights <- matrix(1, ncol = 1, nrow = length(pseudotime))

# Create SingleCellExperiment object
sce <- SingleCellExperiment(assays = list(counts = counts(cds)))
colData(sce)$pseudotime <- pseudotime
colData(sce)$cellWeights <- cellWeights
sce <- fitGAM(counts = counts(sce),
                     pseudotime = pseudotime, 
                     cellWeights = cellWeights,
                     conditions= factor(colData(cds)$type))

#save TradeSeq between condition 
saveRDS(sce, file = "tradseq.condition.fvss.rds")  #this is important!! 
mean(rowData(sce)$tradeSeq$converged)
rowData(sce)$assocRes <- associationTest(sce, lineages = TRUE, l2fc = log2(2))

#TradeSeq !!!
library(tradeSeq)
assocRes <- rowData(sce)$assocRes
mockGenes <-  rownames(assocRes)[
  which(p.adjust(assocRes$`pvalue_lineage1_conditionFirst arthritis`, "fdr") <= 0.05)
]
tgfbGenes <-  rownames(assocRes)[
  which(p.adjust(assocRes$`pvalue_lineage1_conditionSecond arthritis`, "fdr") <= 0.05)
]

length(mockGenes)

UpSetR::upset(fromList(list(mock = mockGenes, Firstarthritis= FirstArthritisGenes))) # did not work!! 
``` 

``` {r}
### based on mean smoother
yhatSmooth <- predictSmooth(sce, gene = mockGenes, nPoints = 50, tidy = FALSE)
heatSmooth <- pheatmap(t(scale(t(yhatSmooth[, 1:50]))),
                       cluster_cols = FALSE,
                       show_rownames = FALSE, 
                       show_colnames = FALSE)
``` 

## the hierarchical trees constructed here, can also be used for 
## clustering of the genes according to their average expression pattern.
``` {r}
cl <- sort(cutree(heatSmooth$tree_row, k = 6))
table(cl)
``` 


conditions <- colData(sce)$pheno$treatment_id
pt1 <- colData(sce)$slingshot$pseudotime
  
### based on fitted values (plotting takes a while to run)
yhatCell <- predictCells(sce, gene=mockGenes)
yhatCellMock <- yhatCell[,conditions == "Mock"]
# order according to pseudotime
ooMock <- order(pt1[conditions == "Mock"], decreasing=FALSE)
yhatCellMock <- yhatCellMock[,ooMock]
pheatmap(t(scale(t(yhatCellMock))), cluster_cols = FALSE,
          show_rownames = FALSE, show_colnames=FALSE)



## C5 category is according to gene ontology grouping: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4707969/pdf/nihms-743907.pdf
``` {r}
geneSets <- msigdbr(species = "Homo sapiens", category = "C5", subcategory = "BP")
### filter background to only include genes that we assessed.
geneSets$gene_symbol <- toupper(geneSets$gene_symbol)
geneSets <- geneSets[geneSets$gene_symbol %in% names(sce),]
m_list <- geneSets %>% split(x = .$gene_symbol, f = .$gs_name)
stats <- assocRes$waldStat_lineage1_conditionMock
names(stats) <- rownames(assocRes)
eaRes <- fgsea(pathways = m_list, stats = stats, nperm = 5e4, minSize = 10)
``` 



``` {r}
# Seurat Wrap 
# 2. Convert to Seurat object to cell_data_set object ------
library(tidyverse)
library(monocle3)
library(SeuratWrappers)
cds <- as.cell_data_set(seurat_obj_cd8) # SeuratWrappers
DefaultAssay(seurat_obj_cd8)


# Learn the trajectory graph
cds <- cluster_cells(cds, reduction_method = "UMAP")  #fit with Seurat ! 

# trajectory analysis
cds <- learn_graph(cds, use_partition = F)

# Order the cells by pseudotime : https://ucdavis-bioinformatics-training.github.io/2021-August-Advanced-Topics-in-Single-Cell-RNA-Seq-Trajectory-and-Velocity/data_analysis/monocle_fixed
cds <- order_cells(cds)
startcell<- Cells(seurat_obj_cd8)[seurat_obj_cd8$seurat_clusters == 1]
cds<- order_cells(cds, 
                  reduction_method = "UMAP",
                  root_cells = startcell)

#plot_cells in pseudotime
library(tidyverse)
plot_cells(cds, color_cells_by = 'pseudotime', 
                       label_branch_points = FALSE, 
           label_leaves = FALSE) + 
  theme_classic() + 
  NoAxes()
``` 

``` {r}
ggsave("Feature.tiff", plot = featureplot, height=5, width = 6, units = "in", dpi = 300)
``` 

# Aboved code ===== Perform Monocle3 ==== old script ======= not for publiciation

# Run differentially expressed genes and annotation! 

# cluster 0
``` {r}
#compare gene expression of cluster 0 to the rest of the cluster
prepsct<- PrepSCTFindMarkers(seurat_obj_cd8)
cluster0<- FindMarkers(prepsct,assay = "SCT", 
                       ident.1 = 0, 
                       only.pos = TRUE, 
                       min.pct = 0.25, 
                       logfc.threshold = 0.25, 
                       verbose = FALSE)
head(cluster0, n = 20)
``` 

**order it from highest log2FC to lowest
``` {r}
cluster0<- cluster0 %>% arrange(desc(avg_log2FC))
head(cluster0)
``` 

# Save deg of cluster 0

``` {r}
write.csv(cluster0, file = "CD8 Subclus/deg_sct_cluster0.csv")
```

#cluster 1
``` {r}
#compare gene expression of cluster 0 to the rest of the cluster
cluster1<- FindMarkers(prepsct,
                       assay = "SCT", 
                       ident.1 = 1, 
                       only.pos = TRUE, 
                       min.pct = 0.25, 
                       logfc.threshold = 0.25, 
                       verbose = FALSE)
head(cluster1, n = 20)
```

**order it from highest log2FC to lowest
``` {r}
cluster1<- cluster1 %>% arrange(desc(avg_log2FC))
head(cluster1)
``` 
# Save deg of cluster 1

``` {r}
write.csv(cluster1, file = "CD8 Subclus/deg_sct_cluster1.csv")
```


#cluster 2
``` {r}
#compare gene expression of cluster 0 to the rest of the cluster
cluster2<- FindMarkers(prepsct,assay = "SCT", 
                       ident.1 = 2, 
                       only.pos = TRUE, 
                       min.pct = 0.25, 
                       logfc.threshold = 0.25, 
                       verbose = FALSE)
head(cluster2, n = 20)
```
**order it from highest log2FC to lowest
``` {r}
cluster2<- cluster2 %>% arrange(desc(avg_log2FC))
head(cluster2)
``` 

# Save deg of cluster 2

``` {r}
write.csv(cluster2, file = "CD8 Subclus/deg_sct_cluster2.csv")
```

#cluster 3
``` {r}
#compare gene expression of cluster 0 to the rest of the cluster
cluster3<- FindMarkers(prepsct,assay = "SCT", 
                       ident.1 = 3, 
                       only.pos = TRUE, 
                       min.pct = 0.25, 
                       logfc.threshold = 0.25, 
                       verbose = FALSE)
head(cluster3, n = 20)
```

**order it from highest log2FC to lowest
``` {r}
cluster3<- cluster3 %>% arrange(desc(avg_log2FC))
head(cluster3)
```

# Save deg of cluster 3

``` {r}
write.csv(cluster3, file = "CD8 Subclus/deg_sct_cluster3.csv")
```

#cluster 4
``` {r}
#compare gene expression of cluster 0 to the rest of the cluster
cluster4<- FindMarkers(prepsct,
                       assay = "SCT", 
                       ident.1 = 4, 
                       only.pos = TRUE, 
                       min.pct = 0.25, 
                       logfc.threshold = 0.25, 
                       verbose = FALSE)
head(cluster4, n = 20)
```
**order it from highest log2FC to lowest
``` {r}
cluster4<- cluster4 %>% arrange(desc(avg_log2FC))
head(cluster4)
```

# Save deg of cluster 4

``` {r}
write.csv(cluster4, file = "CD8 Subclus/deg_sct_cluster4.csv")
```


# monocle 3 for publication! ==================================================

``` {r}
# 2. Convert to Seurat object to cell_data_set object ------
library(monocle3)
library(SeuratWrappers)
cds <- as.cell_data_set(seurat_obj_cd8) # SeuratWrappers
DefaultAssay(seurat_obj_cd8)
``` 
# Learn the trajectory graph ! 
``` {r}
cds <- cluster_cells(cds, 
                     reduction_method = "UMAP", 
                     resolution = 1e-3)  #fit with Seurat !

#plot === optional ===!
plot_cells(cds, color_cells_by = 'ident', 
           label_groups_by_cluster = FALSE,
           group_label_size = 5) +
  theme(legend.position = "right") + 
  theme_bw()# already run and get the same graph! 
```


## trajectory analysis
``` {r}
cds <- learn_graph(cds)

# plot
plot_cells(cds, color_cells_by = 'ident', 
           label_groups_by_cluster = FALSE,
           group_label_size = 5) +
  theme_void() +
  theme(legend.position = "none") + 
  scale_color_manual(values = c("0"= "#FF3349", "1"= "#3633FF", 
                                "2"="#FF33F3", "3"="#33C4FF", "4"= "#FFAF33", 
                                 "5"= "#9933FF", "gd T"= "#FF33D1")) + 
  theme(legend.position = "right")
```
 

``` {r}
ggsave("cd.traj.pdf", plot = node, height=4.5, width = 7, units = "in", dpi = 300)
```


# Create gene annotation file
``` {r}
fData(cds) # gene_annotation
rownames(fData(cds))[1:10]
fData(cds)$gene_short_name <- rownames(fData(cds))
fData(cds)
``` 

# Order the cells by pseudotime : https://ucdavis-bioinformatics-training.github.io/2021-August-Advanced-Topics-in-Single-Cell-RNA-Seq-Trajectory-and-Velocity/data_analysis/monocle_fixed

# Annotation ! 
*0= Th17 like cell*
*1= CD8 effector*
*2= ILC*
*3= Treg*
*4= CD4 Tcm* 
*5= CD8 Tem* 
*6= PD-1hi CXCL13hi CD4+*
*7= Effector CD8* 
*8= Th1-like*
*9= Cytotoxic T*
*10= Recently activated CD4*
*11= Treg*
*12= NKG7+ CD8+ T cells*
*13= NKT*
*14= CD8 Trm*
*15= MAIT*
*16= Cycling T*
*17= KLRB1hi CD8 Tem*
*18= mixed CD4*

# one node=== CD8 effector! 
``` {r, message= FALSE}
https://support.bioconductor.org/p/9145187/
install.packages('htmltools')
cds <- order_cells(cds)

#plot_cells in pseudotime
trjectory<- plot_cells(cds , 
                       color_cells_by = 'pseudotime', 
           label_branch_points = FALSE, 
           label_leaves = FALSE, 
           graph_label_size = 4) + 
  theme_classic() + 
  NoAxes() + 
  theme(legend.position = "right")
trjectory

#save monocle3 object
saveRDS(cds, file = "cds.monocle3.cd8.all.rds")
cds<- readRDS("cds.monocle3.cd8.all.rds")
``` 


``` {r}
ggsave("trajectory.pdf", plot = trjectory, height=4.5, width = 7,  units = "in", dpi = 300)
```

``` {r}
head(pseudotime(cds), 10)
``` 

``` {r}
#resource == https://rpubs.com/mahima_bose/Seurat_and_Monocle3_p 
cds$monocle3_pseudotime <- pseudotime(cds)
data.pseudo <- as.data.frame(colData(cds))

#ggplot! 
ggplot(data.pseudo, aes(monocle3_pseudotime, seurat_clusters, fill = seurat_clusters)) + geom_boxplot()
``` 
# Order the pseudotime! 
``` {r}
library(tidyverse)
data.pseudo <- data.pseudo %>%
  mutate(seurat_clusters_reordered = reorder(seurat_clusters, monocle3_pseudotime))

# Plot using the reordered factor
ggplot(data.pseudo, aes(monocle3_pseudotime, 
                        seurat_clusters_reordered, 
                        fill = seurat_clusters)) + 
  geom_boxplot()
ggplot(data.pseudo, aes(monocle3_pseudotime, 
                        reorder(seurat_clusters, monocle3_pseudotime),
                        fill = seurat_clusters)) + 
  geom_boxplot()
``` 

``` {r}
library(ggplot2)
library(ggridges)

pseudotime<- ggplot(data.pseudo, aes(x = monocle3_pseudotime, 
                                     y = reorder(seurat_clusters, monocle3_pseudotime), 
                                     fill = seurat_clusters)) +
  theme_bw() +
  geom_density_ridges() + 
  theme_classic() + 
  theme(legend.position = "none", 
        axis.title.y  = element_blank(), 
        axis.text = element_text(size= 12, face = "bold"), 
        axis.title.x = element_text(siz= 16, face = "bold"), 
        panel.border = element_rect(colour = "black", fill = NA, size = 1.5)) + 
  labs(x= "Pseudotime")

# New plot! 
pseudotime<- ggplot(data.pseudo, aes(x = monocle3_pseudotime, y = reorder(seurat_clusters, monocle3_pseudotime), fill = seurat_clusters)) +
  geom_density_ridges() + 
  theme_void() + 
  theme(legend.position = "none", 
        axis.title.y  = element_blank(), 
        axis.text = element_text(size= 12, face = "bold"), 
        axis.title.x = element_text(siz= 16, face = "bold")) + 
  labs(x= "Pseudotime") 
pseudotime
``` 

# save plot1 
``` {r}
ggsave("cd8.pseudo.2.pdf", plot = pseudotime, height=4.5, width = 6.5, units = "in", dpi = 300)
```


# save plot1 
``` {r}
ggsave("cd8.pseudotime.pdf", plot = pseudotime, height=4, width = 3.5, units = "in", dpi = 300)
```


#differentially expressed genes along the trajectory! 
``` {r}
deg <- graph_test(cds, neighbor_graph = "principal_graph", cores = 4)

## Additional test ! 
deg <- graph_test(cds, 
                  neighbor_graph = "principal_graph", 
                  model_f,
                  cores = 4)
graph_test()

#save deg obj
saveRDS(deg, file = "deg.rds")
deg<- readRDS("deg.rds")
head(deg)

deg_short<- deg %>% filter(q_value < 0.05, status== "OK")
deg %>% arrange(q_value) %>% filter(status == "OK") %>% 
  head()
class(deg)

# remove col-I gene 
deg<- deg[deg$]
```

#write csv file!

``` {r}
write.csv(deg_short, file = "monocle.csv")
```

# deg as a function of pseudotime!  
``` {r}
gene_speudo<- row.names(subset(deg, q_value < 0.05))
plot_cells(cds, genes=c("TNFRSF18", "SH3BGRL3", "ZNF683", "CSF3R"),
           show_trajectory_graph=FALSE,
           label_cell_groups=FALSE,
           label_leaves=FALSE)

``` 
#gene modules
``` {r}
gene_module_df <- find_gene_modules(cds[gene_speudo,], resolution=1e-2)
cell_group_df <- tibble::tibble(cell=row.names(colData(neurons_cds)), 
                                cell_group=partitions(cds)[colnames(neurons_cds)])
agg_mat <- aggregate_gene_expression(neurons_cds, gene_module_df, cell_group_df)
row.names(agg_mat) <- stringr::str_c("Module ", row.names(agg_mat))
colnames(agg_mat) <- stringr::str_c("Partition ", colnames(agg_mat))

pheatmap::pheatmap(agg_mat, cluster_rows=TRUE, cluster_cols=TRUE,
                   scale="column", clustering_method="ward.D2",
                   fontsize=6)
```

#some feature plots! 
``` {r}
FeaturePlot(seurat_obj_cd8, features = c("TNFRSF18", "TNFRSF25", "ZNF683", "CD52"))
``` 


#add pseudotime values 
``` {r}
seurat_obj_cd8$pseudotime<- pseudotime(cds)
FeaturePlot(seurat_obj_cd8, features = "pseudotime")
``` 


``` {r}
RidgePlot(seurat_obj_cd8, features = c("TNFRSF18", "TNFRSF25", "ZNF683", "CD52"), sort = T, idents = c("0", "1", "2", "3", "4"))
``` 


``` {r}
my_genes <- row.names(subset(fData(cds), gene_short_name %in% c("LINC00115", "KLHL17", "ISG15", "TNFRSF18"))) 
cds_subset <- cds[my_genes,]
plot_genes_in_pseudotime(cds_subset, color_cells_by = "monocle3_pseudotime" )
``` 

``` {r}
library(ggplot2)

# Subset the genes of interest
my_genes <- row.names(subset(fData(cds), gene_short_name %in% c("LINC00115", "KLHL17", "ISG15", "TNFRSF18"))) 
cds_subset <- cds[my_genes,]

# Plot genes in pseudotime
p <- plot_genes_in_pseudotime(cds_subset, color_cells_by = "monocle3_pseudotime" )

# Add density plot
density_plot <- ggplot(data = data.frame(x = rnorm(1000)), aes(x = x)) +
  geom_density()

# Merge plots
p + density_plot +
  geom_text(data = data.frame(y = seq_along(my_genes), label = my_genes),
            aes(x = Inf, y = y, label = label), hjust = 1.1, size = 3)

```


``` {r}
my_genes <- row.names(subset(fData(cds),
          gene_short_name %in% c("LINC00115", "KLHL17", "ISG15", "TNFRSF18"))
cds_subset <- cds[my_genes,]
heatmap <- plot_pseudotime_heatmap(cds_subset, cluster_rows = FALSE, show_rownames = TRUE, return_heatmap = T)
heatmap
```

``` {r}
# https://github.com/cole-trapnell-lab/monocle-release/issues/295 

library(ComplexHeatmap)
library(ggplot2)
library(dplyr)
library(RColorBrewer)
library(circlize)
library(monocle3)

#modulated_genes <- graph_test(cds, neighbor_graph = "principal_graph", cores = 4)
genes <- row.names(subset(deg, q_value< 0.05 & morans_I > 0.20))
genes

pt.matrix <- exprs(cds)[match(genes,rownames(rowData(cds))),order(pseudotime(cds))]
#Can also use "normalized_counts" instead of "exprs" to use various normalization methods, for example:
#normalized_counts(cds, norm_method = "log")

pt.matrix <- t(apply(pt.matrix,1,function(x){smooth.spline(x,df=3)$y}))
pt.matrix <- t(apply(pt.matrix,1,function(x){(x-mean(x))/sd(x)}))
rownames(pt.matrix) <- genes;

#show some genes
num_selected_genes <- 20  # Adjust this number according to your preference

# Randomly select gene labels for visualization
set.seed(123)  # Set seed for reproducibility
selected_genes <- sample(genes, num_selected_genes)

#K means with 6 groups
htkm <- Heatmap(
  pt.matrix,
  name                         = "z-score",
  col                          = colorRamp2(seq(from=-2,to=2,length=11),rev(brewer.pal(11, "Spectral"))),
  show_row_names               = TRUE,
  show_column_names            = TRUE,
  row_names_gp                 = gpar(fontsize = 7),
  km = 4,
  row_title_rot                = 0,
  cluster_rows                 = TRUE,
  cluster_row_slices           = FALSE,
  cluster_columns              = FALSE)

#Ward.D2 Hierarchical Clustering
hthc <- Heatmap(
  pt.matrix,
  name                         = "z-score",
  col                          = colorRamp2(seq(from=-2,to=2,length=11),rev(brewer.pal(11, "Spectral"))),
  show_row_names               = TRUE,
  show_column_names            = TRUE,
  row_names_gp                 = gpar(fontsize = 6),
  clustering_method_rows = "ward.D2",
  clustering_method_columns = "ward.D2",
  row_title_rot                = 0,
  cluster_rows                 = TRUE,
  cluster_row_slices           = FALSE,
  cluster_columns              = FALSE)

print(htkm)
print(hthc)

#save Kmean obj
saveRDS(htkm, file = "htkm.rds")
```

# save quality figure

``` {r}
ggsave("deg.monocle.png", plot = htkm, height=4.5, width = 6, units = "in", dpi = 300)
```

#plot pathways of genes 
``` {r}
library(tidyverse)
monocle_path<- read.csv("CD8_trajec_pathway.csv", stringsAsFactors = TRUE)
head(monocle_path)
```

``` {r}
#filter top 10
monocle_path<- monocle_path[c(1:40),] 

#ggplot 
pathways<- ggplot(monocle_path, aes(reorder(Term,log_pdjust),log_pdjust)) + 
  theme_classic()+ geom_bar(stat = "identity", fill= "#E12695")  + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 13),
        axis.text.y = element_text(size= 10, face = "bold"),
        axis.title.y = element_blank(),
        axis.title.x = element_text(face = "bold", size=15 ),
        axis.text = element_text(face = "bold", size = 8), 
        plot.title = element_text(hjust = 0.5, face = "bold", size = 12)) + 
  labs(y="-log10(P value)", title = "Pathways") + 
  coord_flip()

pathways
``` 

# save pdf file! 
``` {r}
ggsave("path.monocle.pdf", plot = pathways, height=7, width = 5, units = "in", dpi = 300)
```

#deg as a function of pseudotime #did not work1 

``` {r}
diff_test_result <- differentialGeneTest(cds,
fullModelFormulaStr = "~sm.ns(Pseudotime, df=3)",
reducedModelFormulaStr = "~1", relative_expr = TRUE, cores = 4,
verbose = FALSE)

sig_gene_names <- row.names(subset(diff_test_result, qval < 0.00001))
nrow(subset(diff_test_result, qval < 0.00001))
write.csv(diff_test_result, file=paste("diff_test_result.csv",sep=""))

heatmap <- plot_pseudotime_heatmap(cds[sig_gene_names,],
num_clusters = 4,
cores = 6,
show_rownames = FALSE, return_heatmap = T)


``` 



```{r}
deg_qvalue <- row.names(subset(deg, q_value < 0.05)) 
head(deg_qvalue)

``` 
``` {r}
plot_cells(cds, genes=c("LINC00115", "KLHL17", "ISG15", "TNFRSF18"),
           show_trajectory_graph=FALSE,
           label_cell_groups=FALSE,
           label_leaves=FALSE)


``` 





#collect highly variable genes into module
``` {r}
gene_module_df <- find_gene_modules(cds[deg,], resolution=c(10^seq(-6,-1)))
``` 




``` {r}
AFD_genes <- c("LINC00115", "KLHL17", "ISG15", "TNFRSF18")
AFD_lineage_cds <- cds[rowData(cds)$gene_short_name %in% AFD_genes,
                       colData(cds)$cell.type %in% c("AFD")]
AFD_lineage_cds <- order_cells(AFD_lineage_cds)
```

# Analyzing branch! 
``` {r}
cds_subset <- choose_cells(cds)
subset_pr_test_res <- graph_test(cds_subset, neighbor_graph="principal_graph", cores=4)
pr_deg_ids <- row.names(subset(subset_pr_test_res, q_value < 0.05))

```


# Monocle 3 for recurrent arthritis ==== second arthritis =======================================================================

# Subset just second arthritis 

``` {r}
#load Seurat obj!
library(tidyverse)
library(Seurat)
library(monocle3)
library(SeuratWrappers)

#load the Seurat obj! 
seurat_obj_cd8<- readRDS("cd8_subclu.rds")

#Dimplot ! 
DimPlot(seurat_obj_cd8, reduction = "umap", label = TRUE) +
  theme_minimal() +
  NoLegend()
``` 
``` {r}
view(seurat_obj_cd8)
second_arthritis<- subset(seurat_obj_cd8, subset = type == "Second arthritis")
view(second_arthritis)
```

``` {r}
# 2. Convert to Seurat object to cell_data_set object ------
cds_second <- as.cell_data_set(second_arthritis) # SeuratWrappers
DefaultAssay(seurat_obj_cd8)
``` 
# Learn the trajectory graph ! 
``` {r}
cds_second <- cluster_cells(cds_second, 
                     reduction_method = "UMAP", 
                     resolution = 1e-2)  #fit with Seurat !

#plot === optional ===!
plot_cells(cds_second, color_cells_by = 'ident', 
           label_groups_by_cluster = FALSE,
           group_label_size = 5) +
  theme(legend.position = "right") + 
  theme_bw()# already run and get the same graph! 
```


## trajectory analysis
``` {r}
cds_second <- learn_graph(cds_second)

# plot
plot_cells(cds_second, color_cells_by = 'ident', 
           label_groups_by_cluster = FALSE,
           group_label_size = 5) +
  theme_void() +
  theme(legend.position = "none") + 
  scale_color_manual(values = c("0"= "#FF3349", "1"= "#3633FF", 
                                "2"="#FF33F3", "3"="#33C4FF", "4"= "#FFAF33", 
                                 "5"= "#9933FF", "gd T"= "#FF33D1")) + 
  theme(legend.position = "right")
```
 

``` {r}
ggsave("cd.traj.pdf", plot = node, height=4.5, width = 7, units = "in", dpi = 300)
```


# Create gene annotation file
``` {r}
fData(cds) # gene_annotation
rownames(fData(cds))[1:10]
fData(cds)$gene_short_name <- rownames(fData(cds))
fData(cds)
``` 

# Order the cells by pseudotime : https://ucdavis-bioinformatics-training.github.io/2021-August-Advanced-Topics-in-Single-Cell-RNA-Seq-Trajectory-and-Velocity/data_analysis/monocle_fixed

# Annotation ! 
*0= Th17 like cell*
*1= CD8 effector*
*2= ILC*
*3= Treg*
*4= CD4 Tcm* 
*5= CD8 Tem* 
*6= PD-1hi CXCL13hi CD4+*
*7= Effector CD8* 
*8= Th1-like*
*9= Cytotoxic T*
*10= Recently activated CD4*
*11= Treg*
*12= NKG7+ CD8+ T cells*
*13= NKT*
*14= CD8 Trm*
*15= MAIT*
*16= Cycling T*
*17= KLRB1hi CD8 Tem*
*18= mixed CD4*

# one node=== CD8 effector! 
``` {r, message= FALSE}
https://support.bioconductor.org/p/9145187/
install.packages('htmltools')
cds_second <- order_cells(cds_second)

#plot_cells in pseudotime
trjectory_second<- plot_cells(cds_second , 
                       color_cells_by = 'pseudotime', 
           label_branch_points = FALSE, 
           label_leaves = FALSE, 
           graph_label_size = 4) + 
  theme_classic() + 
  NoAxes() + 
  theme(legend.position = "right")
trjectory_second

#save monocle3 object
saveRDS(cds_second, file = "cds_second_arthritis.monocle3.cd8.rds")  # saved on 6/6/2024
cds_second<- readRDS("cds_second_arthritis.monocle3.cd8.rds")
``` 


``` {r}
ggsave("trajectory.pdf", plot = trjectory, height=4.5, width = 7,  units = "in", dpi = 300)
```

``` {r}
head(pseudotime(cds), 10)
``` 

``` {r}
#resource == https://rpubs.com/mahima_bose/Seurat_and_Monocle3_p 
cds_second$monocle3_pseudotime <- pseudotime(cds_second)
data.pseudo_second <- as.data.frame(colData(cds_second))

#ggplot! 
ggplot(data.pseudo_second, aes(monocle3_pseudotime, seurat_clusters, fill = seurat_clusters)) + geom_boxplot()
```
# Order the pseudotime! 
``` {r}
ggplot(data.pseudo_second, aes(monocle3_pseudotime, reorder(seurat_clusters, monocle3_pseudotime), fill = seurat_clusters)) + geom_boxplot()
``` 
#violen plot 
``` {r}
library(ggplot2)
library(ggridges)

pseudotime_second<- ggplot(data.pseudo_second, aes(x = monocle3_pseudotime, y = reorder(seurat_clusters, monocle3_pseudotime), fill = seurat_clusters)) +
  geom_density_ridges() + 
  theme_bw() + 
  theme(legend.position = "none", 
        axis.title.y  = element_blank(), 
        axis.text = element_text(size= 12, face = "bold"), 
        axis.title.x = element_text(siz= 16, face = "bold"), 
        panel.border = element_rect(colour = "black", fill = NA, size = 1.3), 
        panel.grid = element_blank()) + 
  labs(x= "Pseudotime")
pseudotime_second
``` 

# save plot1 
``` {r}
ggsave("cd8.pseudo.2.pdf", plot = pseudotime_second, height=4.5, width = 4, units = "in", dpi = 300)
```
# using yanshuo code !!
``` {r}
## Trying new monocle 3 ! Second arthritis 
second_arthritis<- subset(seurat_obj_cd8, subset = type == c("Second arthritis"))
monocle.cds.s <- as.cell_data_set(second_arthritis)
monocle.cds.s <- cluster_cells(cds = monocle.cds.s, reduction_method = "UMAP")
monocle.cds.s <- learn_graph(monocle.cds.s)

fData(monocle.cds.s) # gene_annotation
rownames(fData(monocle.cds.s))[1:10]
fData(monocle.cds.s)$gene_short_name <- rownames(fData(monocle.cds.s))
fData(monocle.cds.s)

startCells <- Cells(second_arthritis)[second_arthritis$seurat_clusters == 3]
monocle.cds.s <- order_cells(monocle.cds.s, 
                           reduction_method = "UMAP",
                           root_cells = startCells)
cd8_pseudo<- plot_cells(cds = monocle.cds.s,
           color_cells_by = "pseudotime",
           show_trajectory_graph = T,
           trajectory_graph_color = "red",
           trajectory_graph_segment_size = 1,
           graph_label_size = 2,
           cell_size = 0.8,
           label_cell_groups = F,
           label_groups_by_cluster = F,
           label_branch_points = F,
           label_roots = F,
           label_leaves = F) + 
  theme_bw() + 
theme(plot.title = element_text(hjust = 0.5, face = "bold"),
        axis.text = element_text(face = "bold", size= 12), 
        axis.title = element_text(face = "bold", size= 12), 
        panel.border = element_rect(colour = "black", fill = NA, size = 1.5)) + 
  theme(legend.position = "none")
cd8_pseudo

#plot the pseudotime !!
monocle.cds.s$monocle3_pseudotime <- pseudotime(monocle.cds.s)
data.pseudo_second <- as.data.frame(colData(monocle.cds.s))

#ggplot! 
ggplot(data.pseudo_second, aes(monocle3_pseudotime, 
                               seurat_clusters, 
                               fill = seurat_clusters)) + 
  geom_boxplot()

#order the cells !! 
iotox_pseudo_second= ggplot(data.pseudo_second, aes(monocle3_pseudotime, reorder(seurat_clusters, monocle3_pseudotime), fill = seurat_clusters)) + 
  geom_boxplot() + 
  theme_minimal() + 
  NoLegend() + 
  theme(plot.title = element_text(hjust = 0.5, face = "bold"),
        axis.text = element_text(face = "bold", size= 16), 
        axis.title = element_text(face = "bold", size= 18),
         panel.grid.major = element_line(size = 1), 
        panel.grid.minor = element_line(size= 1), 
        axis.title.y = element_blank(), 
        axis.text.y = element_blank()) + 
  theme(legend.position = "none")  +
  labs(x= "pseudotime")

iotox_pseudo_second

#Density plot !! 
library(ggplot2)
library(ggridges)

pseudotime_second<- ggplot(data.pseudo_second, aes(x = monocle3_pseudotime, y = reorder(seurat_clusters, monocle3_pseudotime), fill = seurat_clusters)) +
  geom_density_ridges() + 
  theme_minimal() + 
  theme(legend.position = "none", 
        axis.title.y  = element_blank(), 
        axis.text = element_text(size= 12, face = "bold"), 
        axis.title.x = element_text(siz= 16, face = "bold"), 
        panel.grid.major = element_line(size = 1), 
        panel.grid.minor = element_line(size= 1)) + 
  labs(x= "Pseudotime")
pseudotime_second
ggsave("cd8.pseudo.2.pdf", plot = pseudotime_second, height=3, width= 2.5,  units = "in", dpi = 300)
 panel.border = element_rect(colour = "black", fill = NA, size = 1.3)
``` 
  

#save monocle 3 plot !! 
``` {r}
ggsave("CD8secondtraje.pdf", plot = pseudotime_second, height=3, width = 3.5, units = "in", dpi = 300)
```


# 2024 IOTOX Conference !! 

``` {r}
ggsave("CD8.pseudo.tiff", plot = iotox_pseudo, height=4.5, width = 6, units = "in", dpi = 300)
``` 





#differentially expressed genes along the trajectory! 
``` {r}
deg_second <- graph_test(monocle.cds.s, neighbor_graph = "principal_graph", cores = 4)

#save deg obj
saveRDS(deg_second, file = "deg_second_arthritis_monocle3.rds")
deg_second<- readRDS("deg_second_arthritis_monocle3.rds")
head(deg_second)

deg_second_short<- deg_second %>% filter(q_value < 0.05, status== "OK")
deg_second %>% arrange(q_value) %>% filter(status == "OK") %>% 
  head()

```
# plot genes in pseudotime !!
``` {r}
my_genes<- row.names(subset(fData(monocle.cds.s), gene_short_name %in% c("TNF", "IFNG", "IL2",  "IL17A", "GZMB", "IL10", "TGFB1")))
cds_subset <- monocle.cds.s[my_genes,]
second_plotgenes<- plot_genes_in_pseudotime(cds_subset, 
                         min_expr = 0.5)
``` 
# save figures !!
``` {r}
ggsave("plotgenes_secondIA.pdf", plot = second_plotgenes, height=4.5, width = 3, units = "in", dpi = 300)
```


#write csv file!

``` {r}
write.csv(deg_second_short, file = "monocle3_second_arthritis.csv")
```

# using monocle3 with tradeseq to find differentially expressed genes
``` {r}
BiocManager::install(c("tradeSeq", "SingleCellExperiment", "slingshot"))
library(tradeSeq)
library(SingleCellExperiment)
library(slingshot)


#pseudotime obj
# Extract pseudotime and assume uniform cell weights
pseudotime <- as.numeric(pseudotime(monocle.cds.s))
cellWeights <- matrix(1, ncol = 1, nrow = length(pseudotime))

# Create SingleCellExperiment object
sce <- SingleCellExperiment(assays = list(counts = counts(monocle.cds.s)))
colData(sce)$pseudotime <- pseudotime
colData(sce)$cellWeights <- cellWeights

# Fit the GAM
sce_second <- fitGAM(counts = counts(sce), pseudotime = pseudotime, cellWeights = cellWeights) #
write(sce_second, file= "sce_second.rds")

# Perform differential expression testing as a function of pseudotime! 
diffExpr_second <- associationTest(sce_second)
diffExpr_omitna<- na.omit(diffExpr_second)
significant_genes_second <- rownames(diffExpr_omitna)[which(diffExpr_omitna$pvalue < 0.05)]

#adjust-pvalue ! 
library(data.table)
diffExpr_second_padj <- data.table(gene = rownames(diffExpr_omitna), diffExpr_omitna)
diffExpr_second_padj$padj_all <- p.adjust(diffExpr_second_padj$pvalue, method = "fdr")
diffExpr_second_padj<- na.omit(diffExpr_second_padj)

#filter significant genes 
second_deg_pseudo.adj<- diffExpr_second_padj %>% filter(padj_all < 0.05)
write.csv(second_deg_pseudo.adj, file = "second_deg_pseudo.padj.csv")  

# View significant genes
print(second_deg_pseudo.adj)
```

# write deg back in ! 
``` {r}
deg_second_new<- read.csv("second_deg_pseudo.padj.csv", stringsAsFactors = TRUE)
head(deg_second_new)

# short them by p-value 
# order it from highest log2FC to lowest
deg_second_new_order<- deg_second_new %>% arrange(desc(meanLogFC))
head(deg_second_new_order)

# Save deg of cluster 1
write.csv(deg_second_new_order, file = "deg.second_monocle3_tradeseq.csv") # used this to run gene ontology 
``` 

# read padj deg and order log_fold changes !
``` {r}
install.packages("xfun")
library(tidyverse)
#first arthritis deg of CD8 clusters! 
CD8_second_monocle_deg<- read.csv("second_deg_pseudo.padj.csv", 
                                 stringsAsFactors = TRUE)
CD8second_monocledeg<- CD8_second_monocle_deg %>% arrange(desc(meanLogFC))
head(CD8second_monocledeg)

#save obj ! 
install.packages("openxlsx")
library(openxlsx)

# save obj! 
write.xlsx(CD8second_monocledeg, file = "CD8_second_tradseq.xlsx")
``` 


# deg as a function of pseudotime!  
``` {r}
gene_speudo<- row.names(subset(deg, q_value < 0.05))
plot_cells(cds, genes=c("TNFRSF18", "SH3BGRL3", "ZNF683", "CSF3R"),
           show_trajectory_graph=FALSE,
           label_cell_groups=FALSE,
           label_leaves=FALSE)

``` 
#gene modules
``` {r}
gene_module_df <- find_gene_modules(cds[gene_speudo,], resolution=1e-2)
cell_group_df <- tibble::tibble(cell=row.names(colData(neurons_cds)), 
                                cell_group=partitions(cds)[colnames(neurons_cds)])
agg_mat <- aggregate_gene_expression(neurons_cds, gene_module_df, cell_group_df)
row.names(agg_mat) <- stringr::str_c("Module ", row.names(agg_mat))
colnames(agg_mat) <- stringr::str_c("Partition ", colnames(agg_mat))

pheatmap::pheatmap(agg_mat, cluster_rows=TRUE, cluster_cols=TRUE,
                   scale="column", clustering_method="ward.D2",
                   fontsize=6)
```


#some feature plots! 
``` {r}
FeaturePlot(seurat_obj_cd8, features = c("TNFRSF18", "TNFRSF25", "ZNF683", "CD52"))
``` 


#add pseudotime values 
``` {r}
seurat_obj_cd8$pseudotime<- pseudotime(cds)
FeaturePlot(seurat_obj_cd8, features = "pseudotime")
``` 


``` {r}
RidgePlot(seurat_obj_cd8, features = c("TNFRSF18", "TNFRSF25", "ZNF683", "CD52"), sort = T, idents = c("0", "1", "2", "3", "4"))
``` 



``` {r}
my_genes <- row.names(subset(fData(cds), gene_short_name %in% c("LINC00115", "KLHL17", "ISG15", "TNFRSF18"))) 
cds_subset <- cds[my_genes,]
plot_genes_in_pseudotime(cds_subset, color_cells_by = "monocle3_pseudotime" )
``` 

``` {r}
library(ggplot2)

# Subset the genes of interest
my_genes <- row.names(subset(fData(cds), gene_short_name %in% c("LINC00115", "KLHL17", "ISG15", "TNFRSF18"))) 
cds_subset <- cds[my_genes,]

# Plot genes in pseudotime
p <- plot_genes_in_pseudotime(cds_subset, color_cells_by = "monocle3_pseudotime" )

# Add density plot
density_plot <- ggplot(data = data.frame(x = rnorm(1000)), aes(x = x)) +
  geom_density()

# Merge plots
p + density_plot +
  geom_text(data = data.frame(y = seq_along(my_genes), label = my_genes),
            aes(x = Inf, y = y, label = label), hjust = 1.1, size = 3)

```


``` {r}
my_genes <- row.names(subset(fData(cds),
          gene_short_name %in% c("LINC00115", "KLHL17", "ISG15", "TNFRSF18"))
cds_subset <- cds[my_genes,]
heatmap <- plot_pseudotime_heatmap(cds_subset, cluster_rows = FALSE, show_rownames = TRUE, return_heatmap = T)
heatmap
``

``` {r}
# https://github.com/cole-trapnell-lab/monocle-release/issues/295 

library(ComplexHeatmap)
library(ggplot2)
library(dplyr)
library(RColorBrewer)
library(circlize)
library(monocle3)

#modulated_genes <- graph_test(cds, neighbor_graph = "principal_graph", cores = 4)
genes_second <- row.names(subset(deg_second, q_value< 0.05 & morans_I > 0.19))
genes_second

## Remove col1 from list

genes_second <- setdiff(genes_second, "COTL1")


pt.matrix <- exprs(monocle.cds.s)[match(genes_second,rownames(rowData(monocle.cds.s))),order(pseudotime(monocle.cds.s))]
#Can also use "normalized_counts" instead of "exprs" to use various normalization methods, for example:
#normalized_counts(cds, norm_method = "log")

pt.matrix <- t(apply(pt.matrix,1,function(x){smooth.spline(x,df=3)$y}))
pt.matrix <- t(apply(pt.matrix,1,function(x){(x-mean(x))/sd(x)}))
rownames(pt.matrix) <- genes_second;

## dont run this code! 
if (any(is.na(pt.matrix) | is.nan(pt.matrix) | is.infinite(pt.matrix))) {
  # Replace NA/NaN/Inf values with zero, mean, or another appropriate value
  pt.matrix[is.na(pt.matrix) | is.nan(pt.matrix) | is.infinite(pt.matrix)] <- 0
}

#K means with 6 groups
htkm <- Heatmap(
  pt.matrix,
  name                         = "z-score",
  col                          = colorRamp2(seq(from=-2,to=2,length=11),rev(brewer.pal(11, "Spectral"))),
  show_row_names               = TRUE,
  show_column_names            = TRUE,
  row_names_gp                 = gpar(fontsize = 6),
  km = 4,
  row_title_rot                = 0,
  cluster_rows                 = TRUE,
  cluster_row_slices           = FALSE,
  cluster_columns              = FALSE)

print(htkm)

#save Kmean obj
saveRDS(htkm, file = "second.arthritis_htkm_cluster.rds")
```

# save quality figure

``` {r}
ggsave("deg.monocle.png", plot = htkm, height=4.5, width = 6, units = "in", dpi = 300)
```

#plot pathways of genes 
``` {r}
library(tidyverse)
monocle_path<- read.csv("CD8_second_trajec.csv", stringsAsFactors = TRUE)
monocle_path<- read.csv("CD8_second_trajec_GO.csv", stringsAsFactors = TRUE)
monocle_path<- read.csv("CD8_second_trajec_GO_new.csv", stringsAsFactors = TRUE)
head(monocle_path)
```

``` {r}
#filter top 10
monocle_path<- monocle_path[c(1:20),] 

#ggplot 
pathways<- ggplot(monocle_path, aes(reorder(Term,log_pdjust),log_pdjust)) + 
  theme_classic()+ geom_bar(stat = "identity", fill= "#2b8cbe")  + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 11),
        axis.text.y = element_text(size= 13, face = "bold"),
        axis.title.y = element_blank(),
        axis.title.x = element_text(face = "bold", size=12.5 ),
        axis.text = element_text(face = "bold", size = 8), 
        plot.title = element_text(hjust = 0.5, face = "bold", size = 12)) + 
  ylim(c(0,4)) + 
  labs(y="-log10(padj)") + 
  coord_flip()

pathways
``` 
# save pdf file! 
``` {r}
ggsave("path.monocle.cd8second.pdf", plot = pathways, height=7, width = 5, units = "in", dpi = 300)
```


# First arthritis ===================================================================================

``` {r}
view(seurat_obj_cd8)
First_arthritis<- subset(seurat_obj_cd8, subset = type == "First arthritis")
view(First_arthritis)
```

``` {r}
# 2. Convert to Seurat object to cell_data_set object ------
cds_First <- as.cell_data_set(First_arthritis) # SeuratWrappers
``` 

# Learn the trajectory graph ! 
``` {r}
cds_First <- cluster_cells(cds_First, 
                     reduction_method = "UMAP", 
                     resolution = 0.0001)  #fit with Seurat !

#plot === optional ===!
plot_cells(cds_First, color_cells_by = 'ident', 
           label_groups_by_cluster = FALSE,
           group_label_size = 5) +
  theme(legend.position = "right") + 
  theme_bw()# already run and get the same graph! 
```

## trajectory analysis
``` {r}
cds_First <- learn_graph(cds_First)

# plot
plot_cells(cds_First, color_cells_by = 'ident', 
           label_groups_by_cluster = FALSE,
           group_label_size = 5) +
  theme_void() +
  theme(legend.position = "none") + 
  scale_color_manual(values = c("0"= "#FF3349", "1"= "#3633FF", 
                                "2"="#FF33F3", "3"="#33C4FF", "4"= "#FFAF33", 
                                 "5"= "#9933FF", "gd T"= "#FF33D1")) + 
  theme(legend.position = "right")
```

# one node=== CD8 effector! 
``` {r, message= FALSE}
https://support.bioconductor.org/p/9145187/
install.packages('htmltools')
cds_First <- order_cells(cds_First)

#plot_cells in pseudotime
trjectory_First<- plot_cells(cds_First , 
                       color_cells_by = 'pseudotime', 
           label_branch_points = FALSE, 
           label_leaves = FALSE, 
           graph_label_size = 4) + 
  theme_classic() + 
  NoAxes() + 
  theme(legend.position = "right")
trjectory_First

#save monocle3 object
saveRDS(cds_First, file = "cds_First_arthritis.monocle3.rds")
cds_First<- readRDS("cds_First_arthritis.monocle3.rds")
``` 

``` {r}
#resource == https://rpubs.com/mahima_bose/Seurat_and_Monocle3_p 
cds_First$monocle3_pseudotime <- pseudotime(cds_First)
data.pseudo_First <- as.data.frame(colData(cds_First))

#ggplot! 
ggplot(data.pseudo_First, aes(monocle3_pseudotime, seurat_clusters, fill = seurat_clusters)) + geom_boxplot()
``` 
# Order the pseudotime! 
``` {r}
ggplot(data.pseudo_First, aes(monocle3_pseudotime, reorder(seurat_clusters, monocle3_pseudotime), fill = seurat_clusters)) + geom_boxplot()

``` 




``` {r}
library(ggplot2)
library(ggridges)

pseudotime_First<- ggplot(data.pseudo_First, aes(x = monocle3_pseudotime, y = reorder(seurat_clusters, monocle3_pseudotime), fill = seurat_clusters)) +
  geom_density_ridges() + 
  theme_classic() + 
  theme(legend.position = "none", 
        axis.title.y  = element_blank(), 
        axis.text = element_text(size= 9, face = "bold"), 
        axis.title.x = element_text(siz= 10, face = "bold")) + 
  labs(x= "Pseudotime")
pseudotime_First
``` 
# first arthritis: Yanshuo code !!! 

``` {r}
# First arthritis !!! 
first_arthritis<- subset(seurat_obj_cd8, subset = type == "First arthritis")

monocle.cds.f <- as.cell_data_set(first_arthritis)

monocle.cds.f <- cluster_cells(cds = monocle.cds.f, 
                             reduction_method = "UMAP")

monocle.cds.f <- learn_graph(monocle.cds.f)

startCells <- Cells(first_arthritis)[first_arthritis$seurat_clusters == 3]

monocle.cds.f <- order_cells(monocle.cds.f,
                           reduction_method = "UMAP", 
                           root_cells = startCells)
plot_cells(cds = monocle.cds.f,
           color_cells_by = "pseudotime",
           show_trajectory_graph = T,
           trajectory_graph_color = "grey",
           trajectory_graph_segment_size = 0.5,
           graph_label_size = 2,
           cell_size = 1,
           label_cell_groups = F,
           label_groups_by_cluster = F,
           label_branch_points = F,
           label_roots = F,
           label_leaves = F)


#plot cells !!! 
#plot the pseudotime !!
monocle.cds.f$monocle3_pseudotime <- pseudotime(monocle.cds.f)
data.pseudo_first <- as.data.frame(colData(monocle.cds.f))

#ggplot! 
ggplot(data.pseudo_first, aes(monocle3_pseudotime, seurat_clusters, fill = seurat_clusters)) + geom_boxplot()

#order the cells !! 
iotox_pseudo_first<- ggplot(data.pseudo_first, aes(monocle3_pseudotime, reorder(seurat_clusters, monocle3_pseudotime), fill = seurat_clusters)) + geom_boxplot() + 
  theme_minimal() +
  NoLegend() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold"),
        axis.text = element_text(face = "bold", size= 16), 
        axis.title = element_text(face = "bold", size= 18),
         panel.grid.major = element_line(size = 1), 
        panel.grid.minor = element_line(size= 1), 
        axis.title.y = element_blank(), 
        axis.text.y = element_blank()) + 
  labs(x= "pseudotime")
iotox_pseudo_first

ggsave("cd8.first.speudo.pdf", plot = pseudotime_first, height=3, width = 3.5, units = "in", dpi = 300)

#comined 
library(patchwork)
pseudo_combine<- iotox_pseudo_first + iotox_pseudo_second
pseudo_combine

#save high quality plots !! 
ggsave("cd8.pseudo.comine.tiff", plot = pseudo_combine, height=4.5, width = 3.8, units = "in", dpi = 300)

#Density plot !! 
library(ggplot2)
library(ggridges)

pseudotime_first<- ggplot(data.pseudo_first, aes(x = monocle3_pseudotime, y = reorder(seurat_clusters, monocle3_pseudotime), fill = seurat_clusters)) +
  geom_density_ridges() + 
  theme_minimal() + 
  theme(legend.position = "none", 
        axis.title.y  = element_blank(), 
        axis.text = element_text(size= 12, face = "bold"), 
        axis.title.x = element_text(siz= 16, face = "bold"),
        panel.grid.major = element_line(size = 1), 
        panel.grid.minor = element_line(size= 1)) + 
  labs(x= "Pseudotime")
pseudotime_first
ggsave("cd8.pseudo.1.pdf", plot = pseudotime_first, height=3, width= 2.5,  units = "in", dpi = 300)
 panel.border = element_rect(colour = "black", fill = NA, size = 1.3)
``` 

#differentially expressed genes along the trajectory! 
``` {r}
deg_First <- graph_test(monocle.cds.f, neighbor_graph = "principal_graph", cores = 4)

#save deg obj
saveRDS(deg_First, file = "deg_First_arthritis_monocle3.rds")
deg_First<- readRDS("deg_First_arthritis_monocle3.rds")
head(deg_First)

deg_First_short<- deg_First %>% filter(q_value < 0.05, status== "OK")
deg_First %>% arrange(q_value) %>% filter(status == "OK") %>% 
  head()
```

#write csv file!

``` {r}
write.csv(deg_First_short, file = "monocle_first_arthritis.csv")
```


# using monocle3 with tradeseq to find differentially expressed genes
``` {r}
BiocManager::install(c("tradeSeq", "SingleCellExperiment", "slingshot"))
BiocManager::install(c("BiocParallel", "Matrix"))
library(BiocParallel)
library(tradeSeq)
library(SingleCellExperiment)
library(slingshot)
library(Seurat)
library(monocle3)
library(SeuratWrappers)

# Extract pseudotime and assume uniform cell weights
pseudotime <- as.numeric(pseudotime(monocle.cds.f))
cellWeights <- matrix(1, ncol = 1, nrow = length(pseudotime))


# Convert counts to sparse matrix for memory efficiency
counts_sparse <- as(counts(monocle.cds.f), "dgCMatrix")

# Create SingleCellExperiment object
sce <- SingleCellExperiment(assays = list(counts = counts_sparse))
colData(sce)$pseudotime <- pseudotime
colData(sce)$cellWeights <- cellWeights

# Set up parallel processing
BPPARAM <- SnowParam(workers = 4)  # Adjust based on your system's capabilities

# Fit the GAM
sce <- fitGAM(counts = counts(sce), pseudotime = pseudotime, cellWeights = cellWeights, BPPARAM = BPPARAM)

# Perform differential expression testing as a function of pseudotime! 
diffExpr <- associationTest(sce)
diffExpr_omitna<- na.omit(diffExpr)
significant_genes <- rownames(diffExpr)[which(diffExpr$pvalue < 0.05)]

#adjust-pvalue ! 
library(data.table)
diffExpr_padj <- data.table(gene = rownames(diffExpr), diffExpr)
diffExpr_padj$padj_all <- p.adjust(diffExpr_padj$pvalue, method = "fdr")
diffExpr_padj<- na.omit(diffExpr_padj)

#filter significant genes 
first_deg_pseudo.adj<- diffExpr_padj %>% filter(padj_all < 0.05)
write.csv(first_deg_pseudo.adj, file = "first_deg_pseudo.padj.csv")

# View significant genes
print(significant_genes)

#read it back in 

deg_first_new<- read.csv("first_deg_pseudo.padj.csv", stringsAsFactors = TRUE)
head(deg_first_new)

# short them by p-value 
# order it from highest log2FC to lowest
deg_first_new_order<- deg_first_new %>% arrange(desc(meanLogFC))
head(deg_first_new_order)

# Save deg of cluster 1
write.csv(deg_first_new_order, file = "deg.first_monocle3_tradeseq.csv")
``` 

# read padj deg and order log_fold changes !
``` {r}
install.packages("xfun")
library(tidyverse)
#first arthritis deg of CD8 clusters! 
CD8_first_monocle_deg<- read.csv("first_deg_pseudo.padj.csv", 
                                 stringsAsFactors = TRUE)
CD8first_monocledeg<- CD8_first_monocle_deg %>% arrange(desc(meanLogFC))
head(CD8first_monocledeg)

#save obj ! 
install.packages("openxlsx")
library(openxlsx)

# save obj! 
write.xlsx(CD8first_monocledeg, file = "CD8tradseq.xlsx")
``` 


``` {r}
# https://github.com/cole-trapnell-lab/monocle-release/issues/295 

library(ComplexHeatmap)
library(ggplot2)
library(dplyr)
library(RColorBrewer)
library(circlize)
library(monocle3)

#modulated_genes <- graph_test(cds, neighbor_graph = "principal_graph", cores = 4)
genes_First <- row.names(subset(deg_First, q_value< 0.05 & morans_I > 0.21))
genes_First

## Remove col1 from list

genes_First <- setdiff(genes_First, "COTL1")
pt.matrix.first <- exprs(monocle.cds.f)[match(genes_First,rownames(rowData(monocle.cds.f))),order(pseudotime(monocle.cds.f))]
#Can also use "normalized_counts" instead of "exprs" to use various normalization methods, for example:
#normalized_counts(cds, norm_method = "log")

pt.matrix.first <- t(apply(pt.matrix.first,1,function(x){smooth.spline(x,df=3)$y}))
pt.matrix.first <- t(apply(pt.matrix.first,1,function(x){(x-mean(x))/sd(x)}))
rownames(pt.matrix.first) <- genes_First;


#K means with 6 groups
htkm.first <- Heatmap(
  pt.matrix.first,
  name                         = "z-score",
  col                          = colorRamp2(seq(from=-2,to=2,length=11),rev(brewer.pal(11, "Spectral"))),
  show_row_names               = TRUE,
  show_column_names            = TRUE,
  row_names_gp                 = gpar(fontsize = 6),
  km = 4,
  row_title_rot                = 0,
  cluster_rows                 = TRUE,
  cluster_row_slices           = FALSE,
  cluster_columns              = FALSE, 
  heatmap_legend_param = list(legend_direction= "horizontal"))

print(htkm.first)

#save Kmean obj
saveRDS(htkm.first, file = "htkm.first_arthritis.rds")
```

#plot pathways of genes 
``` {r}
library(tidyverse)
monocle_path<- read.csv("CD8_first_trajec.csv", stringsAsFactors = TRUE)
monocle_path<- read.csv("CD8_first_trajec_new.csv", stringsAsFactors = TRUE)
head(monocle_path)
```

``` {r}
#filter top 10
monocle_path<- monocle_path[c(1:15),] 

#ggplot 
pathways<- ggplot(monocle_path, aes(reorder(Term,log_pdjust),log_pdjust)) + 
  theme_classic()+ geom_bar(stat = "identity", fill= "#2b8cbe")  + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 13),
        axis.text.y = element_text(size= 13, face = "bold"),
        axis.title.y = element_blank(),
        axis.title.x = element_text(face = "bold", size=12),
        axis.text = element_text(face = "bold", size = 8), 
        plot.title = element_text(hjust = 0.5, face = "bold", size = 12)) + 
  labs(y="-log10(padj)", title = "Pathways") + 
  coord_flip()

pathways
``` 
# save pdf file! 
``` {r}
ggsave("pathways.firstarthritis_cd8.pdf", plot = pathways, height=7, width = 5.3, units = "in", dpi = 300)
```

#plot TradeSeq output ! 
``` {r}
go_second_CD8_arthritis<- read.csv("GO_second_arthritis_newp.csv", stringsAsFactors = TRUE)
head(go_second_CD8_arthritis)
go_second_CD8_arthritis$group<- factor(go_second_CD8_arthritis$group, levels = c("First arthritis", "Second arthritis"))

#ggplot----- !! 
GO_CD8_new<- ggplot(go_second_CD8_arthritis, aes(x = reorder(go_terms, gene_ratio), y = gene_ratio)) +
  geom_point(aes(size = gene_count, color = padj)) +
  scale_color_gradient(low = "red", high = "blue") +
  theme_bw() +
  theme(
    axis.text.x = element_text(hjust = 0.5, vjust = 1, face = "bold", size = 15), 
    axis.text.y = element_text(face = "bold", size = 11.6), 
    axis.title.y = element_blank(), 
    axis.title.x = element_text(face = "bold", size = 15), 
    legend.title = element_text(face = "bold"), 
    legend.text = element_text(face = "bold"), 
    panel.grid = element_blank(), 
    panel.border = element_rect(colour = "black", fill = NA, size = 1.5), 
    strip.text = element_text(face= "bold", size=12),
    legend.position = c(0.90, 0.35)) + 
  labs(x = "GO Terms", y = "Gene Ratio", color = "padj", size = "Count") + 
  coord_flip() +
  facet_wrap(~ group)
GO_CD8_new

ggsave("GO_CD8__arthritis.pdf", plot = GO_CD8_new, height=6, width = 9, units = "in", dpi = 300)

``` 


=================================================================================================
# monocle 3 for CD4 ==========================================================================

# orgi.cluster are 
- Th17 like CD4 
- Activated CD4 
- CD4 Tcm
- PD1hi CXCL13hi CD4
- Th1 like 
- MAIT 

``` {r}
cd4.monocle<- readRDS("CD4.obj.rds")
cd4_subclu<- DimPlot(cd4.monocle, 
                    reduction= "umap", 
                    label = FALSE, 
                    sizes.highlight = 20, 
                    pt.size = 1.5) +
  theme_minimal() +
  NoLegend() + 
  theme(plot.title = element_text(hjust = 0.5, face = "bold"),
        axis.text = element_text(face = "bold", size= 16), 
        axis.title = element_text(face = "bold", size= 18),
         panel.grid.major = element_line(size = 1), 
        panel.grid.minor = element_line(size= 1))
cd4_subclu
```  

#save high quality plot 
``` {r}
ggsave("CD4.new.tiff", plot = cd4_subclu, height=4.5, width = 6, units = "in", dpi = 300)

``` 


#cluster 0
``` {r}
#compare gene expression of cluster 0 to the rest of the cluster
prepsct<- PrepSCTFindMarkers(cd4.monocle)
cluster0<- FindMarkers(prepsct,assay = "SCT", 
                       ident.1 = 0, 
                       only.pos = TRUE, 
                       min.pct = 0.25, 
                       logfc.threshold = 0.25, 
                       verbose = FALSE)
head(cluster0, n = 20)
``` 

**order it from highest log2FC to lowest
``` {r}
cluster0<- cluster0 %>% arrange(desc(avg_log2FC))
head(cluster0)
``` 

# Save deg of cluster 0

``` {r}
write.csv(cluster0, file = "CD4 Subclus/deg_sct_cluster0.csv")
```

#cluster 1
``` {r}
#compare gene expression of cluster 0 to the rest of the cluster
cluster1<- FindMarkers(prepsct,
                       assay = "SCT", 
                       ident.1 = 1, 
                       only.pos = TRUE, 
                       min.pct = 0.25, 
                       logfc.threshold = 0.25, 
                       verbose = FALSE)
head(cluster1, n = 20)
```

**order it from highest log2FC to lowest
``` {r}
cluster1<- cluster1 %>% arrange(desc(avg_log2FC))
head(cluster1)
``` 
# Save deg of cluster 1

``` {r}
write.csv(cluster1, file = "CD4 Subclus/deg_sct_cluster1.csv")
```


#cluster 2
``` {r}
#compare gene expression of cluster 0 to the rest of the cluster
cluster2<- FindMarkers(prepsct,assay = "SCT", 
                       ident.1 = 2, 
                       only.pos = TRUE, 
                       min.pct = 0.25, 
                       logfc.threshold = 0.25, 
                       verbose = FALSE)
head(cluster2, n = 20)
```

**order it from highest log2FC to lowest
``` {r}
cluster2<- cluster2 %>% arrange(desc(avg_log2FC))
head(cluster2)
``` 

# Save deg of cluster 2

``` {r}
write.csv(cluster2, file = "CD4 Subclus/deg_sct_cluster2.csv")
```

#cluster 3
``` {r}
#compare gene expression of cluster 0 to the rest of the cluster
cluster3<- FindMarkers(prepsct,assay = "SCT", 
                       ident.1 = 3, 
                       only.pos = TRUE, 
                       min.pct = 0.25, 
                       logfc.threshold = 0.25, 
                       verbose = FALSE)
head(cluster3, n = 20)
```

**order it from highest log2FC to lowest
``` {r}
cluster3<- cluster3 %>% arrange(desc(avg_log2FC))
head(cluster3)
```

# Save deg of cluster 3

``` {r}
write.csv(cluster3, file = "CD4 Subclus/deg_sct_cluster3.csv")
```

#cluster 4
``` {r}
#compare gene expression of cluster 0 to the rest of the cluster
cluster4<- FindMarkers(prepsct,
                       assay = "SCT", 
                       ident.1 = 4, 
                       only.pos = TRUE, 
                       min.pct = 0.25, 
                       logfc.threshold = 0.25, 
                       verbose = FALSE)
head(cluster4, n = 20)
```
**order it from highest log2FC to lowest
``` {r}
cluster4<- cluster4 %>% arrange(desc(avg_log2FC))
head(cluster4)
```
# Save deg of cluster 4

``` {r}
write.csv(cluster4, file = "CD4 Subclus/deg_sct_cluster4.csv")
```

#cluster 5
``` {r}
#compare gene expression of cluster 0 to the rest of the cluster
cluster5<- FindMarkers(prepsct,
                       assay = "SCT", 
                       ident.1 = 5, 
                       only.pos = TRUE, 
                       min.pct = 0.25, 
                       logfc.threshold = 0.25, 
                       verbose = FALSE)
head(cluster5, n = 20)
```

**order it from highest log2FC to lowest
``` {r}
cluster5<- cluster5 %>% arrange(desc(avg_log2FC))
head(cluster5)
```

# Save deg of cluster 5

``` {r}
write.csv(cluster5, file = "CD4 Subclus/deg_sct_cluster5.csv")
```


# Annotation 

``` {r}
DimPlot(cd4.monocle, reduction= "umap", label = TRUE, sizes.highlight = 10) +
  theme_void() +
  NoLegend() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold"))
```  


# Seurat Wrap 

``` {r}
# 2. Convert to Seurat object to cell_data_set object ------
library(monocle3)
library(SeuratWrappers)
#select only first and second IA
cd4.monocle<- subset(cd4.monocle, subset = type %in% c("First arthritis", "Second arthritis"))  # for group selection
cds <- as.cell_data_set(cd4.monocle) # SeuratWrappers
DefaultAssay(cd4.monocle)
``` 


# Learn the trajectory graph ! 
``` {r}
cds <- cluster_cells(cds, 
                     reduction_method = "UMAP", 
                     resolution = 1e-3)  #fit with Seurat !

#plot === optional ===!
plot_cells(cds, color_cells_by = 'ident', 
           label_groups_by_cluster = FALSE,
           group_label_size = 5) +
  theme(legend.position = "right") + 
  theme_bw()# already run and get the same graph! 
```

## trajectory analysis
``` {r}
cds <- learn_graph(cds)

# plot
plot_cells(cds, color_cells_by = 'ident', 
           label_groups_by_cluster = FALSE,
           group_label_size = 5) +
  theme_void() +
  theme(legend.position = "none") + 
  scale_color_manual(values = c("0"= "#FF3349", "1"= "#3633FF", 
                                "2"="#FF33F3", "3"="#33C4FF", "4"= "#FFAF33", 
                                 "5"= "#9933FF", "gd T"= "#FF33D1")) + 
  theme(legend.position = "right")
```

``` {r}
ggsave("cd.traj.pdf", plot = node, height=4.5, width = 7, units = "in", dpi = 300)
```

# Order the cells by pseudotime : https://ucdavis-bioinformatics-training.github.io/2021-August-Advanced-Topics-in-Single-Cell-RNA-Seq-Trajectory-and-Velocity/data_analysis/monocle_fixed

# one node! 
``` {r, message= FALSE}
https://support.bioconductor.org/p/9145187/
install.packages('htmltools')
cds <- order_cells(cds)

#plot_cells in pseudotime
trjectory<- plot_cells(cds, color_cells_by = 'pseudotime', 
           label_branch_points = TRUE, 
           label_leaves = TRUE, 
           graph_label_size = 4) + 
  theme_classic() + 
  NoAxes() + 
  theme(legend.position = "right")
trjectory
``` 


``` {r}
ggsave("trajectory.pdf", plot = trjectory, height=4.5, width = 7,  units = "in", dpi = 300)
```


``` {r}
head(pseudotime(cds), 10)
``` 


``` {r}
#resource == https://rpubs.com/mahima_bose/Seurat_and_Monocle3_p 
cds$monocle3_pseudotime <- pseudotime(cds)
data.pseudo <- as.data.frame(colData(cds))

ggplot(data.pseudo, aes(monocle3_pseudotime, seurat_clusters, fill = seurat_clusters)) + 
  geom_boxplot()
``` 
``` {r}
ggplot(data.pseudo, aes(monocle3_pseudotime, reorder(seurat_clusters, monocle3_pseudotime), fill = seurat_clusters)) + geom_boxplot()
``` 

``` {r}
library(ggplot2)
library(ggridges)

pseudotime<- ggplot(data.pseudo, aes(x = monocle3_pseudotime, 
                                     y = reorder(seurat_clusters, monocle3_pseudotime), fill = seurat_clusters)) +
  geom_density_ridges() + 
  theme_bw() + 
  theme(legend.position = "none", 
        axis.title.y  = element_blank(), 
        axis.text = element_text(size= 12, face = "bold"), 
        axis.title.x = element_text(siz= 14, face = "bold"), 
        panel.border = element_rect(colour = "black", fill = NA, size = 1.5), 
        panel.grid = element_blank()) + 
  labs(x= "Pseudotime")
pseudotime
``` 

# save plot1 
``` {r}
ggsave("cd4.pseudo.pdf", plot = pseudotime, height=2.7, width = 5, units = "in", dpi = 300)
```

#differentially expressed genes along the trajectory! 

``` {r}
deg <- graph_test(cds, neighbor_graph = "principal_graph", cores = 4)
deg_short<- deg %>% filter(q_value < 0.05, status== "OK")
deg %>% arrange(q_value) %>% filter(status == "OK") %>% 
  head()
```


#write csv file!

``` {r}
write.csv(deg_short, file = "monocle.cd4.csv")
```

# deg as a function of pseudotime!  
``` {r}
gene_speudo<- row.names(subset(deg, q_value < 0.05))
plot_cells(cds, genes=c("SMARCC1", "MAP4", "TMA7", "ATRIP"),
           show_trajectory_graph=FALSE,
           label_cell_groups=FALSE,
           label_leaves=FALSE)

``` 
#gene modules
``` {r}
gene_module_df <- find_gene_modules(cds[gene_speudo,], resolution=1e-2)
cell_group_df <- tibble::tibble(cell=row.names(colData(neurons_cds)), 
                                cell_group=partitions(cds)[colnames(neurons_cds)])
agg_mat <- aggregate_gene_expression(neurons_cds, gene_module_df, cell_group_df)
row.names(agg_mat) <- stringr::str_c("Module ", row.names(agg_mat))
colnames(agg_mat) <- stringr::str_c("Partition ", colnames(agg_mat))

pheatmap::pheatmap(agg_mat, cluster_rows=TRUE, cluster_cols=TRUE,
                   scale="column", clustering_method="ward.D2",
                   fontsize=6)
```




#some feature plots! 
``` {r}
FeaturePlot(seurat_obj_cd8, features = c("TNFRSF18", "TNFRSF25", "ZNF683", "CD52"))
``` 


#add pseudotime values 
``` {r}
seurat_obj_cd8$pseudotime<- pseudotime(cds)
FeaturePlot(seurat_obj_cd8, features = "pseudotime")
``` 
``` {r}
RidgePlot(seurat_obj_cd8, features = c("TNFRSF18", "TNFRSF25", "ZNF683", "CD52"), sort = T, idents = c("0", "1", "2", "3", "4"))
``` 



``` {r}
my_genes <- row.names(subset(fData(cds), gene_short_name %in% c("LINC00115", "KLHL17", "ISG15", "TNFRSF18"))) 
cds_subset <- cds[my_genes,]
plot_genes_in_pseudotime(cds_subset, color_cells_by = "monocle3_pseudotime" )
``` 

``` {r}
library(ggplot2)

# Subset the genes of interest
my_genes <- row.names(subset(fData(cds), gene_short_name %in% c("LINC00115", "KLHL17", "ISG15", "TNFRSF18"))) 
cds_subset <- cds[my_genes,]

# Plot genes in pseudotime
p <- plot_genes_in_pseudotime(cds_subset, color_cells_by = "monocle3_pseudotime" )

# Add density plot
density_plot <- ggplot(data = data.frame(x = rnorm(1000)), aes(x = x)) +
  geom_density()

# Merge plots
p + density_plot +
  geom_text(data = data.frame(y = seq_along(my_genes), label = my_genes),
            aes(x = Inf, y = y, label = label), hjust = 1.1, size = 3)

```


``` {r}
my_genes <- row.names(subset(fData(cds),
          gene_short_name %in% c("LINC00115", "KLHL17", "ISG15", "TNFRSF18"))
cds_subset <- cds[my_genes,]
heatmap <- plot_pseudotime_heatmap(cds_subset, cluster_rows = FALSE, show_rownames = TRUE, return_heatmap = T)
heatmap
``

``` {r}
# https://github.com/cole-trapnell-lab/monocle-release/issues/295 

library(ComplexHeatmap)
library(ggplot2)
library(dplyr)
library(RColorBrewer)
library(circlize)
library(monocle3)

#modulated_genes <- graph_test(cds, neighbor_graph = "principal_graph", cores = 4)
genes <- row.names(subset(deg, q_value == 0 & morans_I > 0.22))
genes

pt.matrix <- exprs(cds)[match(genes,rownames(rowData(cds))),order(pseudotime(cds))]
#Can also use "normalized_counts" instead of "exprs" to use various normalization methods, for example:
#normalized_counts(cds, norm_method = "log")

pt.matrix <- t(apply(pt.matrix,1,function(x){smooth.spline(x,df=3)$y}))
pt.matrix <- t(apply(pt.matrix,1,function(x){(x-mean(x))/sd(x)}))
rownames(pt.matrix) <- genes;

#K means with 6 groups
library(ComplexHeatmap)
htkm <- Heatmap(
  pt.matrix,
  name                         = "z-score",
  col                          = colorRamp2(seq(from=-2,to=2,length=11),rev(brewer.pal(11, "Spectral"))),
  show_row_names               = TRUE,
  show_column_names            = FALSE,
  row_names_gp                 = gpar(fontsize = 6),
  km = 4,
  row_title_rot                = 0,
  cluster_rows                 = TRUE,
  cluster_row_slices           = FALSE,
  cluster_columns              = FALSE)
htkm

#Ward.D2 Hierarchical Clustering
hthc <- Heatmap(
  pt.matrix,
  name                         = "z-score",
  col                          = colorRamp2(seq(from=-2,to=2,length=11),rev(brewer.pal(11, "Spectral"))),
  show_row_names               = TRUE,
  show_column_names            = FALSE,
  row_names_gp                 = gpar(fontsize = 6),
  clustering_method_rows = "ward.D2",
  clustering_method_columns = "ward.D2",
  row_title_rot                = 0,
  cluster_rows                 = FALSE,
  cluster_row_slices           = FALSE,
  cluster_columns              = FALSE)

print(htkm)
print(hthc)
```
# save quality figure

``` {r}
ggsave("deg.monocle.png", plot = htkm, height=4.5, width = 6, units = "in", dpi = 300)
```

#plot pathways of genes ====hallmark! 
``` {r}
library(tidyverse)
monocle_path<- read.csv("monocle_deg_cd4.csv", stringsAsFactors = TRUE)
head(monocle_path)
```

``` {r}
#filter top 10
monocle_path<- monocle_path[c(1:30),] 

#ggplot 
pathways<- ggplot(monocle_path, aes(reorder(Term,log_pdjust),log_pdjust)) + 
  theme_classic()+ geom_bar(stat = "identity", fill= "#E12695")  + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 13),
        axis.text.y = element_text(size= 10, face = "bold"),
        axis.title.y = element_blank(),
        axis.title.x = element_text(face = "bold", size=15 ),
        axis.text = element_text(face = "bold", size = 8), 
        plot.title = element_text(hjust = 0.5, face = "bold", size = 12)) + 
  labs(y="-log10(P value)", title = "Pathways") + 
  coord_flip() + 
  scale_fill_gradient(low = "red", high = "blue")

pathways
``` 
# save pdf file! 
``` {r}
ggsave("path.monocle.cd4.pdf", plot = pathways, height=7, width = 5, units = "in", dpi = 300)
```

``` {r}
library(tidyverse)
monocle_path_kegg<- read.csv("monocle_cd4_kegg.csv", stringsAsFactors = TRUE)
head(monocle_path_kegg)
```

``` {r}
#filter top 10
mmonocle_path_kegg<- monocle_path_kegg[c(1:30),] 

#ggplot 
pathways.kegg<- ggplot(mmonocle_path_kegg, aes(reorder(Term,log_pdjust),log_pdjust)) + 
  theme_classic()+ geom_bar(stat = "identity", fill= "#1551F1")  + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 13),
        axis.text.y = element_text(size= 10, face = "bold"),
        axis.title.y = element_blank(),
        axis.title.x = element_text(face = "bold", size=15 ),
        axis.text = element_text(face = "bold", size = 8), 
        plot.title = element_text(hjust = 0.5, face = "bold", size = 12)) + 
  labs(y="-log10(P value)", title = "Pathways") + 
  coord_flip() + 
  scale_fill_gradient(low = "red", high = "blue")

pathways.kegg
``` 

# save pdf file! 
``` {r}
ggsave("path.monocle.cd4kegg.pdf", plot = pathways.kegg, height=7, width = 5, units = "in", dpi = 300)
```

### Split analysis for monocle-3 =================! 

``` {r}
view(cd4.monocle)
second_arthritis.cd4<- subset(cd4.monocle, subset = type == "Second arthritis")
view(second_arthritis)
```

``` {r}
# 2. Convert to Seurat object to cell_data_set object ------
cds_second.cd4 <- as.cell_data_set(second_arthritis.cd4) # SeuratWrappers
``` 

# Learn the trajectory graph ! 
``` {r}
cds_second.cd4 <- cluster_cells(cds_second.cd4, 
                     reduction_method = "UMAP", 
                     resolution = 1e-2)  #fit with Seurat !

#plot === optional ===!
plot_cells(cds_second.cd4, color_cells_by = 'ident', 
           label_groups_by_cluster = FALSE,
           group_label_size = 5) +
  theme(legend.position = "right") + 
  theme_bw()# already run and get the same graph! 
```
## trajectory analysis
``` {r}
cds_second.cd4 <- learn_graph(cds_second.cd4)

# plot
plot_cells(cds_second.cd4, color_cells_by = 'ident', 
           label_groups_by_cluster = FALSE,
           group_label_size = 5) +
  theme_void() +
  theme(legend.position = "none") + 
  scale_color_manual(values = c("0"= "#FF3349", "1"= "#3633FF", 
                                "2"="#FF33F3", "3"="#33C4FF", "4"= "#FFAF33", 
                                 "5"= "#9933FF", "gd T"= "#FF33D1")) + 
  theme(legend.position = "right")
```
 

``` {r}
ggsave("cd.traj.pdf", plot = node, height=4.5, width = 7, units = "in", dpi = 300)
```


# one node= 
``` {r, message= FALSE}
https://support.bioconductor.org/p/9145187/
install.packages('htmltools')
cds_second.cd4 <- order_cells(cds_second.cd4)

#plot_cells in pseudotime
cds_second.cd4.trajec<- plot_cells(cds_second.cd4, 
                       color_cells_by = 'pseudotime', 
           label_branch_points = FALSE, 
           label_leaves = FALSE, 
           graph_label_size = 4) + 
  theme_classic() + 
  NoAxes() + 
  theme(legend.position = "right")
cds_second.cd4.trajec

#save monocle3 object
saveRDS(cds_second.cd4, file = "cds_second_arthritis.monocle3.cd4.rds")  # saved on 6/6/2028
cds_second.cd4<- readRDS("cds_second_arthritis.monocle3.cd4.rds")
``` 


``` {r}
#resource == https://rpubs.com/mahima_bose/Seurat_and_Monocle3_p 
cds_second.cd4$monocle3_pseudotime <- pseudotime(cds_second.cd4)
data.pseudo.cd4.second <- as.data.frame(colData(cds_second.cd4))

ggplot(data.pseudo.cd4.second, aes(monocle3_pseudotime, seurat_clusters, fill = seurat_clusters)) + 
  geom_boxplot()
``` 
``` {r}
ggplot(data.pseudo.cd4.second, aes(monocle3_pseudotime, reorder(seurat_clusters, monocle3_pseudotime), fill = seurat_clusters)) + geom_boxplot()
```
``` {r}
library(ggplot2)
library(ggridges)

pseudotime<- ggplot(data.pseudo.cd4.second, aes(x = monocle3_pseudotime, 
                                     y = reorder(seurat_clusters, monocle3_pseudotime), fill = seurat_clusters)) +
  geom_density_ridges() + 
  theme_classic() + 
  theme(legend.position = "none", 
        axis.title.y  = element_blank(), 
        axis.text = element_text(size= 12, face = "bold"), 
        axis.title.x = element_text(siz= 14, face = "bold")) + 
  labs(x= "Pseudotime")
pseudotime
``` 


#new code from Yanshuo !!! 
``` {r}
# First arthritis !!! 
library(monocle3)
library(SeuratWrappers)
DimPlot(cd4.monocle)
second_arthritis.cd4<- subset(cd4.monocle, subset = type == "Second arthritis")

monocle.cds.cd4s <- as.cell_data_set(second_arthritis.cd4)

monocle.cds.cd4s <- cluster_cells(cds = monocle.cds.cd4s, 
                             reduction_method = "UMAP")
monocle.cds.cd4s <- learn_graph(monocle.cds.cd4s)

startCells <- Cells(second_arthritis.cd4)[second_arthritis.cd4$seurat_clusters == 3]

monocle.cds.cd4s <- order_cells(monocle.cds.cd4s,
                           reduction_method = "UMAP", 
                           root_cells = startCells)
plot_cells(cds = monocle.cds.cd4s,
           color_cells_by = "pseudotime",
           show_trajectory_graph = T,
           trajectory_graph_color = "grey",
           trajectory_graph_segment_size = 0.5,
           graph_label_size = 2,
           cell_size = 1,
           label_cell_groups = F,
           label_groups_by_cluster = F,
           label_branch_points = F,
           label_roots = F,
           label_leaves = F)


#plot cells !!! 
#plot the pseudotime !!
monocle.cds.cd4s$monocle3_pseudotime <- pseudotime(monocle.cds.cd4s)
data.pseudo_second <- as.data.frame(colData(monocle.cds.cd4s))

#ggplot! 
ggplot(data.pseudo_second , aes(monocle3_pseudotime, seurat_clusters, fill = seurat_clusters)) + geom_boxplot()

#order the cells !! 
ggplot(data.pseudo_second, aes(monocle3_pseudotime, reorder(seurat_clusters, monocle3_pseudotime), fill = seurat_clusters)) + geom_boxplot() + 
   theme_bw() + 
  theme(legend.position = "none", 
        axis.title.y  = element_blank(), 
        axis.text = element_text(size= 12, face = "bold"), 
        axis.title.x = element_text(siz= 16, face = "bold"), 
        panel.border = element_rect(colour = "black", fill = NA, size = 1.3), 
        panel.grid = element_blank()) + 
  labs(x= "Pseudotime")

#2024 IOTOX !!! 
cd4_pseudos<- ggplot(data.pseudo_second, aes(monocle3_pseudotime, reorder(seurat_clusters, monocle3_pseudotime), fill = seurat_clusters)) + geom_boxplot() + 
   theme_minimal() + 
  theme(legend.position = "none", 
        axis.title.y  = element_blank(), 
        axis.text = element_text(size= 12, face = "bold"), 
        axis.title.x = element_text(siz= 16, face = "bold"), 
        panel.grid.major = element_line(size = 1), 
        panel.grid.minor = element_line(size= 1), 
        axis.text.y = element_blank()) + 
  labs(x= "Pseudotime") 
cd4_pseudos

#Density plot !! 
library(ggplot2)
library(ggridges)

pseudotime_second<- ggplot(data.pseudo_second, aes(x = monocle3_pseudotime, y = reorder(seurat_clusters, monocle3_pseudotime), fill = seurat_clusters)) +
  geom_density_ridges() + 
  theme_bw() + 
  theme(legend.position = "none", 
        axis.title.y  = element_blank(), 
        axis.text = element_text(size= 15, face = "bold"), 
        axis.title.x = element_text(siz= 18, face = "bold"), 
        panel.border = element_rect(colour = "black", fill = NA, size = 1.3), 
        panel.grid = element_blank()) + 
  labs(x= "Pseudotime")
pseudotime_second
ggsave("cd4.pseudo.s.pdf", plot = pseudotime_second, height=4.5, width = 4, units = "in", dpi = 300)
``` 

#differentially expressed genes along the trajectory! 

``` {r}
deg.cd4.second.arthritis <- graph_test(monocle.cds.cd4s, neighbor_graph = "principal_graph", cores = 4)

#save deg object! 
saveRDS(deg.cd4.second.arthritis, file = "deg.cd4.second.arthritis.rds")
deg.cd4.second.arthritis<- readRDS("deg.cd4.second.arthritis.rds")


#short and save deg! 
cd4.deg.second.arthritis_short<- deg.cd4.second.arthritis %>% filter(q_value < 0.05, status== "OK")
deg.cd4.second.arthritis %>% arrange(q_value) %>% filter(status == "OK")
```

#write csv file!

``` {r}
write.csv(cd4.deg.second.arthritis_short, file = "monocle.cd4.second.arthritis.csv")
```


``` {r}
# https://github.com/cole-trapnell-lab/monocle-release/issues/295 

library(ComplexHeatmap)
library(ggplot2)
library(dplyr)
library(RColorBrewer)
library(circlize)
library(monocle3)

#modulated_genes <- graph_test(cds, neighbor_graph = "principal_graph", cores = 4)
genes.cd4.second <- row.names(subset(deg.cd4.second.arthritis, q_value==0  & morans_I > 0.19))
genes.cd4.second

genes.cd4.second <- setdiff(genes.cd4.second, "COTL1")

pt.matrix <- exprs(monocle.cds.cd4s)[match(genes.cd4.second,rownames(rowData(monocle.cds.cd4s))),order(pseudotime(monocle.cds.cd4s))]
#Can also use "normalized_counts" instead of "exprs" to use various normalization methods, for example:
#normalized_counts(cds, norm_method = "log")

pt.matrix <- t(apply(pt.matrix,1,function(x){smooth.spline(x,df=3)$y}))
pt.matrix <- t(apply(pt.matrix,1,function(x){(x-mean(x))/sd(x)}))
rownames(pt.matrix) <- genes.cd4.second;

#K means with 6 groups
library(ComplexHeatmap)
htkm.cd4.second <- Heatmap(
  pt.matrix,
  name                         = "z-score",
  col                          = colorRamp2(seq(from=-2,to=2,length=11),rev(brewer.pal(11, "Spectral"))),
  show_row_names               = TRUE,
  show_column_names            = FALSE,
  row_names_gp                 = gpar(fontsize = 6.5),
  km = 4,
  row_title_rot                = 0,
  cluster_rows                 = TRUE,
  cluster_row_slices           = FALSE,
  cluster_columns              = FALSE)
saveRDS(htkm.cd4.second, file= "htkm.cd4.second.rds")
htkm.cd4.second <- readRDS("htkm.cd4.second.rds")
print(htkm.cd4.second)
```

# perform TradeSeq for second arthritis 
``` {r}
#first step 
cds_second.cd4 <- cluster_cells(cds_second.cd4, 
                     reduction_method = "UMAP", 
                     resolution = 1e-2)  #fit with Seurat !

# First display, coloring by the cell types from Paul et al
plot_cells(cds_second.cd4, 
           label_groups_by_cluster = FALSE, 
           cell_size = 1,
           color_cells_by = "ident")
``` 
# Construct the graph
# Note that, for the rest of the code to run, the graph should be fully connected
``` {r}
cds_second.cd4 <- learn_graph(cds_second.cd4, use_partition = FALSE)
``` 
``` {r}
# We find all the cells that are close to the starting point
cell_ids <- colnames(cds_second.cd4)[second_arthritis.cd4$seurat_clusters ==  "3"]
closest_vertex <- cds_second.cd4@principal_graph_aux[["UMAP"]]$pr_graph_cell_proj_closest_vertex
closest_vertex <- as.matrix(closest_vertex[colnames(cds_second.cd4), ])
closest_vertex <- closest_vertex[cell_ids, ]
closest_vertex <- as.numeric(names(which.max(table(closest_vertex))))
mst <- principal_graph(cds)$UMAP
root_pr_nodes <- igraph::V(mst)$name[closest_vertex]

# We compute the trajectory
cds_second.cd4 <- order_cells(cds_second.cd4, root_pr_nodes = root_pr_nodes)
plot_cells(cds_second.cd4, color_cells_by = "pseudotime")
``` 

``` {r}
library(magrittr)
# Get the closest vertice for every cell
y_to_cells <-  principal_graph_aux(cds_second.cd4)$UMAP$pr_graph_cell_proj_closest_vertex %>%
  as.data.frame()
y_to_cells$cells <- rownames(y_to_cells)
y_to_cells$Y <- y_to_cells$V1

# Get the root vertices
# It is the same node as above
root <- cds_second.cd4@principal_graph_aux$UMAP$root_pr_nodes

# Get the other endpoints
endpoints <- names(which(igraph::degree(mst) == 1))
endpoints <- endpoints[!endpoints %in% root]

# For each endpoint
cellWeights <- lapply(endpoints, function(endpoint) {
  # We find the path between the endpoint and the root
  path <- igraph::shortest_paths(mst, root, endpoint)$vpath[[1]]
  path <- as.character(path)
  # We find the cells that map along that path
  df <- y_to_cells[y_to_cells$Y %in% path, ]
  df <- data.frame(weights = as.numeric(colnames(cds_second.cd4) %in% df$cells))
  colnames(df) <- endpoint
  return(df)
  }) %>% do.call(what = 'cbind', args = .) %>%
    as.matrix()
rownames(cellWeights) <- colnames(cds_second.cd4)
pseudotime <- matrix(pseudotime(cds_second.cd4), ncol = ncol(cellWeights),
                     nrow = ncol(cds_second.cd4), byrow = FALSE)

# Check for cells with no weights   ==== use it in case of no positive cell weight
if (any(rowSums(cellWeights) == 0)) {
  stop("Some cells have no positive cell weights.")
}

# Exclude cells with no positive weights
valid_cells <- rowSums(cellWeights) > 0
cellWeights <- cellWeights[valid_cells, ]
pseudotime <- pseudotime[valid_cells, ]
```

# Fit GAM!

``` {r}
# Convert counts to sparse matrix for memory efficiency
# Ensure the counts matrix matches the valid cells
counts <- as(counts(cds_second.cd4), "dgCMatrix") # good code 

#fit GAM
for (nknots in seq(10, 50, by = 5)) {
  cat("Fitting GAM with", nknots, "knots...\n")
  sce <- fitGAM(counts = counts,
                pseudotime = pseudotime,
                cellWeights = cellWeights,
                nknots = nknots)
  if (!any(grepl("Impossible to place a knot", warnings()))) {
    break
  }
}

#fit GAM ! 
sce_cd4_second <- fitGAM(counts = counts,
                pseudotime = pseudotime,
                cellWeights = cellWeights,
                nknots = )
``` 

## new fitGAM ! 
``` {r}
# Extract pseudotime and assume uniform cell weights
pseudotime <- as.numeric(pseudotime(cds_second.cd4))
cellWeights <- matrix(1, ncol = 1, nrow = length(pseudotime))


# Convert counts to sparse matrix for memory efficiency
counts_sparse <- as(counts(cds_second.cd4), "dgCMatrix")

# Create SingleCellExperiment object
sce_cd4_second <- SingleCellExperiment(assays = list(counts = counts_sparse))
colData(sce_cd4_second)$pseudotime <- pseudotime
colData(sce_cd4_second)$cellWeights <- cellWeights

# Set up parallel processing
BPPARAM <- SnowParam(workers = 4)  # Adjust based on your system's capabilities

# Fit the GAM
sce_GAM <- fitGAM(counts = counts(sce_cd4_second), 
                  pseudotime = pseudotime, 
                  cellWeights = cellWeights, 
                  BPPARAM = BPPARAM)


# Perform differential expression testing as a function of pseudotime! 
diffExpr_cd4second <- associationTest(sce_GAM)
diffExpr_cd4second_omitna<- na.omit(diffExpr_cd4second)
diffExpr_cd4second.significant_genes <- rownames(diffExpr_cd4second_omitna)[which(diffExpr_cd4second_omitna$pvalue < 0.05)]

write.csv(diffExpr_cd4second.significant_genes, file = "second_CD4_deg_pseudo.csv")

#adjust-pvalue ! 
library(data.table)
diffExpr_CD4_second_padj <- data.table(gene = rownames(diffExpr_cd4second_omitna), diffExpr_cd4second_omitna)
diffExpr_CD4_second_padj$padj_all <- p.adjust(diffExpr_CD4_second_padj$pvalue, method = "fdr")
diffExpr_CD4_second_padj<- na.omit(diffExpr_CD4_second_padj)

#filter significant genes 
second_deg_CD4_pseudo.adj<- diffExpr_CD4_second_padj %>% filter(padj_all < 0.05)
write.csv(diffExpr_CD4_second_padj, file = "second_CD4_deg_pseudo.padj.csv")

# View significant genes
print(significant_genes)


# read it back in !! 
CD4_deg_tradeseq_second<- read.csv("second_CD4_deg_pseudo.padj.csv")
CD4_deg_tradeseq_second<- CD4_deg_tradeseq_second %>% filter(padj_all < 0.05)
write.csv(CD4_deg_tradeseq_second, file= "CD4_second_traseq.padj0.05.csv")

# read  it back in and order by logfc
CD4.tradeseq.logfc<- read.csv("CD4_second_traseq.padj0.05.csv", 
                              stringsAsFactors = TRUE)

# order it from highest log2FC to lowest
deg_second_CD4_order<- CD4.tradeseq.logfc %>% arrange(desc(meanLogFC))
head(deg_second_CD4_order)

# Save deg of cluster 1
write.csv(deg_second_CD4_order, file = "CD4_second_tradeseq_logfc_order.csv")
``` 


#plot pathways of genes ====hallmark! 
``` {r}
library(tidyverse)
monocle_path<- read.csv("monocle_deg_cd4.csv", stringsAsFactors = TRUE)
monocle_path<- read.csv("monocle_deg_cd4_second.csv", stringsAsFactors = TRUE)
head(monocle_path)
```

``` {r}
#filter top 10
monocle_path<- monocle_path[c(1:30),] 

#ggplot 
pathways<- ggplot(monocle_path, aes(reorder(Term,log_pdjust),log_pdjust)) + 
  theme_classic()+ geom_bar(stat = "identity", fill= "#E12695")  + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 13),
        axis.text.y = element_text(size= 11, face = "bold"),
        axis.title.y = element_blank(),
        axis.title.x = element_text(face = "bold", size=13),
        axis.text = element_text(face = "bold", size = 8), 
        plot.title = element_text(hjust = 0.5, face = "bold", size = 12)) + 
  ylim(c(0,4)) + 
  labs(y="-log10(P value)", title = "Pathways") + 
  coord_flip() + 
  scale_fill_gradient(low = "red", high = "blue")
pathways
``` 
```{r}
pathways <- ggplot(monocle_path, aes(reorder(Term, log_pdjust), log_pdjust)) + 
  theme_classic() +
  geom_point(size = 8, color = "#E12695") + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size = 13),
        axis.text.y = element_text(size = 11, face = "bold"),
        axis.title.y = element_blank(),
        axis.title.x = element_text(face = "bold", size = 13),
        axis.text = element_text(face = "bold", size = 8), 
        plot.title = element_text(hjust = 0.5, face = "bold", size = 12)) + 
  ylim(c(0, 4)) + 
  labs(y = "-log10(P value)", title = "Pathways") + 
  coord_flip()
pathways
``` 



# save plots ! 
``` {r}
ggsave("pathways.second_arthrits_CD4.pdf", plot = pathways, height=7, width = 5.5, units = "in", dpi = 300)
```

# CD4 trajectory for first arthritis !

``` {r}
First_arthritis.cd4<- subset(cd4.monocle, subset = type == "First arthritis")
view(second_arthritis)
```

``` {r}
# 2. Convert to Seurat object to cell_data_set object ------
cds_First.cd4 <- as.cell_data_set(First_arthritis.cd4) # SeuratWrappers
``` 

# Learn the trajectory graph ! 
``` {r}
cds_First.cd4 <- cluster_cells(cds_First.cd4, 
                     reduction_method = "UMAP", 
                     resolution = 1e-3)  #fit with Seurat !

#plot === optional ===!
plot_cells(cds_First.cd4, color_cells_by = 'ident', 
           label_groups_by_cluster = FALSE,
           group_label_size = 5) +
  theme(legend.position = "right") + 
  theme_bw()# already run and get the same graph! 
```
## trajectory analysis
``` {r}
cds_First.cd4 <- learn_graph(cds_First.cd4)

# plot
plot_cells(cds_First.cd4, color_cells_by = 'ident', 
           label_groups_by_cluster = FALSE,
           group_label_size = 5) +
  theme_void() +
  theme(legend.position = "none") + 
  scale_color_manual(values = c("0"= "#FF3349", "1"= "#3633FF", 
                                "2"="#FF33F3", "3"="#33C4FF", "4"= "#FFAF33", 
                                 "5"= "#9933FF", "gd T"= "#FF33D1")) + 
  theme(legend.position = "right")
```

# one node= 
``` {r, message= FALSE}
https://support.bioconductor.org/p/9145187/
install.packages('htmltools')
cds_First.cd4 <- order_cells(cds_First.cd4)

#plot_cells in pseudotime
cds_First.cd4.trajec<- plot_cells(cds_First.cd4, 
                       color_cells_by = 'pseudotime', 
           label_branch_points = FALSE, 
           label_leaves = FALSE, 
           graph_label_size = 4) + 
  theme_classic() + 
  NoAxes() + 
  theme(legend.position = "right")
cds_First.cd4.trajec

#save monocle3 object
saveRDS(cds_First.cd4, file = "cds_First_arthritis.monocle3.cd4.rds")  # saved on 6/6/2028
cds_First.cd4<- readRDS("cds_First_arthritis.monocle3.cd4.rds")
``` 


``` {r}
#resource == https://rpubs.com/mahima_bose/Seurat_and_Monocle3_p 
cds_First.cd4$monocle3_pseudotime <- pseudotime(cds_First.cd4)
data.pseudo.cd4.first <- as.data.frame(colData(cds_First.cd4))

ggplot(data.pseudo.cd4.first, aes(monocle3_pseudotime, seurat_clusters, fill = seurat_clusters)) + 
  geom_boxplot()
``` 
``` {r}
ggplot(data.pseudo.cd4.first, aes(monocle3_pseudotime, reorder(seurat_clusters, monocle3_pseudotime), fill = seurat_clusters)) + geom_boxplot()
```

#monocle 3: Yanshuo code !!! 
``` {r}
# First arthritis !!! 
first_arthritis.cd4<- subset(cd4.monocle, subset = type == "First arthritis")

monocle.cds.cd4f <- as.cell_data_set(first_arthritis.cd4)

monocle.cds.cd4f <- cluster_cells(cds = monocle.cds.cd4f, 
                             reduction_method = "UMAP")
monocle.cds.cd4f <- learn_graph(monocle.cds.cd4f)

startCells <- Cells(first_arthritis.cd4)[first_arthritis.cd4$seurat_clusters == 3]

monocle.cds.cd4f<- order_cells(monocle.cds.cd4f,
                           reduction_method = "UMAP", 
                           root_cells = startCells)
plot_cells(cds = monocle.cds.cd4f,
           color_cells_by = "pseudotime",
           show_trajectory_graph = T,
           trajectory_graph_color = "grey",
           trajectory_graph_segment_size = 0.5,
           graph_label_size = 2,
           cell_size = 1,
           label_cell_groups = F,
           label_groups_by_cluster = F,
           label_branch_points = F,
           label_roots = F,
           label_leaves = F)


#plot cells !!! 
#plot the pseudotime !!
monocle.cds.cd4f$monocle3_pseudotime<- pseudotime(monocle.cds.cd4f)
data.pseudo_first <- as.data.frame(colData(monocle.cds.cd4f))

#ggplot! 
ggplot(data.pseudo_first , aes(monocle3_pseudotime, seurat_clusters, fill = seurat_clusters)) + geom_boxplot()

#order the cells !! 
ggplot(data.pseudo_first, aes(monocle3_pseudotime, reorder(seurat_clusters, monocle3_pseudotime), fill = seurat_clusters)) + geom_boxplot() + 
   theme_bw() + 
  theme(legend.position = "none", 
        axis.title.y  = element_blank(), 
        axis.text = element_text(size= 12, face = "bold"), 
        axis.title.x = element_text(siz= 16, face = "bold"), 
        panel.border = element_rect(colour = "black", fill = NA, size = 1.3), 
        panel.grid = element_blank()) + 
  labs(x= "Pseudotime")

#2024 IOTOX !! 
library(tidyverse)
cd4_pseudof<- ggplot(data.pseudo_first, aes(monocle3_pseudotime, reorder(seurat_clusters, monocle3_pseudotime), fill = seurat_clusters)) + geom_boxplot() + 
   theme_minimal() + 
  theme(legend.position = "none", 
        axis.title.y  = element_blank(), 
        axis.text.x  = element_text(size= 12, face = "bold"), 
        axis.title.x = element_text(siz= 16, face = "bold"),
         panel.grid.major = element_line(size = 1), 
        panel.grid.minor = element_line(size= 1), 
        axis.text.y = element_blank()) + 
  labs(x= "Pseudotime")
cd4_pseudof

#comined 
library(patchwork)
pseudo_combine<- cd4_pseudof + cd4_pseudos
pseudo_combine

#save high quality plots !! 
ggsave("cd4.pseudo.comine.tiff", plot = pseudo_combine, height=4.5, width = 4.5, units = "in", dpi = 300)

#Density plot !! 
library(ggplot2)
library(ggridges)

pseudotime_first<- ggplot(data.pseudo_first, aes(x = monocle3_pseudotime, y = reorder(seurat_clusters, monocle3_pseudotime), fill = seurat_clusters)) +
  geom_density_ridges() + 
  theme_bw() + 
  theme(legend.position = "none", 
        axis.title.y  = element_blank(), 
        axis.text = element_text(size= 15, face = "bold"), 
        axis.title.x = element_text(siz= 18, face = "bold"), 
        panel.border = element_rect(colour = "black", fill = NA, size = 1.3), 
        panel.grid = element_blank()) + 
  labs(x= "Pseudotime")
pseudotime_first
ggsave("cd4.pseudo.f.pdf", plot = pseudotime_first, height=4.5, width = 4, units = "in", dpi = 300)
``` 

#differentially expressed genes along the trajectory! 
``` {r}
deg.cd4.First.arthritis <- graph_test(monocle.cds.cd4f, neighbor_graph = "principal_graph", cores = 4)

#save deg object! 
saveRDS(deg.cd4.First.arthritis, file = "deg.cd4.First.arthritis.rds")
deg.cd4.First.arthritis<- readRDS("deg.cd4.First.arthritis.rds")

#short and save deg! 
cd4.deg.First.arthritis_short<- deg.cd4.First.arthritis %>% filter(q_value < 0.05, status== "OK")
deg.cd4.First.arthritis %>% arrange(q_value) %>% filter(status == "OK")
```


## new fitGAM ! 
``` {r}
# Extract pseudotime and assume uniform cell weights
library(tradeSeq)
pseudotime <- as.numeric(pseudotime(cds_First.cd4))
cellWeights <- matrix(1, ncol = 1, nrow = length(pseudotime))


# Convert counts to sparse matrix for memory efficiency
counts_sparse <- as(counts(cds_First.cd4), "dgCMatrix")

# Create SingleCellExperiment object
sce_cd4_First <- SingleCellExperiment(assays = list(counts = counts_sparse))
colData(sce_cd4_First)$pseudotime <- pseudotime
colData(sce_cd4_First)$cellWeights <- cellWeights

# Fit the GAM
sce_GAM_cd4_first <- fitGAM(counts = counts(cds_First.cd4), 
                  pseudotime = pseudotime, 
                  cellWeights = cellWeights)

#save this obj
saveRDS(sce_GAM_cd4_first, file= "fitGAM_cd4_first.rds")


# Perform differential expression testing as a function of pseudotime! 
diffExpr_cd4first <- associationTest(sce_GAM_cd4_first)
diffExpr_cd4first_omitna<- na.omit(diffExpr_cd4first)
diffExpr_cd4first.significant_genes <- rownames(diffExpr_cd4first_omitna)[which(diffExpr_cd4first_omitna$pvalue < 0.05)]

# Write a csv file !! 
write.csv(diffExpr_cd4first.significant_genes, file = "first_CD4_deg_pseudo.csv")

#adjust-pvalue ! 
library(data.table)
diffExpr_CD4_first_padj <- data.table(gene = rownames(diffExpr_cd4first_omitna), diffExpr_cd4first_omitna)
diffExpr_CD4_first_padj$padj_all <- p.adjust(diffExpr_CD4_first_padj$pvalue, method = "fdr")
diffExpr_CD4_first_padj<- na.omit(diffExpr_CD4_first_padj)

#filter significant genes 
first_deg_CD4_pseudo.adj<- diffExpr_CD4_first_padj %>% filter(padj_all < 0.05)
write.csv(first_deg_CD4_pseudo.adj, file = "first_CD4_deg_pseudo.padj.csv")

# View significant genes
print(significant_genes)
``` 

# arranged base on logfc changes

``` {r}
library(tidyverse)
#first arthritis deg of CD8 clusters! 
first_deg_CD4_pseudo.adj<- read.csv("first_CD4_deg_pseudo.padj.csv", 
                                 stringsAsFactors = TRUE)
cd4first_monocledeg<- first_deg_CD4_pseudo.adj %>% arrange(desc(meanLogFC))
head(cd4first_monocledeg)

#save obj ! 
install.packages("openxlsx")
library(openxlsx)

# save obj! 
write.xlsx(cd4first_monocledeg, file = "CD4tradseq_first_arthritis.xlsx")  ### done! 
``` 

#write csv file!

``` {r}
write.csv(cd4.deg.First.arthritis_short, file = "monocle.cd4.first.arthritis.csv")
```

# plot 

#plot pathways of genes ====Gene Ontology! 
``` {r}
library(tidyverse)
monocle_path<- read.csv("monocle_deg_cd4_first.csv", stringsAsFactors = TRUE)
head(monocle_path)
```

``` {r}
#filter top 10
monocle_path<- monocle_path[c(1:15),] 

#ggplot 
pathways<- ggplot(monocle_path, aes(reorder(Term,log_pdjust),log_pdjust)) + 
  theme_classic()+ geom_bar(stat = "identity", fill= "#E12695")  + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 13),
        axis.text.y = element_text(size= 11, face = "bold"),
        axis.title.y = element_blank(),
        axis.title.x = element_text(face = "bold", size=13),
        axis.text = element_text(face = "bold", size = 8), 
        plot.title = element_text(hjust = 0.5, face = "bold", size = 12)) + 
  labs(y="-log10(P value)", title = "Pathways") + 
  coord_flip()
pathways
``` 

# save plots ! 
``` {r}
ggsave("pathways.first_arthrits_CD4.pdf", plot = pathways, height=7, width = 5.5, units = "in", dpi = 300)
```


``` {r}
# https://github.com/cole-trapnell-lab/monocle-release/issues/295 

library(ComplexHeatmap)
library(ggplot2)
library(dplyr)
library(RColorBrewer)
library(circlize)
library(monocle3)

#modulated_genes <- graph_test(cds, neighbor_graph = "principal_graph", cores = 4)
genes.cd4.first <- row.names(subset(deg.cd4.First.arthritis, q_value==0  & morans_I > 0.28))
genes.cd4.first

pt.matrix <- exprs(monocle.cds.cd4f)[match(genes.cd4.first,rownames(rowData(monocle.cds.cd4f))),order(pseudotime(monocle.cds.cd4f))]
#Can also use "normalized_counts" instead of "exprs" to use various normalization methods, for example:
#normalized_counts(cds, norm_method = "log")

pt.matrix <- t(apply(pt.matrix,1,function(x){smooth.spline(x,df=3)$y}))
pt.matrix <- t(apply(pt.matrix,1,function(x){(x-mean(x))/sd(x)}))
rownames(pt.matrix) <- genes.cd4.first;

#K means with 6 groups
library(ComplexHeatmap)
htkm.cd4.first <- Heatmap(
  pt.matrix,
  name                         = "z-score",
  col                          = colorRamp2(seq(from=-2,to=2,length=11),rev(brewer.pal(11, "Spectral"))),
  show_row_names               = TRUE,
  show_column_names            = FALSE,
  row_names_gp                 = gpar(fontsize = 6.5),
  km = 4,
  row_title_rot                = 0,
  cluster_rows                 = TRUE,
  cluster_row_slices           = FALSE,
  cluster_columns              = FALSE)
saveRDS(htkm.cd4.first, file= "htkm.cd4.first.rds")

htkm.cd4.first
```

# TradeSeq for gene ontology for CD4 Trajectory !! 
``` {r}

go__CD4_arthritis<- read.csv("GO_CD4_arthritis_newp .csv", stringsAsFactors = TRUE)
head(go__CD4_arthritis)
go__CD4_arthritis$group<- factor(go__CD4_arthritis$group, levels = c("First arthritis", "Second arthritis"))

#ggplot----- !! 
GO_CD4_new<- ggplot(go__CD4_arthritis, aes(x = reorder(go_terms, gene_ratio), y = gene_ratio)) +
  geom_point(aes(size = gene_count, color = padj), stroke= 0.5) +
  scale_color_gradient(low = "red", high = "blue") +
  theme_bw() +
  theme(
    axis.text.x = element_text(face = "bold", size = 12), 
    axis.text.y = element_text(face = "bold", size = 11.6), 
    axis.title.y = element_blank(), 
    axis.title.x = element_text(face = "bold", size = 15), 
    legend.title = element_text(face = "bold"), 
    legend.text = element_text(face = "bold"), 
    panel.grid = element_blank(), 
    panel.border = element_rect(colour = "black", fill = NA, size = 1.5), 
    strip.text = element_text(face= "bold", size=12),
    legend.position = c(0.90, 0.35)) + 
  labs(x = "GO Terms", y = "Gene Ratio", color = "padj", size = "Count") + 
  coord_flip() +
  facet_wrap(~ group) + 
  scale_y_continuous(breaks = seq(0.1, 0.6, by = 0.1))

geom_point(aes(size=pct.exp), shape = 21, stroke=0.5)

GO_CD4_new

ggsave("GO_CD4__arthritis.pdf", plot = GO_CD4_new, height=6, width = 9, units = "in", dpi = 300)
``` 

# Try new ways to run geneset enrichment analysis in R















#=============================================================== 
              # Treg 
              
# Run deg for both cluster ! 

#cluster 0
``` {r}
Treg.T.pc10<- readRDS("Treg.obj.rds")
DimPlot(Treg.T.pc10)
 
#compare gene expression of cluster 0 to the rest of the cluster
prepsct<- PrepSCTFindMarkers(Treg.T.pc10)
cluster0<- FindMarkers(prepsct,assay = "SCT", 
                       ident.1 = 0, 
                       only.pos = TRUE, 
                       min.pct = 0.25, 
                       logfc.threshold = 0.25, 
                       verbose = FALSE)
head(cluster0, n = 20)
``` 

**order it from highest log2FC to lowest
``` {r}
cluster0<- cluster0 %>% arrange(desc(avg_log2FC))
head(cluster0)
``` 
#cluster 1
``` {r}
#compare gene expression of cluster 0 to the rest of the cluster
prepsct<- PrepSCTFindMarkers(Treg.T.pc10)
cluster1<- FindMarkers(prepsct,assay = "SCT", 
                       ident.1 = 1, 
                       only.pos = TRUE, 
                       min.pct = 0.25, 
                       logfc.threshold = 0.25, 
                       verbose = FALSE)
head(cluster1, n = 20)
```

**order it from highest log2FC to lowest
``` {r}
cluster1<- cluster1 %>% arrange(desc(avg_log2FC))
head(cluster1)
``` 

#dot plots! 
``` {r}
features<- c("CD4", "FOXP3", "CTLA4", "PDCD1", "TIGIT","LAG3","IL2RA","IKZF2", "ICOS","KLRG1","IL10", "TGFB1")

#plot
dotplot<- DotPlot(Treg.annota, features = features, cols = c("blue", "red"), dot.scale = 6) + 
  theme_classic() +
  RotatedAxis() + 
  coord_flip() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=15), 
        axis.title.y= element_text(size = 14, face = "bold"), 
        axis.text = element_text(size=13, face = "bold"), 
        axis.text.x = element_text(angle=90,hjust=0.95,vjust=0.2, face = "bold", size= 12), 
        axis.title.x = element_blank(),
        legend.title = element_text(size= 12, face = "bold"), 
        legend.text = element_text(size= 10, face = "bold")) 
dotplot
``` 
# rename the clusters ! 
``` {r}
Treg.annota <- RenameIdents(Treg.T.pc10, `0` ="iTreg", `1` ="nTreg")
DimPlot(Treg.annota)
``` 

# save this plots 
``` {r}
ggsave("tregnew.pdf", plot = dotplot, height=3.5, width = 6,  units = "in", dpi = 300)
```


# Feature plots ! 
``` {r, fig.width= 7, fig.height= 8}
FeaturePlot(Treg.T.pc10, features = c("IKZF2", "LAG3", "IL2RA", "CD4", "FOXP3", "CD19" ), pt.size = 0.2,
    ncol = 2)
```


# Save deg of cluster 0

``` {r}
write.csv(cluster1, file = "Treg/deg_sct_cluster1.csv")
```

# Cluster proportion! 
``` {r}

# patient 164S
p.164s <- subset(Treg.T.pc10, subset = orig.ident == "164S")
view(p.164s)

# cluster_frequency
cluster_frequency.164s <- p.164s@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.164s")

#write csv file to save the result
write.csv(cluster_frequency.164s, file="Treg/cluster_frequency.164S.CSV")


# patient 164S2
p.164S2 <- subset(Treg.T.pc10, subset = orig.ident == "164S2")
view(p.164S2)

# cluster_frequency
cluster_frequency.164S2 <- p.164S2@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.164S2")

write csv file to save the result
write.csv(cluster_frequency.164S2, file="Treg/cluster_frequency.164S2.CSV")


# patient 164S2
p.184S2 <- subset(Treg.T.pc10, subset = orig.ident == "184S2")
view(p.184S2)

# cluster_frequency
cluster_frequency.184S2 <- p.184S2@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.184S2")

write csv file to save the result
write.csv(cluster_frequency.184S2, file="Treg/cluster_frequency.184S2.CSV")

# patient 164S2
p.184S3 <- subset(Treg.T.pc10, subset = orig.ident == "184S3")
view(p.184S3)

# cluster_frequency
cluster_frequency.184S3 <- p.184S3@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.184S3")

write csv file to save the result
write.csv(cluster_frequency.184S3, file="Treg/cluster_frequency.184S3.CSV")


# patient 164S2
p.218S <- subset(Treg.T.pc10, subset = orig.ident == "218S")
view(p.218S)

# cluster_frequency
cluster_frequency.218S <- p.218S@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.218S")

write csv file to save the result
write.csv(cluster_frequency.218S, file="Treg/cluster_frequency.218S.CSV")


# cluster_frequency
# patient 164S2
p.218S2 <- subset(Treg.T.pc10, subset = orig.ident == "218S2")
view(p.218S2)
cluster_frequency.218S2 <- p.218S2@meta.data %>%
  group_by(seurat_clusters) %>%
  summarise(count=n()) %>%
  mutate(relative_frequency = count*100/sum(count)) %>%
  mutate(data_set = "p.218S2")

write csv file to save the result
write.csv(cluster_frequency.218S2, file="Treg/cluster_frequency.218S2.CSV")
``` 


#plot the frequency! 

#plot cluster frequency 
``` {r}
#read csv file back in 
proportion<- read.csv("cluster_frequency_treg.csv", stringsAsFactors = TRUE)
proportion
``` 

## cluster proportion
``` {r}
iTreg<- proportion %>% filter(cluster_id == "iTreg")
iTreg$group<- factor(iTreg$group, levels = c("first arthritis", "second arthritis"))


#visualization ! 
p1<- ggplot(iTreg, aes(x= group, y= frequency, fill= group)) +
  theme_classic() +
  geom_bar(position = 'dodge', stat = 'summary', fun.y='mean') +
  geom_errorbar(stat = 'summary', position = 'dodge', width=0.1) +
  geom_point(size= 4, aes(x=group), shape=21, position = position_dodge(width = 0)) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=14
  ), axis.title.y = element_text(size = 16, face = "bold"), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=14, face = "bold"), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% iTreg of foxp3+ cells", title = "inducible Treg")
p1
``` 

# save this plots 
``` {r}
ggsave("itreg.pdf", plot = p1, height=4, width = 3,  units = "in", dpi = 300)
```


## cluster proportion
``` {r}
nTreg<- proportion %>% filter(cluster_id == "nTreg")
nTreg$group<- factor(nTreg$group, levels = c("first arthritis", "second arthritis"))


#visualization ! 
p2<- ggplot(nTreg, aes(x= group, y= frequency, fill= group)) +
  theme_classic() +
  geom_bar(position = 'dodge', stat = 'summary', fun.y='mean') +
  geom_errorbar(stat = 'summary', position = 'dodge', width=0.1) +
  geom_point(size= 4, aes(x=group), shape=21, position = position_dodge(width = 0)) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=14
  ), axis.title.y = element_text(size = 16, face = "bold"), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=14, face = "bold"), 
  axis.text.x = element_blank(), 
  legend.position = "none") + 
  labs(x= "Groups", y= "% nTreg of foxp3+ cells", title = "natural Treg")
p2
``` 
# save this plots 
``` {r}
ggsave("ntreg.pdf", plot = p2, height=4, width = 3,  units = "in", dpi = 300)
```
## Violen plot of TNF, IFNg, GranB, 

``` {r}
# subset only first and second arthritis ! 
Treg.T.pc10_filter<- subset(Treg.annota, subset = type %in% c("First arthritis", "Second arthritis"))  # for group selection 

Treg.split<- DotPlot(Treg.T.pc10_filter, 
        features = c("IFNG", "TNF", "IL1B", "IL17", "IL10", "TGFB1", "CTLA4", "PDCD1","IL2RA", "IKZF2","FOXP3"), 
        cols = c("blue", "red"), 
        dot.scale = 10,
        split.by = "type") + 
  theme_classic() +
  RotatedAxis() + 
  coord_flip() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=15), 
        axis.title.y= element_blank(), 
        axis.text = element_text(size=22, face = "bold"), 
        axis.text.x = element_blank(), 
        axis.title.x = element_blank(),
        legend.title = element_text(size= 16, face = "bold"), 
        legend.text = element_text(size= 14, face = "bold"), 
        legend.position = "top")
Treg.split
``` 
# save high quality plot! 
``` {r}
ggsave("dotplot_Treg_clusternw.pdf", 
       plot = Treg.split, 
       height=4, 
       width = 7,  
       units = "in", dpi = 300)

``` 

#legend 
``` {r}
ggsave("legend_IFNG_TNF.pdf", 
       plot = Treg.split, 
       height=4, 
       width = 8,  
       units = "in", dpi = 300)
```


``` {r}
Treg.split_regulatory<- DotPlot(Treg.T.pc10_filter, 
        features = c("IL10", "TGFB1"), 
        cols = c("blue", "red"), 
        dot.scale = 6, 
        split.by = "type") + 
  theme_classic() +
  RotatedAxis() + 
  coord_flip() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=15), 
        axis.title.y= element_blank(), 
        axis.text = element_text(size=18, face = "bold"), 
        axis.text.x = element_blank(), 
        axis.title.x = element_blank(),
        legend.title = element_text(size= 8, face = "bold"), 
        legend.text = element_text(size= 6, face = "bold"), 
        legend.position = "none")
Treg.split_regulatory
``` 

# save high quality plot! 
``` {r}
ggsave("dotplot_Treg_IL10_TGFB.pdf", 
       plot = Treg.split_regulatory, 
       height=4, 
       width = 5,  
       units = "in", dpi = 300)
``` 

#legend 
``` {r}
ggsave("legend_IL10_TGFB.pdf", 
       plot = Treg.split_regulatory, 
       height=4, 
       width = 8,  
       units = "in", dpi = 300)
``` 

``` {r}
VlnPlot(object = Treg.T.pc10, features = 'TNF', split.by = 'type')

``` 



# gene expression and gene set enrichment1 of Treg clusters! 

``` {r}
# Identify differentially expressed genes across conditions
view (Treg.annota@meta.data)
Idents(Treg.annota)
Treg.annota$celltypes<-Idents(Treg.annota)
view (Treg.annota@meta.data)
Treg.annota$arthritis_celltype <- paste(Treg.annota$type, sep = "_", Idents(Treg.annota))
view (Treg.annota@meta.data)
Idents(Treg.annota) <- "arthritis_celltype"
Idents(Treg.annota)

#Calculate differentially expressed genes between first and second arthritis! 
prepsct<- PrepSCTFindMarkers(Treg.annota)
nTreg_deg<-  FindMarkers(prepsct,
                                assay = "SCT", 
                                ident.1 = "Second arthritis_nTreg", 
                               ident.2 = "First arthritis_nTreg", 
                    min.pct = 0.25, 
                    logfc.threshold = 0.25, 
                    verbose = FALSE)
head(nTreg_deg, n = 20)


# order it from highest log2FC to lowest
nTregicivsosteo_order<- nTreg_deg %>% arrange(desc(avg_log2FC))
head(nTregicivsosteo_order)


nTregicivsosteo_order<- nTregicivsosteo_order[nTregicivsosteo_order$p_val_adj<0.05, ]

# Save deg of cluster 1
write.csv(nTregicivsosteo_order, file = "deg.nTreg.csv")
``` 

# volcano plot for nTreg
# new volcano plot for CXCL13 

``` {r}
# Loading relevant libraries 
library(tidyverse) # includes ggplot2, for data visualisation. dplyr, for data manipulation.
library(RColorBrewer) # for a colourful plot
library(ggrepel) # for nice annotations

# Set input path
path <- "C:/Users/Documents/Biostatsquid/Volcano_plots/Tutorial/"  # I can set later
setwd(path)

# Import DGE results
deg.first.vs.second.nTreg<- read.csv("deg_between conditions/deg_second_vs_first_nTreg.csv")
head(deg.first.vs.second.nTreg)
``` 

``` {r}

deg.first.vs.second.nTreg$diffexpressed<- "NO"
deg.first.vs.second.nTreg$diffexpressed[deg.first.vs.second.nTreg$avg_log2FC > 0.5 & deg.first.vs.second.nTreg$p_val_adj < 0.05] <- "UP"
deg.first.vs.second.nTreg$diffexpressed[deg.first.vs.second.nTreg$avg_log2FC < -0.5 & deg.first.vs.second.nTreg$p_val_adj < 0.05] <- "DOWN"
head(deg.first.vs.second.nTreg[order(deg.first.vs.second.nTreg$p_val_adj) & deg.first.vs.second.nTreg$diffexpressed == 'DOWN', ])
``` 

# Create a new column "delabel" to de, that will contain the name of the top 30 differentially expressed genes (NA in case they are not)
``` {r}
deg.first.vs.second.nTreg$delabel <- ifelse(deg.first.vs.second.nTreg$gene_symbol %in% head(deg.first.vs.second.nTreg[order(deg.first.vs.second.nTreg$p_val_adj), "gene_symbol"], 20), deg.first.vs.second.nTreg$gene_symbol, NA)
``` 



``` {r}
volcano_nTreg_secondvsfirst<- ggplot(data = deg.first.vs.second.nTreg,
                                      aes(x = avg_log2FC, y = -log10(p_val_adj), 
                                                                   col = diffexpressed, 
                                          label= delabel)) +
  geom_vline(xintercept = c(-0.5, 0.5), col = "gray", linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), col = "gray", linetype = 'dashed') +
  geom_point(size = 2) +
  theme(plot.title = element_text(hjust = 0.5, size = 11, face = "bold"), 
        axis.title = element_text(face = "bold", size = 12), 
        axis.text = element_text(face = "bold", size=11), 
        legend.title = element_blank(), 
        legend.text = element_text(size = 6), 
        legend.position = c(0.17, 0.87)) +
  scale_color_manual(values = c("#00AFBB", "grey", "#bb0c00"), 
                     labels = c("Downregulated", "Not significant", "Upregulated"))  + 
 ggtitle("Second vs. first arthritis (nTreg)") + 
   geom_text_repel(max.overlaps = Inf, size=2.3) # To show all labels 
volcano_nTreg_secondvsfirst
``` 

# save high quality plot! 
``` {r}
ggsave("volcano_secondvsfirst_nTreg.pdf", plot = volcano_nTreg_secondvsfirst, height=4, width = 4.9, units = "in", dpi = 300)
```

#deg plot for volcano! 
``` {r}
## Volcano plot @ 1 day 

volcano.nTreg<- nTreg_deg[order(nTreg_deg$p_val_adj),]

# create data.frame
results.nTreg<- as.data.frame(mutate(as.data.frame(volcano.nTreg), 
                                    cut_off=ifelse(volcano.nTreg$p_val_adj<0.05, "significant", "insignificant")), 
                             row.names=rownames(volcano.nTreg))
## ggplot plus ggrepel
options(ggrepel.max.overlaps = Inf)
library(ggrepel)
volcano.nTreg$cut_off<- factor(results.nTreg$cut_off, levels = c('significant', 'insignificant'))

# plot
volcano_nTreg.ggplot<- ggplot(results.nTreg, aes(avg_log2FC, -log10(p_val_adj))) + 
  theme_classic() +
  geom_point(aes(col = cut_off)) +
  scale_color_manual(values = c("black", "red")) +
  theme(plot.title = element_text(hjust = 0.5, size = 8, face = "bold"), 
        axis.title = element_text(face = "bold", size = 12), 
        axis.text = element_text(face = "bold", size=8), 
        legend.title = element_text(size = 8, face = "bold"), 
        legend.text = element_text(size = 6), 
        legend.position = c(0.2, 0.87)) + 
  geom_text_repel(data=results.nTreg[1:20,], 
                  aes(label=rownames(results.nTreg[1:20,])), size= 2) +
  ggtitle("Second vs. First arthritis (nTreg)") 
volcano_nTreg.ggplot
``` 

# save high quality plot! 
``` {r}
ggsave("volcano.nTreg.pdf", plot = volcano_nTreg.ggplot, height=2.8, width = 3, units = "in", dpi = 300)
```


# Perform gene set enrichment analysis. === Second vs. first arthritis! 
``` {r}
## perform gene set enrichment 

BiocManager::install("fgsea")
library(fgsea)
nTreg_deg$gene<- rownames(nTreg_deg)
nTreg_deg<- nTreg_deg %>% arrange(desc(avg_log2FC))

#write csv file for deg
write.csv(DEG.activatedCD4, file = "Y:/Recurrent__Arthritis__Single cell/DEG.activatedCD4.csv")

fold_changes<- nTreg_deg$avg_log2FC
names(fold_changes)<- nTreg_deg$gene
``` 

#Load geneset ! 
``` {r}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("fgsea")
Reactome <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.reactome.v2023.1.Hs.symbols.gmt")
hallmark <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/h.all.v2023.1.Hs.symbols.gmt")
KEGG <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.kegg.v2023.1.Hs.symbols.gmt")
GO <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c5.go.bp.v2023.1.Hs.symbols.gmt")

#combine the four gsea
Four_gene_sets <- c(hallmark, KEGG, GO, Reactome)

#getset for all pathway 
Allgenesets <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/msigdb.v2023.1.Hs.symbols.gmt")

# GSEA analysis
gsea_Myo_nTreg<- fgsea(pathways = hallmark,
                  stats = fold_changes,
                  eps = 0.0,
                  minSize=15,
                  maxSize=500)
```

# plot enrichment 
``` {r}
library(tidyverse)

## 3 hrs
gseanTreg<- read.csv("gsea.nTreg.csv", stringsAsFactors = TRUE, header = TRUE)

# ggplot
gseantreg.plots<- ggplot(gseanTreg, aes(reorder(Pathway, NES), NES, fill= padj)) + 
  theme_classic() +
  geom_bar(stat = "identity")  + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 8),
        axis.title.x = element_text(face = "bold", size = 12),
        axis.title.y  = element_blank(), 
        axis.text = element_text(face = "bold", size = 10), 
        plot.title = element_text(hjust = 0.5, face = "bold", size = 10)) + 
  labs(x="Hallmark pathways", y="normalized enrichment score",
       title="Second vs. first arthritis (nTreg)") + 
  coord_flip() + 
  scale_fill_gradient(low = "red", high = "blue")
gseantreg.plots
``` 

``` {r}
ggsave("gseanTreg.pdf", plot = gseantreg.plots, height=3, width =6.5, units = "in", dpi = 300)
``` 

# iTreg !
``` {r}
iTreg_deg<-  FindMarkers(prepsct,
                                assay = "SCT", 
                                ident.1 = "Second arthritis_iTreg", 
                               ident.2 = "First arthritis_iTreg", 
                    min.pct = 0.25, 
                    logfc.threshold = 0.25, 
                    verbose = FALSE)
head(iTreg_deg, n = 20)


# order it from highest log2FC to lowest
iTregicivsosteo_order<- iTreg_deg %>% arrange(desc(avg_log2FC))
head(nTregicivsosteo_order)


iTregicivsosteo_order<- iTregicivsosteo_order[iTregicivsosteo_order$p_val_adj<0.05, ]

# Save deg of cluster 1
write.csv(iTregicivsosteo_order, file = "deg.iTreg.csv")
``` 

#volcano plot for iTreg! 
``` {r}
# Loading relevant libraries 
library(tidyverse) # includes ggplot2, for data visualisation. dplyr, for data manipulation.
library(RColorBrewer) # for a colourful plot
library(ggrepel) # for nice annotations

# Set input path
path <- "C:/Users/Documents/Biostatsquid/Volcano_plots/Tutorial/"  # I can set later
setwd(path)

# Import DGE results
deg.first.vs.second.iTreg<- read.csv("deg_between conditions/deg_second_vs_first_iTreg.csv")
head(deg.first.vs.second.nTreg)
``` 

``` {r}

deg.first.vs.second.iTreg$diffexpressed<- "NO"
deg.first.vs.second.iTreg$diffexpressed[deg.first.vs.second.iTreg$avg_log2FC > 0.5 & deg.first.vs.second.iTreg$p_val_adj < 0.05] <- "UP"
deg.first.vs.second.iTreg$diffexpressed[deg.first.vs.second.iTreg$avg_log2FC < -0.5 & deg.first.vs.second.iTreg$p_val_adj < 0.05] <- "DOWN"
head(deg.first.vs.second.iTreg[order(deg.first.vs.second.iTreg$p_val_adj) & deg.first.vs.second.iTreg$diffexpressed == 'DOWN', ])
``` 

# Create a new column "delabel" to de, that will contain the name of the top 30 differentially expressed genes (NA in case they are not)
``` {r}
deg.first.vs.second.iTreg$delabel <- ifelse(deg.first.vs.second.iTreg$gene_symbol %in% head(deg.first.vs.second.iTreg[order(deg.first.vs.second.iTreg$p_val_adj), "gene_symbol"], 20), deg.first.vs.second.iTreg$gene_symbol, NA)
``` 



``` {r}
volcano_iTreg_secondvsfirst<- ggplot(data = deg.first.vs.second.iTreg,
                                      aes(x = avg_log2FC, y = -log10(p_val_adj), 
                                                                   col = diffexpressed, 
                                          label= delabel)) +
  geom_vline(xintercept = c(-0.5, 0.5), col = "gray", linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), col = "gray", linetype = 'dashed') +
  geom_point(size = 2) +
  theme(plot.title = element_text(hjust = 0.5, size = 11, face = "bold"), 
        axis.title = element_text(face = "bold", size = 12), 
        axis.text = element_text(face = "bold", size=11), 
        legend.title = element_blank(), 
        legend.text = element_text(size = 6), 
        legend.position = c(0.87, 0.9)) +
  scale_color_manual(values = c("#00AFBB", "grey", "#bb0c00"), 
                     labels = c("Downregulated", "Not significant", "Upregulated"))  + 
 ggtitle("Second vs. first arthritis (iTreg)") + 
   geom_text_repel(max.overlaps = Inf, size=2.3) # To show all labels 
volcano_iTreg_secondvsfirst
``` 
# save high quality plot! 
``` {r}
ggsave("volcano_secondvsfirst_iTreg.pdf", plot = volcano_iTreg_secondvsfirst, height=4, width = 4.9, units = "in", dpi = 300)
```


#deg plot for volcano! 
``` {r}
## Volcano plot @ 1 day 

volcano.iTreg<- iTreg_deg[order(iTreg_deg$p_val_adj),]

# create data.frame
results.iTreg<- as.data.frame(mutate(as.data.frame(volcano.iTreg), 
                                    cut_off=ifelse(volcano.iTreg$p_val_adj<0.05, "significant", "insignificant")), 
                             row.names=rownames(volcano.iTreg))
## ggplot plus ggrepel
options(ggrepel.max.overlaps = Inf)
library(ggrepel)
volcano.iTreg$cut_off<- factor(results.iTreg$cut_off, levels = c('significant', 'insignificant'))

# plot
volcano_iTreg.ggplot<- ggplot(results.iTreg, aes(avg_log2FC, -log10(p_val_adj))) + 
  theme_classic() +
  geom_point(aes(col = cut_off)) +
  scale_color_manual(values = c("black", "red")) +
  theme(plot.title = element_text(hjust = 0.5, size = 8, face = "bold"), 
        axis.title = element_text(face = "bold", size = 12), 
        axis.text = element_text(face = "bold", size=8), 
        legend.title = element_text(size = 8, face = "bold"), 
        legend.text = element_text(size = 6), 
        legend.position = c(0.2, 0.87)) + 
  geom_text_repel(data=results.iTreg[1:30,], 
                  aes(label=rownames(results.iTreg[1:30,])), size= 2) +
  ggtitle("Second vs. First arthritis (iTreg)") 
volcano_iTreg.ggplot
``` 

# save high quality plot! 
``` {r}
ggsave("volcano.iTreg.pdf",
       plot = volcano_iTreg.ggplot, 
       height=2.8, 
       width = 3.25, 
       units = "in", 
       dpi = 300)
```

# Perform gene set enrichment analysis. === Second vs. first arthritis! 
``` {r}
## perform gene set enrichment 

BiocManager::install("fgsea")
library(fgsea)
iTreg_deg$gene<- rownames(iTreg_deg)
iTreg_deg<- iTreg_deg %>% arrange(desc(avg_log2FC))

#write csv file for deg
write.csv(DEG.activatedCD4, file = "Y:/Recurrent__Arthritis__Single cell/DEG.activatedCD4.csv")

fold_changes<- iTreg_deg$avg_log2FC
names(fold_changes)<- iTreg_deg$gene
``` 

#Load geneset ! 
``` {r}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("fgsea")
Reactome <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.reactome.v2023.1.Hs.symbols.gmt")
hallmark <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/h.all.v2023.1.Hs.symbols.gmt")
KEGG <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.kegg.v2023.1.Hs.symbols.gmt")
GO <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c5.go.bp.v2023.1.Hs.symbols.gmt")

#combine the four gsea
Four_gene_sets <- c(hallmark, KEGG, GO, Reactome)

#getset for all pathway 
Allgenesets <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/msigdb.v2023.1.Hs.symbols.gmt")

# GSEA analysis
gsea_Myo_iTreg<- fgsea(pathways = hallmark,
                  stats = fold_changes,
                  eps = 0.0,
                  minSize=15,
                  maxSize=500)
```


# plot enrichment 
``` {r}
library(tidyverse)

## 3 hrs
gseaniTreg<- read.csv("gsea.iTreg.csv", stringsAsFactors = TRUE, header = TRUE)

# ggplot
gseaitreg.plots<- ggplot(gseaniTreg, aes(reorder(Pathway, NES), NES, fill= padj)) + 
  theme_classic() +
  geom_bar(stat = "identity")  + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 8),
        axis.title.x = element_text(face = "bold", size = 12),
        axis.title.y  = element_blank(), 
        axis.text = element_text(face = "bold", size = 10), 
        plot.title = element_text(hjust = 0.5, face = "bold", size = 10)) + 
  labs(x="Hallmark pathways", y="normalized enrichment score",
       title="Second vs. first arthritis (iTreg)") + 
  coord_flip() + 
  scale_fill_gradient(low = "green", high = "blue")
gseaitreg.plots
``` 

``` {r}
ggsave("gseaiTreg.pdf", plot = gseaitreg.plots, height=3, width =6.5, units = "in", dpi = 300)
``` 

# Analysis of only one Treg population !! 
``` {r}
Treg.obj<- readRDS("Treg.obj.annotated_one.rds")
onetreg<- DimPlot(Treg.obj, label = FALSE) + 
  NoLegend() + 
  theme_minimal() + 
  theme(axis.text = element_text(size= 18, face = "bold"), 
        axis.title = element_text(size= 18, face = "bold"))
onetreg
``` 
# save this plots 
``` {r}
ggsave("seurat.plot.pdf", plot = onetreg, height=3.5, width = 5,  units = "in", dpi = 300)
``` 



# dot plots between first and second arthritis of various markers !! 
``` {r}
# subset only first and second arthritis ! 
Treg.T.pc10_filter<- subset(Treg.obj, subset = type %in% c("First arthritis", "Second arthritis"))  # for group selection 

Treg.split<- DotPlot(Treg.T.pc10_filter, 
        features = c("IFNG", "TNF", "IL1B", "IL17", "IL10", "TGFB1", "CTLA4", "PDCD1","IL2RA", "IKZF2","FOXP3"), 
        cols = c("blue", "red"), 
        dot.scale = 10,
        split.by = "type") + 
  theme_classic() +
  RotatedAxis() + 
  coord_flip() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=15), 
        axis.title.y= element_blank(), 
        axis.text = element_text(size=22, face = "bold"), 
        axis.text.x = element_text(), 
        axis.title.x = element_text(),
        legend.title = element_text(size= 16, face = "bold"), 
        legend.text = element_text(size= 14, face = "bold"), 
        legend.position = "right")
Treg.split
``` 

``` {r}
#feature dotplot !! 
features<- c( "CTLA4", "PDCD1", "TIGIT","LAG3","IL2RA", "ICOS","KLRG1", 
             "IFNG", "TNF", "IL1B", "IL17", "IL10", "TGFB1", "IKZF2", "FOXP3","CD4")

#plot
dotplot<- DotPlot(Treg.T.pc10_filter,
                  features = features, 
                  dot.scale = 6, 
                  cols = "RdBu",
                  split.by = "type") +
  geom_point(aes(size=pct.exp), shape = 21, stroke=0.5) +
  RotatedAxis() + 
  coord_flip() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=15), 
        axis.title.y= element_blank(), 
        axis.text = element_text(size=15, face = "bold"), 
        axis.text.x = element_blank(), 
        axis.title.x = element_blank(),
        legend.title = element_text(size= 14, face = "bold"), 
        legend.text = element_text(size= 12, face = "bold")) 

dotplot
``` 


# save this plots 
``` {r}
ggsave("allTreg.pdf", plot = dotplot, height=3.5, width = 5,  units = "in", dpi = 300)
``` 

``` {r}
#Differentially gene expression! 
# Identify differentially expressed genes across conditions
view (Treg.obj@meta.data)
Idents(Treg.obj)
Treg.obj$celltypes<-Idents(Treg.obj)
view (Treg.obj@meta.data)
Treg.obj$arthritis_celltype <- paste(Treg.obj$type, sep = "_", Idents(Treg.obj))
view (Treg.obj@meta.data)
Idents(Treg.obj) <- "arthritis_celltype"
Idents(Treg.obj)
``` 

# Run differentially expressed genes
``` {r}
#Calculate differentially expressed genes between first and second arthritis! 
prepsct<- PrepSCTFindMarkers(Treg.obj)
Tregsecondvsfirst<-  FindMarkers(prepsct,assay = "SCT", ident.1 = "Second arthritis_Treg", 
                                   ident.2 = "First arthritis_Treg", 
                                   min.pct = 0.25, 
                                   logfc.threshold = 0.25, 
                                   verbose = FALSE, 
                                   recorrect_umi = FALSE)
head(Tregsecondvsfirst, n = 20)
Tregorder<- Tregsecondvsfirst %>% arrange(desc(avg_log2FC))
head(Tregorder)

Tregorder<-Tregorder[Tregorder$p_val_adj<0.05, ]

# Save deg of cluster 1
write.csv(Tregorder, file = "deg_tregall.csv")







# Order the Th1 data frame by adjusted p-value
volcano_Treg <- Tregsecondvsfirst[order(Tregsecondvsfirst$p_val_adj),]

# Create data frame with cut_off based on significance and avg_log2FC thresholds
results_Treg <- as.data.frame(mutate(as.data.frame(volcano_Treg), 
                                    cut_off = case_when(
                                      p_val_adj < 0.05 & avg_log2FC < -0.5 ~ "Downregulated",
                                      p_val_adj < 0.05 & avg_log2FC > 0.5 ~ "Upregulated",
                                      TRUE ~ "Not significant")), 
                             row.names = rownames(volcano_Treg))

# Ensure the cut_off column is a factor with the correct levels
results_Treg$cut_off <- factor(results_Treg$cut_off, levels = c('Downregulated', 'Upregulated', 'Not significant'))

# Load necessary library
options(ggrepel.max.overlaps = Inf)
library(ggrepel)
library(ggrepel)
genes_to_label <- c("TNF", "IFNG", "JUN", "CD69", "KLF6", "JUNB", "IER2", "NFKB1", "DNAJA1", "IRF4", "DNAJB1", "CCNL1",
                    "CXCR6", "MT1E", "AC131971.1", "TSC22D3", "MT2A")  # Replace with your gene names

# Plot
volcano_Treg_plot <- ggplot(results_Treg, aes(avg_log2FC, -log10(p_val_adj))) + 
  theme_classic() +
  geom_point(aes(col = cut_off), size= 2) +
  scale_color_manual(values = c("#00AFBB","#bb0c00", "grey")) +
  geom_vline(xintercept = c(-0.5, 0.5), col = "black", linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), col = "black", linetype = 'dashed') +
  theme(plot.title = element_blank(), 
        axis.title = element_text(face = "bold", size = 14), 
        axis.text = element_text(face = "bold", size = 13), 
        legend.title = element_blank(), 
        legend.text = element_text(size = 6.5), 
        legend.position = c(0.32, 0.88)) + 
  geom_text_repel(data = results_Treg[1:25,], 
                  aes(label = rownames(results_Treg[1:25,])), size = 1.8) + 
  ggtitle("Second vs. first arthritis (PD1hi CXCL13hi CD4)")
volcano_Treg_plot


#geom_text_repel(data = subset(results_Treg, rownames(results_Treg) %in% genes_to_label), 
 #                 aes(label = rownames(subset(results_cxcl13cd4, rownames(results_Treg) %in% genes_to_label))), size = 2.3)

``` 
``` {r}
ggsave("volcano.Tregall.pdf", plot = volcano_Treg_plot, height=3, width = 3.2, units = "in", dpi = 300)
``` 


# plot cluster frequency of Treg !! 
``` {r}
Treg_frequency<- read.csv("cluster_frequency/cluster_frequency.csv", stringsAsFactors = TRUE)
Treg_frequency<- Treg_frequency %>% 
  filter(Arthritis %in% c("first arthritis", "second arthritis")) %>% 
  filter(cluster_name == "Treg")
``` 
#plot cluster frequency 
``` {r}
Tregfreq<- ggplot(Treg_frequency, aes(x= Arthritis, y=frequency, fill= Arthritis)) + 
  theme_classic() +
  geom_violin(trim = FALSE) +
  geom_point(size= 5, aes(x=Arthritis), shape=21, position = position_dodge(width = 0)) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=14
  ), axis.title.y = element_text(size= 20, face = "bold"), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=16, face = "bold"), 
  axis.text.x = element_blank()) + 
  theme(legend.position = "none") + 
  ylab("% of total cells") + 
  ggtitle("Regulatory T cell")
Tregfreq


``` 
``` {r}
ggsave("Treg_proportion.pdf", plot = Tregfreq, height=3.5, width =4, units = "in", dpi = 300) 
``` 

# Run Treg gene set enrichment !!! 

``` {r}
BiocManager::install("fgsea")
library(fgsea)
Tregsecondvsfirst$gene<- rownames(Tregsecondvsfirst)
Tregsecondvsfirst<- Tregsecondvsfirst %>% arrange(desc(avg_log2FC))

#write csv file for deg
write.csv(DEG.activatedCD4, file = "Y:/Recurrent__Arthritis__Single cell/DEG.activatedCD4.csv")

fold_changes<- Tregsecondvsfirst$avg_log2FC
names(fold_changes)<- Tregsecondvsfirst$gene
 

#Load geneset ! 

if (!require("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install("fgsea")
Reactome <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.reactome.v2023.1.Hs.symbols.gmt")
hallmark <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/h.all.v2023.1.Hs.symbols.gmt")
KEGG <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.kegg.v2023.1.Hs.symbols.gmt")
GO <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c5.go.bp.v2023.1.Hs.symbols.gmt")

#combine the four gsea
Four_gene_sets <- c(hallmark, KEGG, GO, Reactome)

#getset for all pathway 
Allgenesets <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/msigdb.v2023.1.Hs.symbols.gmt")

# GSEA analysis
gsea_Myo_Treg<- fgsea(pathways = hallmark,
                       stats = fold_changes,
                       eps = 0.0,
                       minSize=15,
                       maxSize=500)

# make a table plot for a bunch of selected pathways:
topPathwaysUp <- gsea_Myo_nTreg[ES > 0][head(order(pval), n=10), pathway]
topPathwaysDown <- gsea_Myo_nTreg[ES < 0][head(order(pval), n=10), pathway]
topPathways <- c(topPathwaysUp, rev(topPathwaysDown))
plotGseaTable(hallmark [topPathways], fold_changes, gsea_Myo_nTreg, 
              gseaParam=0.5)

# make an enrichment plot for a pathway:
TNF<- plotEnrichment(hallmark[["HALLMARK_TNFA_SIGNALING_VIA_NFKB"]],
               fold_changes) + labs(title="TNFA_SIGNALING_VIA_NFKB") + 
  theme_classic()



IL2<- plotEnrichment(hallmark[["HALLMARK_IL2_STAT5_SIGNALING"]],
               fold_changes) + labs(title="IL2_STAT5_SIGNALING") + 
  theme_classic()


IFNG<- plotEnrichment(hallmark[["HALLMARK_INTERFERON_GAMMA_RESPONSE"]],
               fold_changes) + labs(title="INTERFERON_GAMMA_RESPONSE") + 
  theme_classic()


#Patchwork !! 
library(patchwork)
nTreg<- TNF + IL2 + IFNG

ggsave("gase_Tregall.pdf", plot = nTreg, height=3, width =6.5, units = "in", dpi = 300)
``` 

# Analysis of only one Treg population !! 
``` {r}
library(Seurat)
Treg.obj<- readRDS("Treg.obj.annotated_one.rds")

#subset the first and second IA !!! 
Treg.obj<- subset(Treg.obj, subset = type %in% c("First arthritis","Second arthritis")) 

Tregone<- DimPlot(Treg.obj, label = FALSE, 
                  group.by = "type") + 
  theme_classic()  +
    theme(plot.title = element_text(hjust = 0.5, face = "bold"),
        axis.text = element_text(face = "bold", size= 16), 
        axis.title = element_text(face = "bold", size= 18)) 
Tregone
``` 
#save for iotox 
``` {r}
ggsave("Tregkkk.pdf", plot = Tregone, height=3, width = 4, units = "in", dpi = 300)
ggsave("Tregkkklegend.pdf", plot = Tregone, height=3, width = 4, units = "in", dpi = 300)
``` 


# dot plots between first and second arthritis of various markers !! 
``` {r}
# subset only first and second arthritis ! 
Treg.T.pc10_filter<- subset(Treg.obj, subset = type %in% c("First arthritis", "Second arthritis"))  # for group selection 

Treg.split<- DotPlot(Treg.T.pc10_filter, 
        features = c("IFNG", "TNF", "IL1B", "IL17", "IL10", "TGFB1", "CTLA4", "PDCD1","IL2RA", "IKZF2","FOXP3"), 
        cols = c("blue", "red"), 
        dot.scale = 10,
        split.by = "type") + 
  theme_classic() +
  RotatedAxis() + 
  coord_flip() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=15), 
        axis.title.y= element_blank(), 
        axis.text = element_text(size=22, face = "bold"), 
        axis.text.x = element_text(), 
        axis.title.x = element_text(),
        legend.title = element_text(size= 16, face = "bold"), 
        legend.text = element_text(size= 14, face = "bold"), 
        legend.position = "right")
Treg.split
``` 

``` {r}
#feature dotplot !! 
features<- c( "CTLA4", "PDCD1", "TIGIT","LAG3","IL2RA", "ICOS","KLRG1", 
             "IFNG", "TNF", "IL1B", "IL17", "IL10", "TGFB1", "IKZF2", "FOXP3","CD4")

#plot
dotplot<- DotPlot(Treg.T.pc10_filter,
                  features = features, 
                  dot.scale = 6, 
                  cols = "RdBu",
                  split.by = "type") +
  geom_point(aes(size=pct.exp), shape = 21, stroke=0.5) +
  RotatedAxis() + 
  coord_flip() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=15), 
        axis.title.y= element_blank(), 
        axis.text = element_text(size=15, face = "bold"), 
        axis.text.x = element_blank(), 
        axis.title.x = element_blank(),
        legend.title = element_text(size= 14, face = "bold"), 
        legend.text = element_text(size= 12, face = "bold"))

dotplot

#plot
dotplot<- DotPlot(Treg.T.pc10_filter, 
                         features = features, 
                         cols = c("blue", "red"),
                         dot.scale = 6, 
                         group.by = "type") + 
  theme_classic() +
  RotatedAxis() + 
  coord_flip() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=15), 
        axis.title.y= element_blank(), 
        axis.text = element_text(size=16, face = "bold"), 
        axis.text.x = element_blank(), 
        axis.title.x = element_blank(),
        legend.title = element_text(size= 14, face = "bold"), 
        legend.text = element_text(size= 14, face = "bold")) + 
  theme(legend.position = "right")
dotplot








``` 

#iotox !! 
``` {r}
dotplot<- DotPlot(Treg.T.pc10_filter,
                  features = features, 
                  dot.scale = 6, 
                  cols = "RdBu",
                  split.by = "type") +
  geom_point(aes(size=pct.exp), shape = 21, stroke=0.5) +
  RotatedAxis() + 
  coord_flip() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=15), 
        axis.title.y= element_blank(), 
        axis.text = element_text(size=15, face = "bold"), 
        axis.text.x = element_blank(), 
        axis.title.x = element_blank(),
        legend.title = element_text(size= 14, face = "bold"), 
        legend.text = element_text(size= 12, face = "bold")) 
dotplot
```

# save this plots 
``` {r}
ggsave("allTreg.pdf", plot = dotplot, height=3.5, width = 6,  units = "in", dpi = 300)

#IOTOX !! 
ggsave("dotplot_treg.tiff", plot = dotplot, height=3, width = 6,  units = "in", dpi = 300)
``` 

``` {r}
#Differentially gene expression! 
# Identify differentially expressed genes across conditions
view (Treg.obj@meta.data)
Idents(Treg.obj)
Treg.obj$celltypes<-Idents(Treg.obj)
view (Treg.obj@meta.data)
Treg.obj$arthritis_celltype <- paste(Treg.obj$type, sep = "_", Idents(Treg.obj))
view (Treg.obj@meta.data)
Idents(Treg.obj) <- "arthritis_celltype"
Idents(Treg.obj)
``` 

# Run differentially expressed genes
``` {r}
#Calculate differentially expressed genes between first and second arthritis! 
prepsct<- PrepSCTFindMarkers(Treg.obj)
Tregsecondvsfirst<-  FindMarkers(prepsct,assay = "SCT", ident.1 = "Second arthritis_Treg", 
                                   ident.2 = "First arthritis_Treg", 
                                   min.pct = 0.25, 
                                   logfc.threshold = 0.25, 
                                   verbose = FALSE, 
                                   recorrect_umi = FALSE)
head(Tregsecondvsfirst, n = 20)
Tregorder<- Tregsecondvsfirst %>% arrange(desc(avg_log2FC))
head(Tregorder)

Tregorder<-Tregorder[Tregorder$p_val_adj<0.05, ]

# Save deg of cluster 1
write.csv(Tregorder, file = "deg_tregall.csv")


# Order the Th1 data frame by adjusted p-value
volcano_Treg <- Tregsecondvsfirst[order(Tregsecondvsfirst$p_val_adj),]

# Create data frame with cut_off based on significance and avg_log2FC thresholds
results_Treg <- as.data.frame(mutate(as.data.frame(volcano_Treg), 
                                    cut_off = case_when(
                                      p_val_adj < 0.05 & avg_log2FC < -0.5 ~ "Downregulated",
                                      p_val_adj < 0.05 & avg_log2FC > 0.5 ~ "Upregulated",
                                      TRUE ~ "Not significant")), 
                             row.names = rownames(volcano_Treg))

# Ensure the cut_off column is a factor with the correct levels
results_Treg$cut_off <- factor(results_Treg$cut_off, levels = c('Downregulated', 'Upregulated', 'Not significant'))

# Load necessary library
options(ggrepel.max.overlaps = Inf)
library(ggrepel)
library(ggrepel)
genes_to_label <- c("TNF", "IFNG", "JUN", "CD69", "KLF6", "JUNB", "IER2", "NFKB1", "DNAJA1", "IRF4", "DNAJB1", "CCNL1",
                    "CXCR6", "MT1E", "AC131971.1", "TSC22D3", "MT2A")  # Replace with your gene names

# Plot
volcano_Treg_plot <- ggplot(results_Treg, aes(avg_log2FC, -log10(p_val_adj))) + 
  theme_classic() +
  geom_point(aes(col = cut_off), size= 2) +
  scale_color_manual(values = c("#00AFBB","#bb0c00", "grey")) +
  geom_vline(xintercept = c(-0.5, 0.5), col = "black", linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), col = "black", linetype = 'dashed') +
  theme(plot.title = element_blank(), 
        axis.title = element_text(face = "bold", size = 14), 
        axis.text = element_text(face = "bold", size = 13), 
        legend.title = element_blank(), 
        legend.text = element_blank()) + 
  geom_text_repel(data = results_Treg[1:25,], 
                  aes(label = rownames(results_Treg[1:25,])), size = 2) + 
   theme(legend.position = "none") +
  ggtitle("Second vs. first arthritis (PD1hi CXCL13hi CD4)")
volcano_Treg_plot


#geom_text_repel(data = subset(results_Treg, rownames(results_Treg) %in% genes_to_label), 
 #                 aes(label = rownames(subset(results_cxcl13cd4, rownames(results_Treg) %in% genes_to_label))), size = 2.3)


#IOTOX conference 
volcano_Treg_plot <- ggplot(results_Treg, aes(avg_log2FC, -log10(p_val_adj))) + 
  theme_minimal() +
  geom_point(aes(col = cut_off), size= 2) +
  scale_color_manual(values = c("#00AFBB","#bb0c00", "grey")) +
  geom_vline(xintercept = c(-0.5, 0.5), col = "black", linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), col = "black", linetype = 'dashed') +
  theme(plot.title = element_blank(), 
        axis.title = element_text(face = "bold", size = 18), 
        axis.text = element_text(face = "bold", size = 13), 
        legend.title = element_blank(), 
        legend.text = element_blank(), 
        panel.grid.major = element_line(size = 1), 
        panel.grid.minor = element_line(size= 1)) + 
  geom_text_repel(data = results_Treg[1:25,], 
                  aes(label = rownames(results_Treg[1:25,])), size = 5) + 
  ggtitle("Second vs. first arthritis (PD1hi CXCL13hi CD4)") + 
  theme(legend.position = "none")
volcano_Treg_plot

ggsave("volcano_secondvsfirst_Tregone.tiff", plot = volcano_Treg_plot, height=4, width = 4.9, units = "in", dpi = 300)





``` 
``` {r}
ggsave("volcano.Tregall.pdf", plot = volcano_Treg_plot, height=3, width = 4, units = "in", dpi = 300)
``` 

#run enrichment analysis !! 
``` {r}
library(fgsea)
Tregsecondvsfirst$gene<- rownames(Tregsecondvsfirst)
Tregsecondvsfirst<- Tregsecondvsfirst %>% arrange(desc(avg_log2FC))

#write csv file for deg
write.csv(DEG.activatedCD4, file = "Y:/Recurrent__Arthritis__Single cell/DEG.activatedCD4.csv")

fold_changes<- Tregsecondvsfirst$avg_log2FC
names(fold_changes)<- Tregsecondvsfirst$gene

#pathway genes
Reactome <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.reactome.v2023.1.Hs.symbols.gmt")
hallmark <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/h.all.v2023.1.Hs.symbols.gmt")
KEGG <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c2.cp.kegg.v2023.1.Hs.symbols.gmt")
GO <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/c5.go.bp.v2023.1.Hs.symbols.gmt")

#combine the four gsea
Four_gene_sets <- c(hallmark, KEGG, GO, Reactome)

#getset for all pathway 
Allgenesets <- fgsea::gmtPathways("Y:/Recurrent__Arthritis__Single cell/Single_Cell_Arthritis/msigdb.v2023.1.Hs.symbols.gmt")

# GSEA analysis
gsea_Myo_treg <- fgsea(pathways = hallmark,
                              stats = fold_changes,
                              eps = 0.0,
                              minSize=15,
                              maxSize=500)

head(gsea_Myo_treg[order(pval), ]) 



# make a table plot for a bunch of selected pathways:
topPathwaysUp <- gsea_Myo_treg [ES > 0][head(order(pval), n=10), pathway]
topPathwaysDown <- gsea_Myo_treg [ES < 0][head(order(pval), n=10), pathway]
topPathways <- c(topPathwaysUp, rev(topPathwaysDown))
plotGseaTable(hallmark [topPathways], fold_changes, gsea_Myo_treg , 
              gseaParam=0.5)

# make an enrichment plot for a pathway:
TNF<- plotEnrichment(hallmark[["HALLMARK_TNFA_SIGNALING_VIA_NFKB"]],
                     fold_changes) + 
  theme_classic()
TNF

IL2<- plotEnrichment(hallmark[["HALLMARK_IL2_STAT5_SIGNALING"]],
                     fold_changes) +  
  theme_classic()
IL2

IFNG<- plotEnrichment(hallmark[["HALLMARK_INTERFERON_GAMMA_RESPONSE"]],
                     fold_changes) + 
  theme_classic()
IFNG

fatty<- plotEnrichment(hallmark[["HALLMARK_GLYCOLYSIS"]],
                      fold_changes)  + 
  theme_classic()
fatty

oxidative<- plotEnrichment(hallmark[["HALLMARK_OXIDATIVE_PHOSPHORYLATION"]],
                     fold_changes) +
  theme_classic()
oxidative

adipo<- plotEnrichment(hallmark[["HALLMARK_ADIPOGENESIS"]],
                     fold_changes) +
  theme_classic()
adipo

# Downregulated pathways 



#Patchwork !! 
library(patchwork)
Treg<- TNF + IL2 
  
  
 Treg<-   oxidative +  fatty + adipo

#2024 IOTOX !! 
ggsave("gase_Treg.tiff", plot = Treg, height=2.5, width =5, units = "in", dpi = 300)

ggsave("gase_Treg.pdf", plot = Treg, height=3, width =6, units = "in", dpi = 300)
``` 

# plot cluster frequency of Treg !! 
``` {r}
Treg_frequency<- read.csv("cluster_frequency/cluster_frequency.csv", stringsAsFactors = TRUE)
Treg_frequency<- Treg_frequency %>% 
  filter(Arthritis %in% c("first arthritis", "second arthritis")) %>% 
  filter(cluster_name == "Treg")
``` 
#plot cluster frequency 
``` {r}
Tregfreq<- ggplot(Treg_frequency, aes(x= Arthritis, y=frequency, fill= Arthritis)) + 
  theme_classic() +
  geom_violin(trim = FALSE) +
  geom_point(size= 5, aes(x=Arthritis), shape=21, position = position_dodge(width = 0)) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold", size=14
  ), axis.title.y = element_text(size= 20, face = "bold"), 
  axis.title.x = element_blank(),
  axis.text = element_text(size=16, face = "bold"), 
  axis.text.x = element_blank()) + 
  theme(legend.position = "none") + 
  ylab("% of total cells") + 
  ggtitle("Regulatory T cell")
Tregfreq
``` 





















``` {r}
sessionInfo()
``` 



















